// @scalar/api-reference/dist/components/ShowMoreButton.vue2.js
import { defineComponent, useTemplateRef, createElementBlock, openBlock, createTextVNode, createVNode, unref } from "vue";
import { ScalarIconCaretDown } from "@scalar/icons";
import { emitCustomEvent } from "@scalar/workspace-store/events";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ShowMoreButton",
  props: {
    id: {}
  },
  setup(__props) {
    const el = useTemplateRef("el");
    const handleClick = () => {
      emitCustomEvent(el.value, "scalar-on-show-more", { id: __props.id });
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("button", {
        ref_key: "el",
        ref: el,
        class: "show-more",
        type: "button",
        onClick: handleClick
      }, [
        _cache[0] || (_cache[0] = createTextVNode(" Show More ", -1)),
        createVNode(unref(ScalarIconCaretDown), {
          class: "text-c-2 mt-0.25 size-3",
          weight: "bold"
        })
      ], 512);
    };
  }
});
export {
  _sfc_main as default
};
