// @scalar/api-client/dist/v2/features/modal/Modal.vue2.js
import { defineComponent as k, ref as m, useId as h, watch as b, nextTick as E, onBeforeMount as y, onBeforeUnmount as g, withDirectives as u, createElementBlock as T, openBlock as x, createElementVNode as o, unref as r, createVNode as l, withCtx as C, vShow as S } from "vue";
import { addScalarClassesToHeadless as F, ScalarTeleportRoot as O } from "@scalar/components";
import { useFocusTrap as B } from "@vueuse/integrations/useFocusTrap";
import U from "../../components/TempReplaceMe.vue.js";
import { useSidebarState as L } from "../../hooks/use-sidebar-state.js";
import M from "../../components/sidebar/Sidebar.vue.js";
const N = { class: "scalar scalar-app" }, W = { class: "scalar-container" }, _ = ["id"], $ = { class: "flex flex-1 flex-row" }, I = {}, j = /* @__PURE__ */ k({
  ...I,
  __name: "Modal",
  props: {
    activeEntities: {},
    workspaceStore: {},
    modalState: {}
  },
  setup(s) {
    const n = m(null), d = h(), { activate: f, deactivate: w } = B(n, {
      allowOutsideClick: !0,
      fallbackFocus: `#${d}`
    }), { sidebarState: v } = L({
      workspaceStore: s.workspaceStore,
      workspaceSlug: "default",
      documentSlug: "",
      exampleName: "",
      method: "get",
      path: ""
    }), c = (e) => e.key === "Escape" && s.modalState.hide(), p = () => {
      window.removeEventListener("keydown", c), document.documentElement.style.removeProperty("overflow"), w();
    };
    b(
      () => s.modalState.open,
      (e) => {
        e ? (window.addEventListener("keydown", c), document.documentElement.style.overflow = "hidden", f({ checkCanFocusTrap: () => E() })) : p();
      }
    ), y(() => F()), g(() => {
      p();
    });
    const i = m(!0);
    return (e, t) => u((x(), T("div", N, [
      o("div", W, [
        o("div", {
          id: r(d),
          ref_key: "client",
          ref: n,
          "aria-label": "API Client",
          "aria-modal": "true",
          class: "scalar-app-layout scalar-client flex",
          role: "dialog",
          tabindex: "-1"
        }, [
          l(r(O), null, {
            default: C(() => [
              o("main", $, [
                u(l(r(M), {
                  isSidebarOpen: i.value,
                  "onUpdate:isSidebarOpen": t[0] || (t[0] = (a) => i.value = a),
                  sidebarState: r(v),
                  layout: "modal",
                  sidebarWidth: e.workspaceStore.workspace["x-scalar-sidebar-width"] ?? 288,
                  "onUpdate:sidebarWidth": t[1] || (t[1] = (a) => e.workspaceStore.update("x-scalar-sidebar-width", a))
                }, null, 8, ["isSidebarOpen", "sidebarState", "sidebarWidth"]), [
                  [S, i.value]
                ]),
                l(U, {
                  documentSlug: e.activeEntities.documentSlug,
                  example: e.activeEntities.example,
                  layout: "modal",
                  method: e.activeEntities.method,
                  path: e.activeEntities.path,
                  workspaceStore: e.workspaceStore,
                  workspaceSlug: "default"
                }, null, 8, ["documentSlug", "example", "method", "path", "workspaceStore"])
              ])
            ]),
            _: 1
          })
        ], 8, _),
        o("div", {
          class: "scalar-app-exit",
          onClick: t[2] || (t[2] = (a) => e.modalState.hide())
        })
      ])
    ], 512)), [
      [S, e.modalState.open]
    ]);
  }
});
export {
  j as default
};
