// @scalar/api-client/dist/v2/components/sidebar/Sidebar.vue.js
import { defineComponent as v, mergeModels as c, useModel as n, ref as g, createBlock as a, openBlock as r, unref as s, withCtx as l, createVNode as M, renderSlot as b, createElementVNode as S, createCommentVNode as u } from "vue";
import { ScalarIconButton as O, ScalarSidebarSearchInput as V } from "@scalar/components";
import { ScalarIconMagnifyingGlass as I } from "@scalar/icons";
import { ScalarSidebar as B } from "@scalar/sidebar";
import W from "./SidebarMenu.vue.js";
import $ from "./SidebarToggle.vue.js";
import h from "../resize/Resize.vue.js";
const x = { class: "bg-sidebar-b-1 sticky top-0 z-1 flex flex-col gap-3 px-3 pt-3" }, C = { class: "flex items-center justify-between" }, R = /* @__PURE__ */ v({
  __name: "Sidebar",
  props: /* @__PURE__ */ c({
    sidebarState: {},
    layout: {}
  }, {
    workspace: {
      default: "default"
    },
    workspaceModifiers: {},
    isSidebarOpen: { type: Boolean, required: !0 },
    isSidebarOpenModifiers: {},
    sidebarWidth: {
      required: !0,
      default: 288
    },
    sidebarWidthModifiers: {}
  }),
  emits: /* @__PURE__ */ c(["selectItem"], ["update:workspace", "update:isSidebarOpen", "update:sidebarWidth"]),
  setup(i, { emit: k }) {
    const w = k, y = (t, ...e) => {
      console.log("[LOG] event name: ", t), console.log("[LOG]", ...e);
    }, p = n(i, "workspace"), m = n(i, "isSidebarOpen"), d = g(!1), f = n(i, "sidebarWidth");
    return (t, e) => (r(), a(s(h), {
      width: f.value,
      "onUpdate:width": e[5] || (e[5] = (o) => f.value = o),
      class: "flex flex-col"
    }, {
      default: l(() => [
        M(s(B), {
          class: "flex w-auto flex-1 pt-2",
          isExpanded: t.sidebarState.isExpanded,
          isSelected: t.sidebarState.isSelected,
          items: t.sidebarState.items.value,
          layout: "client",
          onSelectItem: e[3] || (e[3] = (o) => w("selectItem", o)),
          onReorder: e[4] || (e[4] = (...o) => y("reorder", ...o))
        }, {
          search: l(() => [
            S("div", x, [
              S("div", C, [
                t.layout === "desktop" ? (r(), a(W, {
                  key: 0,
                  workspace: p.value,
                  "onUpdate:workspace": e[0] || (e[0] = (o) => p.value = o)
                }, null, 8, ["workspace"])) : t.layout === "modal" ? (r(), a($, {
                  key: 1,
                  modelValue: m.value,
                  "onUpdate:modelValue": e[1] || (e[1] = (o) => m.value = o)
                }, null, 8, ["modelValue"])) : u("", !0),
                t.layout !== "web" ? (r(), a(s(O), {
                  key: 2,
                  icon: s(I),
                  label: "Search",
                  onClick: e[2] || (e[2] = (o) => d.value = !d.value)
                }, null, 8, ["icon"])) : u("", !0)
              ]),
              d.value || t.layout === "web" ? (r(), a(s(V), {
                key: 0,
                autofocus: t.layout !== "web"
              }, null, 8, ["autofocus"])) : u("", !0)
            ])
          ]),
          before: l(() => [
            b(t.$slots, "workspaceButton")
          ]),
          footer: l(() => [
            b(t.$slots, "footer")
          ]),
          _: 3
        }, 8, ["isExpanded", "isSelected", "items"])
      ]),
      _: 3
    }, 8, ["width"]));
  }
});
export {
  R as default
};
