// @scalar/api-client/dist/v2/components/sidebar/SidebarMenu.vue.js
import { defineComponent as d, mergeModels as k, computed as m, useModel as f, createBlock as M, openBlock as S, unref as o, withCtx as r, createVNode as t, createTextVNode as n } from "vue";
import { ScalarMenu as w, ScalarMenuSection as W, ScalarMenuWorkspacePicker as C, ScalarMenuLink as V, ScalarMenuResources as v, ScalarMenuSupport as x } from "@scalar/components";
import { ScalarIconGear as b } from "@scalar/icons";
import { RouterLink as g } from "vue-router";
const T = /* @__PURE__ */ d({
  __name: "SidebarMenu",
  props: {
    workspace: {
      required: !0,
      default: "default"
    },
    workspaceModifiers: {}
  },
  emits: /* @__PURE__ */ k(["createWorkspace"], ["update:workspace"]),
  setup(s, { emit: u }) {
    const i = u, p = m(() => [
      {
        id: "default",
        label: "Default Workspace"
      },
      {
        id: "fake",
        label: "Fake Workspace"
      }
    ]), a = f(s, "workspace");
    return (O, e) => (S(), M(o(w), null, {
      products: r(() => [...e[2] || (e[2] = [])]),
      sections: r(({ close: c }) => [
        t(o(W), null, {
          title: r(() => [...e[3] || (e[3] = [
            n("Team", -1)
          ])]),
          default: r(() => [
            t(o(C), {
              modelValue: a.value,
              "onUpdate:modelValue": e[0] || (e[0] = (l) => a.value = l),
              workspaceOptions: p.value,
              onCreateWorkspace: e[1] || (e[1] = (l) => i("createWorkspace"))
            }, null, 8, ["modelValue", "workspaceOptions"]),
            t(o(V), {
              is: o(g),
              icon: o(b),
              to: "/settings",
              onClick: c
            }, {
              default: r(() => [...e[4] || (e[4] = [
                n(" Settings ", -1)
              ])]),
              _: 1
            }, 8, ["is", "icon", "onClick"])
          ]),
          _: 2
        }, 1024),
        t(o(v)),
        t(o(x))
      ]),
      _: 1
    }));
  }
});
export {
  T as default
};
