// @scalar/api-client/dist/views/Request/RequestRoot.vue2.js
import { defineComponent as $, ref as c, computed as C, onMounted as H, onBeforeUnmount as Q, createElementBlock as F, openBlock as m, normalizeClass as N, unref as o, createBlock as U, createCommentVNode as _, createElementVNode as O, isRef as G, createVNode as K } from "vue";
import { isDefined as X } from "@scalar/oas-utils/helpers";
import { safeJSON as Y } from "@scalar/object-utils/parse";
import { useToasts as Z } from "@scalar/use-toasts";
import { RouterView as ee } from "vue-router";
import te from "../../components/Sidebar/SidebarToggle.vue.js";
import { useAnalytics as oe } from "../../hooks/useAnalytics.js";
import { useClientConfig as re } from "../../hooks/useClientConfig.js";
import { useSidebar as se } from "../../hooks/useSidebar.js";
import { validateParameters as ae } from "../../libs/validate-parameters.js";
import { useActiveEntities as ne } from "../../store/active-entities.js";
import { useOpenApiWatcher as le } from "./hooks/useOpenApiWatcher.js";
import ue from "./RequestSidebar.vue.js";
import { usePluginManager as ie } from "../../plugins/hooks/usePluginManager.js";
import { createRequestOperation as ce } from "../../libs/send-request/create-request-operation.js";
import { ERRORS as me } from "../../libs/errors.js";
import { useWorkspace as fe } from "../../store/store.js";
import { useLayout as de } from "../../hooks/useLayout.js";
const pe = { class: "flex h-full" }, ve = { class: "flex h-full flex-1 flex-col" }, Te = /* @__PURE__ */ $({
  __name: "RequestRoot",
  emits: ["newTab"],
  setup(Re) {
    const V = fe(), { toast: u } = Z(), { layout: f } = de(), w = re(), { isSidebarOpen: s } = se(), B = oe(), {
      activeCollection: n,
      activeExample: i,
      activeEnvironment: d,
      activeRequest: l,
      activeWorkspace: p,
      activeServer: A
    } = ne(), { cookies: T, requestHistory: P, showSidebar: v, securitySchemes: D, events: r } = V, M = ie(), z = c(), R = c(), S = C(
      () => ae(i.value ?? null)
    ), q = c(null), y = C(
      () => (n.value?.useCollectionSecurity ? n.value?.selectedSecuritySchemeUids : l.value?.selectedSecuritySchemeUids) ?? []
    ), x = async () => {
      if (!l.value || !i.value || !n.value)
        return;
      if (S.value.hasBlockingErrors) {
        u("Path parameters must have values.", "error"), r.requestStatus.emit("abort");
        return;
      }
      const t = typeof d.value == "object" ? d.value.value : "{}", e = Y.parse(t);
      e.error && console.error("INVALID ENVIRONMENT!");
      const a = e.error || typeof e.data != "object" ? {} : e.data ?? {}, W = p.value?.cookies.map((L) => T[L]).filter(X) ?? [], j = n.value?.info?.title === "Drafts" ? void 0 : A.value, [h, g] = ce({
        request: l.value,
        example: i.value,
        selectedSecuritySchemeUids: y.value,
        proxyUrl: p.value?.proxyUrl ?? "",
        environment: a,
        globalCookies: W,
        status: r.requestStatus,
        securitySchemes: D,
        server: j,
        pluginManager: M
      });
      if (w.value?.onRequestSent?.(l.value.path ?? ""), h) {
        u(h.message, "error");
        return;
      }
      R.value = g.controller;
      const [k, E] = await g.sendRequest();
      q.value = E, k ? u(k.message, "error") : P.push(J(E));
    }, I = async () => R.value?.abort(me.REQUEST_ABORTED);
    function b() {
      B?.capture("client-send-request");
    }
    H(() => {
      r.executeRequest.on(x), r.executeRequest.on(b), r.cancelRequest.on(I);
    }), le(), Q(() => {
      r.executeRequest.off(x), r.executeRequest.off(b);
    });
    const J = (t) => {
      try {
        return structuredClone(t);
      } catch {
        const e = { ...t };
        return t.response?.data && (t.response.data instanceof Blob || t.response.data instanceof ArrayBuffer ? e.response.data = t.response.data : e.response.data = JSON.parse(JSON.stringify(t.response.data))), e;
      }
    };
    return (t, e) => (m(), F("div", {
      ref_key: "element",
      ref: z,
      class: N(["bg-b-1 relative z-0 flex h-full flex-1 flex-col overflow-hidden pt-0", {
        "!mr-0 !mb-0 !border-0": o(f) === "modal"
      }])
    }, [
      o(v) ? (m(), U(te, {
        key: 0,
        modelValue: o(s),
        "onUpdate:modelValue": e[0] || (e[0] = (a) => G(s) ? s.value = a : null),
        class: N(["absolute top-2 left-3 z-50", [
          { hidden: o(s) },
          { "xl:!flex": !o(s) },
          { "!flex": o(f) === "modal" }
        ]])
      }, null, 8, ["modelValue", "class"])) : _("", !0),
      O("div", pe, [
        o(v) ? (m(), U(ue, {
          key: 0,
          onNewTab: e[1] || (e[1] = (a) => t.$emit("newTab", a))
        })) : _("", !0),
        O("div", ve, [
          K(o(ee), {
            invalidParams: S.value.invalidParams,
            requestResult: q.value,
            selectedSecuritySchemeUids: y.value
          }, null, 8, ["invalidParams", "requestResult", "selectedSecuritySchemeUids"])
        ])
      ])
    ], 2));
  }
});
export {
  Te as default
};
