// @scalar/api-client/dist/v2/features/collection/components/Authentication.vue2.js
import { defineComponent as u, computed as l, createElementBlock as m, openBlock as d, createElementVNode as n, createVNode as i, unref as c, normalizeClass as a } from "vue";
import { ScalarToggle as p } from "@scalar/components";
import v from "../../../blocks/scalar-auth-selector-block/components/AuthSelector.vue.js";
const y = { class: "flex flex-col gap-4" }, f = { class: "flex flex-col gap-2" }, h = { class: "flex h-8 items-center justify-between" }, x = /* @__PURE__ */ u({
  __name: "Authentication",
  props: {
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    workspaceStore: {},
    collectionType: {}
  },
  setup(o) {
    const s = l(
      () => o.document?.["x-scalar-document-security"] ?? !1
    ), r = l(
      () => o.document?.servers?.find(
        ({ url: e }) => e === o.document?.["x-scalar-selected-server"]
      )
    );
    return (e, t) => (d(), m("div", y, [
      n("div", f, [
        n("div", h, [
          t[1] || (t[1] = n("h3", { class: "font-bold" }, "Authentication", -1)),
          i(c(p), {
            class: "w-4",
            modelValue: s.value,
            "onUpdate:modelValue": t[0] || (t[0] = () => e.eventBus.emit("document:toggle:document-security"))
          }, null, 8, ["modelValue"])
        ]),
        t[2] || (t[2] = n("p", { class: "pr-6 text-sm" }, " If enabled, all selected authentication will apply to all operations in this document. You can override this by disabling the toggle and authentication will then be applied at the operation level. ", -1))
      ]),
      n("div", {
        class: a(!s.value && "cursor-not-allowed")
      }, [
        i(c(v), {
          class: a([
            "scalar-collection-auth !border-none",
            !s.value && "pointer-events-none opacity-50 mix-blend-luminosity"
          ]),
          environment: e.environment,
          eventBus: e.eventBus,
          isStatic: "",
          meta: { type: "document" },
          security: e.document?.security ?? [],
          securitySchemes: e.document?.components?.securitySchemes ?? {},
          selectedSecurity: e.document?.["x-scalar-selected-security"],
          server: r.value,
          title: "Authentication"
        }, null, 8, ["class", "environment", "eventBus", "security", "securitySchemes", "selectedSecurity", "server"])
      ], 2)
    ]));
  }
});
export {
  x as default
};
