// @scalar/api-client/dist/v2/features/collection/components/Environment.vue2.js
import { defineComponent as s, computed as c, createElementBlock as l, openBlock as a, createStaticVNode as i, createVNode as r, unref as m } from "vue";
import v from "../../environments/EnvironmentsList.vue.js";
const d = { class: "flex flex-col gap-4" }, f = /* @__PURE__ */ s({
  __name: "Environment",
  props: {
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    workspaceStore: {},
    collectionType: {}
  },
  setup(e) {
    const o = c(
      () => (e.collectionType === "document" ? e.document["x-scalar-environments"] : e.workspaceStore.workspace["x-scalar-environments"]) ?? {}
    );
    return (t, n) => (a(), l("div", d, [
      n[0] || (n[0] = i('<div class="flex items-start justify-between gap-2"><div class="flex flex-col gap-2"><div class="flex h-8 items-center"><h3 class="font-bold">Environment Variables</h3></div><p class="text-c-2 mb-4 text-sm"> Set environment variables at your collection level. Use <code class="font-code text-c-2"> {{ variable }} </code> to add / search among the selected environment&#39;s variables in your request inputs. </p></div></div>', 1)),
      r(m(v), {
        collectionType: t.collectionType,
        environments: o.value,
        eventBus: t.eventBus
      }, null, 8, ["collectionType", "environments", "eventBus"])
    ]));
  }
});
export {
  f as default
};
