// @scalar/api-client/dist/v2/features/collection/components/Servers.vue2.js
import { defineComponent as M, ref as A, computed as x, createElementBlock as v, openBlock as c, Fragment as S, createElementVNode as r, createVNode as l, createTextVNode as g, renderList as I, createBlock as y, unref as s, toDisplayString as T, withCtx as f, createCommentVNode as V } from "vue";
import { useModal as P, ScalarMarkdown as z, ScalarButton as _, ScalarModal as L } from "@scalar/components";
import { debounce as O } from "@scalar/helpers/general/debounce";
import { ScalarIconTrash as R, ScalarIconPlus as j } from "@scalar/icons";
import E from "../../../../components/Sidebar/Actions/DeleteSidebarListElement.vue.js";
import F from "./Form.vue.js";
import W from "../../../../components/Server/ServerVariablesForm.vue.js";
const Y = { class: "flex flex-col gap-4" }, q = { class: "flex flex-col gap-4" }, G = { class: "bg-b-2 flex items-center justify-between rounded-t-lg px-3 py-1 text-sm" }, H = {
  key: 1,
  class: "self-center"
}, J = { class: "divide-0 flex w-full flex-col divide-y rounded-b-lg text-sm" }, K = { class: "text-c-3 flex h-full items-center justify-center rounded-lg border p-4" }, se = /* @__PURE__ */ M({
  __name: "Servers",
  props: {
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    workspaceStore: {},
    collectionType: {}
  },
  setup(o) {
    const u = P(), n = A(-1), k = x(
      () => o.document?.servers?.[n.value]
    ), $ = x(() => o.document?.servers ?? []), B = [
      {
        label: "URL",
        key: "url",
        placeholder: "https://void.scalar.com"
      },
      {
        label: "Description",
        key: "description",
        placeholder: "Production"
      }
    ], w = (t) => {
      n.value = t, u.show();
    }, p = () => {
      u.hide(), n.value = -1;
    }, N = () => {
      n.value < 0 || (o.eventBus.emit("server:delete:server", { index: n.value }), p());
    }, { execute: b } = O({ delay: 328, maxWait: 1e3 }), U = (t, e, a) => b(
      `${t}-${e}`,
      () => o.eventBus.emit("server:update:server", {
        index: t,
        server: { [e]: a }
      })
    ), C = (t, e, a) => b(
      `${t}-${e}`,
      () => o.eventBus.emit("server:update:variables", {
        index: t,
        key: e,
        value: a
      })
    ), D = () => o.eventBus.emit("server:add:server"), h = (t, e = 0) => t?.description || `Server ${e + 1}`;
    return (t, e) => (c(), v(S, null, [
      r("div", Y, [
        e[1] || (e[1] = r("div", { class: "flex flex-col gap-2" }, [
          r("h3", { class: "font-bold" }, "Servers"),
          r("p", { class: "text-sm" }, [
            g(" Add different base URLs for your API. You can use "),
            r("code", { class: "font-code text-c-2" }, "{variables}"),
            g(" for dynamic parts. ")
          ])
        ], -1)),
        r("div", q, [
          (c(!0), v(S, null, I($.value, (a, d) => (c(), v("div", {
            key: d,
            class: "rounded-lg border"
          }, [
            r("div", G, [
              a.description ? (c(), y(s(z), {
                key: 0,
                class: "self-center",
                value: a.description
              }, null, 8, ["value"])) : (c(), v("span", H, T(h(a, d)), 1)),
              l(s(_), {
                class: "hover:bg-b-3 hover:text-c-1 h-fit p-1.25",
                "data-testid": "delete-server-button",
                variant: "ghost",
                onClick: (i) => w(d)
              }, {
                default: f(() => [
                  l(s(R), { class: "size-3.5" })
                ]),
                _: 1
              }, 8, ["onClick"])
            ]),
            r("div", J, [
              l(F, {
                data: a,
                environment: t.environment,
                onUpdate: (i, m) => U(d, i, m),
                options: B
              }, null, 8, ["data", "environment", "onUpdate"]),
              a.variables ? (c(), y(s(W), {
                key: 0,
                variables: a.variables,
                "onUpdate:variable": (i, m) => C(d, i, m)
              }, null, 8, ["variables", "onUpdate:variable"])) : V("", !0)
            ])
          ]))), 128))
        ]),
        r("div", K, [
          l(s(_), {
            class: "hover:bg-b-2 hover:text-c-1 flex items-center gap-2",
            size: "sm",
            variant: "ghost",
            onClick: D
          }, {
            default: f(() => [
              l(s(j)),
              e[0] || (e[0] = r("span", null, "Add Server", -1))
            ]),
            _: 1
          })
        ])
      ]),
      l(s(L), {
        size: "xxs",
        state: s(u),
        title: `Delete ${h(k.value, n.value)}`
      }, {
        default: f(() => [
          l(E, {
            variableName: "Server",
            warningMessage: "Are you sure you want to delete this server? This action cannot be undone.",
            onClose: p,
            onDelete: N
          })
        ]),
        _: 1
      }, 8, ["state", "title"])
    ], 64));
  }
});
export {
  se as default
};
