// @scalar/api-client/dist/v2/features/collection/components/Settings.vue2.js
import { defineComponent as b, createElementBlock as a, openBlock as i, Fragment as r, createElementVNode as t, createVNode as n, unref as l, createTextVNode as c, toDisplayString as w, withCtx as m } from "vue";
import { useModal as h, ScalarToggle as v, ScalarIcon as p, ScalarButton as x, ScalarModal as y } from "@scalar/components";
import D from "../../../../components/Sidebar/Actions/DeleteSidebarListElement.vue.js";
const M = { class: "flex flex-col gap-2" }, k = { class: "bg-b-2 rounded-lg border text-sm" }, U = { class: "bg-b-1 flex items-center justify-between gap-4 rounded-t-lg p-3" }, N = { class: "text-c-1 flex items-center overflow-x-auto border-t py-1.5 whitespace-nowrap" }, S = { class: "flex items-center" }, C = ["href"], V = { class: "flex flex-col gap-4" }, B = { class: "flex items-center justify-between rounded-lg border p-3 text-sm" }, I = /* @__PURE__ */ b({
  __name: "Settings",
  props: {
    documentUrl: {},
    watchMode: { type: Boolean },
    title: {}
  },
  emits: ["deleteDocument", "update:watchMode"],
  setup(T, { emit: f }) {
    const u = f, s = h(), g = () => {
      u("deleteDocument"), s.hide();
    };
    return (o, e) => (i(), a(r, null, [
      t("div", M, [
        e[6] || (e[6] = t("div", { class: "flex h-8 items-center" }, [
          t("h3", { class: "font-bold" }, "Features")
        ], -1)),
        t("div", k, [
          t("div", U, [
            e[3] || (e[3] = t("div", null, [
              t("h4", null, "Watch Mode"),
              t("p", { class: "text-c-2 mt-1" }, " When enabled, the OpenAPI document will be polled for changes. The collection will be updated automatically. ")
            ], -1)),
            n(l(v), {
              class: "w-4",
              disabled: !o.documentUrl,
              modelValue: o.watchMode ?? !1,
              "onUpdate:modelValue": e[0] || (e[0] = (d) => u("update:watchMode", d))
            }, null, 8, ["disabled", "modelValue"])
          ]),
          t("div", N, [
            t("div", S, [
              o.documentUrl ? (i(), a(r, { key: 0 }, [
                e[4] || (e[4] = t("span", { class: "bg-b-2 sticky left-0 pr-2 pl-3" }, "Source", -1)),
                t("a", {
                  class: "text-c-2 group rounded pr-3 no-underline hover:underline",
                  href: o.documentUrl,
                  target: "_blank"
                }, [
                  c(w(o.documentUrl) + " ", 1),
                  n(l(p), {
                    class: "ml-1 hidden w-2.5 group-hover:inline",
                    icon: "ExternalLink"
                  })
                ], 8, C)
              ], 64)) : (i(), a(r, { key: 1 }, [
                n(l(p), {
                  class: "text-c-2 mr-2 ml-3 w-4",
                  icon: "NotAllowed",
                  size: "sm"
                }),
                e[5] || (e[5] = t("span", { class: "text-c-2 pr-3" }, " No URL configured. Try importing an OpenAPI document from an URL. ", -1))
              ], 64))
            ])
          ])
        ])
      ]),
      t("div", V, [
        e[9] || (e[9] = t("h3", { class: "font-bold" }, "Danger Zone", -1)),
        t("div", B, [
          e[8] || (e[8] = t("div", null, [
            t("h4", null, "Delete Collection"),
            t("p", { class: "text-c-2 mt-1" }, " Be careful, my friend. Once deleted, there is no way to recover the collection. ")
          ], -1)),
          n(l(x), {
            class: "custom-scroll h-8 gap-1.5 px-2.5 font-medium whitespace-nowrap shadow-none focus:outline-none",
            variant: "danger",
            onClick: e[1] || (e[1] = (d) => l(s).show())
          }, {
            default: m(() => [...e[7] || (e[7] = [
              c(" Delete Collection ", -1)
            ])]),
            _: 1
          })
        ])
      ]),
      n(l(y), {
        size: "xxs",
        state: l(s),
        title: `Delete ${o.title}`
      }, {
        default: m(() => [
          n(D, {
            variableName: o.title ?? "",
            warningMessage: "This action cannot be undone.",
            onClose: e[2] || (e[2] = (d) => l(s).hide()),
            onDelete: g
          }, null, 8, ["variableName"])
        ]),
        _: 1
      }, 8, ["state", "title"])
    ], 64));
  }
});
export {
  I as default
};
