// @scalar/api-reference/dist/blocks/scalar-auth-selector-block/components/AuthSelector.vue2.js
import { defineComponent, computed, useId, ref, useTemplateRef, createBlock, createCommentVNode, unref, openBlock, withCtx, createVNode, createElementVNode, createElementBlock, Fragment, createTextVNode, toDisplayString, withModifiers, normalizeClass } from "vue";
import { ViewLayoutCollapse } from "@scalar/api-client/components/ViewLayout";
import { useLayout } from "@scalar/api-client/hooks";
import { useWorkspace, useActiveEntities } from "@scalar/api-client/store";
import { getSecurityRequirements, formatComplexScheme, formatScheme, getSchemeOptions } from "@scalar/api-client/views/Request/libs";
import { useModal, ScalarComboboxMultiselect, ScalarButton, ScalarListboxCheckbox, ScalarIconButton } from "@scalar/components";
import { safeLocalStorage, CLIENT_LS_KEYS } from "@scalar/helpers/object/local-storage";
import { ScalarIconCaretDown, ScalarIconTrash } from "@scalar/icons";
import { isDefined } from "@scalar/oas-utils/helpers";
import { emitCustomEvent } from "@scalar/workspace-store/events";
import _sfc_main$1 from "./DeleteRequestAuthModal.vue.js";
import RequestAuthDataTable from "./RequestAuthDataTable.vue.js";
const _hoisted_1 = ["id"];
const _hoisted_2 = { class: "flex flex-1" };
const _hoisted_3 = { class: "min-w-0 flex-1 truncate" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "AuthSelector",
  props: {
    layout: {},
    operation: {},
    persistAuth: { type: Boolean, default: false },
    server: {},
    title: {}
  },
  setup(__props) {
    const { layout: clientLayout } = useLayout();
    const { securitySchemes } = useWorkspace();
    const {
      activeWorkspace: workspace,
      activeCollection: collection,
      activeEnvVariables: envVariables,
      activeEnvironment: environment
    } = useActiveEntities();
    const selectedSecuritySchemeUids = computed(
      () => collection.value?.selectedSecuritySchemeUids ?? []
    );
    const titleId = useId();
    const comboboxButtonRef = ref(null);
    const deleteSchemeModal = useModal();
    const selectedScheme = ref(
      null
    );
    const isViewLayoutOpen = ref(false);
    const securityRequirements = computed(() => {
      const requirements = getSecurityRequirements(__props.operation, collection.value);
      const filteredRequirements = requirements.filter((r) => Object.keys(r).length);
      return { filteredRequirements, requirements };
    });
    const authIndicator = computed(() => {
      const { filteredRequirements, requirements } = securityRequirements.value;
      if (!requirements.length) {
        return null;
      }
      const hasComplexRequirement = requirements.some(
        (req) => Object.keys(req).length > 1
      );
      const isOptional = !hasComplexRequirement && filteredRequirements.length < requirements.length;
      const icon = isOptional ? "Unlock" : "Lock";
      const text = isOptional ? "Optional" : "Required";
      return { icon, text };
    });
    const selectedSchemeOptions = computed(
      () => selectedSecuritySchemeUids.value.map((s) => {
        if (Array.isArray(s)) {
          return formatComplexScheme(s, securitySchemes);
        }
        const scheme = securitySchemes[s ?? ""];
        if (!scheme) {
          return void 0;
        }
        return formatScheme(scheme);
      }).filter(isDefined)
    );
    function updateSelectedAuth(entries) {
      const addNewOption = entries.find((e) => e.payload);
      const _entries = entries.filter((e) => !e.payload).map(({ id }) => {
        const arr = id.split(",");
        return arr.length > 1 ? arr : id;
      });
      if (addNewOption?.payload) {
        emitCustomEvent(wrapperRef.value?.$el, "scalar-add-auth-option", {
          payload: addNewOption.payload
        });
        if (addNewOption.payload.uid) {
          _entries.push(addNewOption.payload.uid);
        }
      }
      editSelectedSchemeUids(_entries);
    }
    const editSelectedSchemeUids = (uids) => {
      if (collection.value?.useCollectionSecurity) {
        emitCustomEvent(wrapperRef.value?.$el, "scalar-select-security-schemes", {
          uids
        });
        if (!__props.persistAuth) {
          return;
        }
        const nameKeys = uids.map((u) => {
          if (Array.isArray(u)) {
            return u.map((uid) => securitySchemes[uid]?.nameKey);
          }
          return securitySchemes[u]?.nameKey;
        });
        safeLocalStorage().setItem(
          CLIENT_LS_KEYS.SELECTED_SECURITY_SCHEMES,
          JSON.stringify(nameKeys)
        );
      } else if (__props.operation?.uid) {
        emitCustomEvent(
          wrapperRef.value?.$el,
          "scalar-select-operation-security-schemes",
          {
            operationUid: __props.operation.uid,
            uids
          }
        );
      }
    };
    function handleDeleteScheme({ id, label }) {
      selectedScheme.value = { id, label };
      deleteSchemeModal.show();
    }
    const unselectAuth = (unSelectUid) => {
      if (!unSelectUid) {
        return;
      }
      const newUids = selectedSecuritySchemeUids.value.filter((uid) => {
        const arr = unSelectUid.split(",");
        if (arr.length > 1 && Array.isArray(uid) && arr.length === uid.length) {
          return uid.every((u) => !arr.includes(u));
        }
        return uid !== unSelectUid;
      });
      editSelectedSchemeUids(newUids);
      comboboxButtonRef.value?.$el.focus();
      deleteSchemeModal.hide();
    };
    const schemeOptions = computed(
      () => getSchemeOptions(
        securityRequirements.value.filteredRequirements,
        collection.value?.securitySchemes ?? [],
        securitySchemes,
        clientLayout === "modal" || __props.layout === "reference"
      )
    );
    const openAuthCombobox = (event) => {
      if (isViewLayoutOpen.value) {
        event.stopPropagation();
      }
      comboboxButtonRef.value?.$el.click();
    };
    const wrapperRef = useTemplateRef("wrapperRef");
    return (_ctx, _cache) => {
      return unref(collection) && unref(workspace) ? (openBlock(), createBlock(unref(ViewLayoutCollapse), {
        key: 0,
        ref_key: "wrapperRef",
        ref: wrapperRef,
        class: "group/params relative",
        itemCount: selectedSchemeOptions.value.length,
        layout: _ctx.layout,
        "onUpdate:modelValue": _cache[2] || (_cache[2] = ($event) => isViewLayoutOpen.value = $event)
      }, {
        title: withCtx(() => [
          createElementVNode("div", {
            id: unref(titleId),
            class: "inline-flex items-center gap-0.5 leading-[20px]"
          }, [
            createElementVNode("span", null, toDisplayString(_ctx.title), 1),
            authIndicator.value ? (openBlock(), createElementBlock("span", {
              key: 0,
              class: normalizeClass(["text-c-3 hover:bg-b-3 hover:text-c-1 -mr-1 cursor-pointer rounded px-1 py-0.5 text-xs leading-[normal]", { "text-c-1": authIndicator.value.text === "Required" }]),
              onClick: openAuthCombobox
            }, toDisplayString(authIndicator.value.text), 3)) : createCommentVNode("", true)
          ], 8, _hoisted_1)
        ]),
        actions: withCtx(() => [
          createElementVNode("div", _hoisted_2, [
            createVNode(unref(ScalarComboboxMultiselect), {
              class: "w-72 text-xs",
              modelValue: selectedSchemeOptions.value,
              multiple: "",
              options: schemeOptions.value,
              placement: "bottom-end",
              teleport: "",
              onDelete: handleDeleteScheme,
              "onUpdate:modelValue": updateSelectedAuth
            }, {
              option: withCtx(({ option, selected }) => [
                createVNode(unref(ScalarListboxCheckbox), {
                  multiselect: "",
                  selected
                }, null, 8, ["selected"]),
                createElementVNode("div", _hoisted_3, toDisplayString(option.label), 1),
                option.isDeletable ?? (unref(clientLayout) !== "modal" && _ctx.layout !== "reference") ? (openBlock(), createBlock(unref(ScalarIconButton), {
                  key: 0,
                  class: "-m-0.5 shrink-0 p-0.5 opacity-0 group-hover/item:opacity-100",
                  icon: unref(ScalarIconTrash),
                  label: `Delete ${option.label}`,
                  size: "xs",
                  onClick: withModifiers(($event) => handleDeleteScheme(option), ["stop"])
                }, null, 8, ["icon", "label", "onClick"])) : createCommentVNode("", true)
              ]),
              default: withCtx(() => [
                createVNode(unref(ScalarButton), {
                  ref_key: "comboboxButtonRef",
                  ref: comboboxButtonRef,
                  "aria-describedby": unref(titleId),
                  class: "group/combobox-button hover:text-c-1 text-c-2 flex h-fit w-full items-center gap-1 px-0.75 py-0.25 text-base font-normal transition-transform",
                  variant: "ghost"
                }, {
                  default: withCtx(() => [
                    selectedSchemeOptions.value.length === 1 ? (openBlock(), createElementBlock(Fragment, { key: 0 }, [
                      _cache[3] || (_cache[3] = createElementVNode("span", { class: "sr-only" }, "Selected Auth Type:", -1)),
                      createTextVNode(" " + toDisplayString(selectedSchemeOptions.value[0]?.label), 1)
                    ], 64)) : selectedSchemeOptions.value.length > 1 ? (openBlock(), createElementBlock(Fragment, { key: 1 }, [
                      _cache[4] || (_cache[4] = createTextVNode(" Multiple ", -1)),
                      _cache[5] || (_cache[5] = createElementVNode("span", { class: "sr-only" }, "Auth Types Selected", -1))
                    ], 64)) : (openBlock(), createElementBlock(Fragment, { key: 2 }, [
                      _cache[6] || (_cache[6] = createElementVNode("span", { class: "sr-only" }, "Select", -1)),
                      _cache[7] || (_cache[7] = createTextVNode(" Auth Type ", -1))
                    ], 64)),
                    createVNode(unref(ScalarIconCaretDown), {
                      class: "size-3 shrink-0 transition-transform duration-100 group-aria-expanded/combobox-button:rotate-180",
                      weight: "bold"
                    })
                  ]),
                  _: 1
                }, 8, ["aria-describedby"])
              ]),
              _: 1
            }, 8, ["modelValue", "options"])
          ])
        ]),
        default: withCtx(() => [
          createVNode(RequestAuthDataTable, {
            collection: unref(collection),
            envVariables: unref(envVariables),
            environment: unref(environment),
            layout: _ctx.layout,
            persistAuth: _ctx.persistAuth,
            selectedSchemeOptions: selectedSchemeOptions.value,
            server: _ctx.server,
            workspace: unref(workspace)
          }, null, 8, ["collection", "envVariables", "environment", "layout", "persistAuth", "selectedSchemeOptions", "server", "workspace"]),
          createVNode(_sfc_main$1, {
            scheme: selectedScheme.value,
            state: unref(deleteSchemeModal),
            onClose: _cache[0] || (_cache[0] = ($event) => unref(deleteSchemeModal).hide()),
            onDelete: _cache[1] || (_cache[1] = ($event) => unselectAuth(selectedScheme.value?.id))
          }, null, 8, ["scheme", "state"])
        ]),
        _: 1
      }, 8, ["itemCount", "layout"])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
