// @scalar/api-reference/dist/blocks/scalar-auth-selector-block/components/DeleteRequestAuthModal.vue.js
import { defineComponent, useTemplateRef, createBlock, openBlock, unref, withCtx, createElementVNode, toDisplayString, createVNode, createTextVNode } from "vue";
import { ScalarModal, ScalarButton } from "@scalar/components";
import { emitCustomEvent } from "@scalar/workspace-store/events";
const _hoisted_1 = { class: "text-c-2 mb-4 text-sm leading-normal" };
const _hoisted_2 = { class: "flex justify-between gap-2" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "DeleteRequestAuthModal",
  props: {
    state: {},
    scheme: {}
  },
  emits: ["close", "delete"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const deleteScheme = () => {
      if (props.scheme?.id) {
        emitCustomEvent(wrapperRef.value?.$el, "scalar-delete-security-scheme", {
          uid: props.scheme.id
        });
      }
      emit("delete");
    };
    const wrapperRef = useTemplateRef("wrapperRef");
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(ScalarModal), {
        ref_key: "wrapperRef",
        ref: wrapperRef,
        size: "xxs",
        state: _ctx.state,
        title: "Delete Security Scheme"
      }, {
        default: withCtx(() => [
          createElementVNode("p", _hoisted_1, " This cannot be undone. You're about to delete the " + toDisplayString(_ctx.scheme?.label) + " security scheme from the collection. ", 1),
          createElementVNode("div", _hoisted_2, [
            createVNode(unref(ScalarButton), {
              class: "flex h-8 cursor-pointer items-center gap-1.5 px-3 shadow-none focus:outline-none",
              type: "button",
              variant: "outlined",
              onClick: _cache[0] || (_cache[0] = ($event) => emit("close"))
            }, {
              default: withCtx(() => [..._cache[1] || (_cache[1] = [
                createTextVNode(" Cancel ", -1)
              ])]),
              _: 1
            }),
            createVNode(unref(ScalarButton), {
              class: "flex h-8 cursor-pointer items-center gap-1.5 px-3 shadow-none focus:outline-none",
              type: "submit",
              onClick: deleteScheme
            }, {
              default: withCtx(() => [
                createTextVNode(" Delete " + toDisplayString(_ctx.scheme?.label), 1)
              ]),
              _: 1
            })
          ])
        ]),
        _: 1
      }, 8, ["state"]);
    };
  }
});
export {
  _sfc_main as default
};
