// @scalar/api-reference/dist/blocks/scalar-auth-selector-block/components/RequestAuthDataTableInput.vue.js
import { defineComponent, useId, createBlock, openBlock, unref, mergeProps, withCtx, renderSlot, createElementVNode } from "vue";
import { DataTableInput } from "@scalar/api-client/components/DataTable";
const _hoisted_1 = ["for"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "RequestAuthDataTableInput",
  props: {
    type: {},
    containerClass: {},
    required: { type: Boolean, default: false },
    modelValue: {},
    readOnly: { type: Boolean, default: false },
    environment: {},
    envVariables: {},
    workspace: {}
  },
  emits: ["update:modelValue", "inputFocus", "inputBlur", "selectVariable"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const id = useId();
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(DataTableInput), mergeProps({
        id: unref(id),
        canAddCustomEnumValue: !props.readOnly,
        containerClass: props.containerClass
      }, _ctx.$attrs, {
        envVariables: props.envVariables,
        environment: props.environment,
        lineWrapping: "",
        modelValue: props.modelValue,
        readOnly: props.readOnly,
        required: props.required,
        type: props.type,
        workspace: props.workspace,
        onInputBlur: _cache[0] || (_cache[0] = ($event) => emit("inputBlur")),
        onInputFocus: _cache[1] || (_cache[1] = ($event) => emit("inputFocus")),
        onSelectVariable: _cache[2] || (_cache[2] = ($event) => emit("selectVariable", $event)),
        "onUpdate:modelValue": _cache[3] || (_cache[3] = ($event) => emit("update:modelValue", $event))
      }), {
        default: withCtx(() => [
          createElementVNode("label", { for: unref(id) }, [
            renderSlot(_ctx.$slots, "default")
          ], 8, _hoisted_1)
        ]),
        icon: withCtx(() => [
          renderSlot(_ctx.$slots, "icon")
        ]),
        _: 3
      }, 16, ["id", "canAddCustomEnumValue", "containerClass", "envVariables", "environment", "modelValue", "readOnly", "required", "type", "workspace"]);
    };
  }
});
export {
  _sfc_main as default
};
