// @scalar/api-reference/dist/blocks/scalar-auth-selector-block/components/RequestAuthTab.vue.js
import { defineComponent, computed, ref, onMounted, useTemplateRef, createElementBlock, openBlock, Fragment, createElementVNode, renderList, createBlock, createCommentVNode, unref, withCtx, createVNode, toDisplayString, mergeProps, createTextVNode, normalizeClass, capitalize } from "vue";
import { DataTableRow, DataTableCell } from "@scalar/api-client/components/DataTable";
import { useWorkspace } from "@scalar/api-client/store";
import { ScalarMarkdownSummary } from "@scalar/components";
import { safeLocalStorage, CLIENT_LS_KEYS } from "@scalar/helpers/object/local-storage";
import { isDefined } from "@scalar/oas-utils/helpers";
import { emitCustomEvent } from "@scalar/workspace-store/events";
import { updateScheme } from "../helpers/update-scheme.js";
import _sfc_main$2 from "./OAuth2.vue.js";
import _sfc_main$1 from "./RequestAuthDataTableInput.vue.js";
const _hoisted_1 = { class: "bg-b-1 text-c-2 outline-b-3 top-0 z-1 h-full w-full overflow-hidden px-3 py-1.25 text-ellipsis group-hover/auth:absolute group-hover/auth:h-auto group-hover/auth:border-b *:first:line-clamp-1 *:first:text-ellipsis group-hover/auth:*:first:line-clamp-none" };
const _hoisted_2 = {
  key: 0,
  class: "flex min-h-8 border-t text-base"
};
const _hoisted_3 = { class: "flex h-8 max-w-full gap-2.5 overflow-x-auto px-3" };
const _hoisted_4 = ["onClick"];
const _hoisted_5 = { class: "relative z-10" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "RequestAuthTab",
  props: {
    collection: {},
    environment: {},
    envVariables: {},
    layout: {},
    persistAuth: { type: Boolean, default: false },
    securitySchemeUids: {},
    server: {},
    workspace: {}
  },
  setup(__props) {
    const storeContext = useWorkspace();
    const { securitySchemes } = storeContext;
    const security = computed(
      () => __props.securitySchemeUids.map((uid) => ({
        scheme: securitySchemes[uid]
      }))
    );
    const activeFlow = ref("");
    const generateLabel = (scheme) => {
      const description = scheme.description ? `: ${scheme.description}` : "";
      const baseLabel = `${capitalize(scheme.nameKey)}${description || `: ${scheme.type}`}`;
      if (scheme.type === "apiKey") {
        return `${capitalize(scheme.nameKey)}${description || `: ${scheme.in}`}`;
      }
      if (scheme.type === "oauth2") {
        const firstFlow = Object.values(scheme.flows ?? {})[0];
        return `${capitalize(scheme.nameKey)}: ${activeFlow.value ? activeFlow.value : firstFlow?.type ?? ""}${description}`;
      }
      if (scheme.type === "http") {
        return `${capitalize(scheme.nameKey)}: ${scheme.scheme}${description}`;
      }
      return `${baseLabel}${description}`;
    };
    const updateScheme$1 = (uid, path, value) => {
      updateScheme(uid, path, value, storeContext, wrapperRef.value, __props.persistAuth);
    };
    onMounted(() => {
      if (!__props.persistAuth) {
        return;
      }
      const auth = JSON.parse(
        safeLocalStorage().getItem(CLIENT_LS_KEYS.AUTH) ?? "{}"
      );
      const dict = Object.keys(securitySchemes).reduce(
        (acc, key) => {
          const scheme = securitySchemes[key];
          if (scheme) {
            acc[scheme.nameKey] = scheme.uid;
          }
          return acc;
        },
        {}
      );
      Object.entries(auth).forEach(([key, entry]) => {
        const uid = dict[key];
        if (uid) {
          const entries = Object.entries(entry);
          entries.forEach(([path, value]) => {
            emitCustomEvent(wrapperRef.value, "scalar-edit-security-scheme", {
              uid,
              path,
              value
            });
          });
        }
      });
      try {
        const selectedSchemeUids = JSON.parse(
          safeLocalStorage().getItem(CLIENT_LS_KEYS.SELECTED_SECURITY_SCHEMES) ?? "null"
        );
        const uids = selectedSchemeUids?.map((nameKeys) => {
          if (Array.isArray(nameKeys)) {
            return nameKeys.map((key) => dict[key]).filter(isDefined);
          }
          return dict[nameKeys];
        }).filter(isDefined);
        if (uids) {
          emitCustomEvent(wrapperRef.value, "scalar-select-security-schemes", {
            uids
          });
        }
      } catch {
      }
    });
    const dataTableInputProps = {
      environment: __props.environment,
      envVariables: __props.envVariables,
      workspace: __props.workspace
    };
    const wrapperRef = useTemplateRef("wrapperRef");
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(Fragment, null, [
        createElementVNode("div", {
          ref_key: "wrapperRef",
          ref: wrapperRef
        }, null, 512),
        (openBlock(true), createElementBlock(Fragment, null, renderList(security.value, ({ scheme }) => {
          return openBlock(), createElementBlock(Fragment, {
            key: scheme?.uid
          }, [
            security.value.length > 1 && scheme ? (openBlock(), createBlock(unref(DataTableRow), { key: 0 }, {
              default: withCtx(() => [
                createVNode(unref(DataTableCell), {
                  "aria-label": generateLabel(scheme),
                  class: "text-c-2 group/auth flex items-center leading-[22px] whitespace-nowrap outline-none hover:whitespace-normal"
                }, {
                  default: withCtx(() => [
                    createElementVNode("p", _hoisted_1, toDisplayString(generateLabel(scheme)), 1)
                  ]),
                  _: 2
                }, 1032, ["aria-label"])
              ]),
              _: 2
            }, 1024)) : createCommentVNode("", true),
            scheme?.description && security.value.length <= 1 ? (openBlock(), createBlock(unref(DataTableRow), { key: 1 }, {
              default: withCtx(() => [
                createVNode(unref(DataTableCell), {
                  "aria-label": scheme.description,
                  class: "max-h-[auto]"
                }, {
                  default: withCtx(() => [
                    createVNode(unref(ScalarMarkdownSummary), {
                      class: "auth-description bg-b-1 text-c-2 min-w-0 flex-1 px-3 py-1.25",
                      value: scheme.description
                    }, null, 8, ["value"])
                  ]),
                  _: 2
                }, 1032, ["aria-label"])
              ]),
              _: 2
            }, 1024)) : createCommentVNode("", true),
            scheme?.type === "http" ? (openBlock(), createElementBlock(Fragment, { key: 2 }, [
              scheme.scheme === "bearer" ? (openBlock(), createBlock(unref(DataTableRow), { key: 0 }, {
                default: withCtx(() => [
                  createVNode(_sfc_main$1, mergeProps({ ref_for: true }, dataTableInputProps, {
                    containerClass: _ctx.layout === "reference" && "border-t",
                    modelValue: scheme.token,
                    placeholder: "Token",
                    type: "password",
                    "onUpdate:modelValue": (v) => updateScheme$1(scheme.uid, "token", v)
                  }), {
                    default: withCtx(() => [..._cache[0] || (_cache[0] = [
                      createTextVNode(" Bearer Token ", -1)
                    ])]),
                    _: 1
                  }, 16, ["containerClass", "modelValue", "onUpdate:modelValue"])
                ]),
                _: 2
              }, 1024)) : scheme?.scheme === "basic" ? (openBlock(), createElementBlock(Fragment, { key: 1 }, [
                createVNode(unref(DataTableRow), null, {
                  default: withCtx(() => [
                    createVNode(_sfc_main$1, mergeProps({ ref_for: true }, dataTableInputProps, {
                      class: "text-c-2",
                      modelValue: scheme.username,
                      placeholder: "janedoe",
                      required: "",
                      "onUpdate:modelValue": (v) => updateScheme$1(scheme.uid, "username", v)
                    }), {
                      default: withCtx(() => [..._cache[1] || (_cache[1] = [
                        createTextVNode(" Username ", -1)
                      ])]),
                      _: 1
                    }, 16, ["modelValue", "onUpdate:modelValue"])
                  ]),
                  _: 2
                }, 1024),
                createVNode(unref(DataTableRow), null, {
                  default: withCtx(() => [
                    createVNode(_sfc_main$1, mergeProps({ ref_for: true }, dataTableInputProps, {
                      modelValue: scheme.password,
                      placeholder: "********",
                      type: "password",
                      "onUpdate:modelValue": (v) => updateScheme$1(scheme.uid, "password", v)
                    }), {
                      default: withCtx(() => [..._cache[2] || (_cache[2] = [
                        createTextVNode(" Password ", -1)
                      ])]),
                      _: 1
                    }, 16, ["modelValue", "onUpdate:modelValue"])
                  ]),
                  _: 2
                }, 1024)
              ], 64)) : createCommentVNode("", true)
            ], 64)) : scheme?.type === "apiKey" ? (openBlock(), createElementBlock(Fragment, { key: 3 }, [
              createVNode(unref(DataTableRow), null, {
                default: withCtx(() => [
                  createVNode(_sfc_main$1, mergeProps({ ref_for: true }, dataTableInputProps, {
                    containerClass: _ctx.layout === "reference" && "border-t",
                    modelValue: scheme.name,
                    placeholder: "api-key",
                    "onUpdate:modelValue": (v) => updateScheme$1(scheme.uid, "name", v)
                  }), {
                    default: withCtx(() => [..._cache[3] || (_cache[3] = [
                      createTextVNode(" Name ", -1)
                    ])]),
                    _: 1
                  }, 16, ["containerClass", "modelValue", "onUpdate:modelValue"])
                ]),
                _: 2
              }, 1024),
              createVNode(unref(DataTableRow), null, {
                default: withCtx(() => [
                  createVNode(_sfc_main$1, mergeProps({ ref_for: true }, dataTableInputProps, {
                    modelValue: scheme.value,
                    placeholder: "QUxMIFlPVVIgQkFTRSBBUkUgQkVMT05HIFRPIFVT",
                    type: "password",
                    "onUpdate:modelValue": (v) => updateScheme$1(scheme.uid, "value", v)
                  }), {
                    default: withCtx(() => [..._cache[4] || (_cache[4] = [
                      createTextVNode(" Value ", -1)
                    ])]),
                    _: 1
                  }, 16, ["modelValue", "onUpdate:modelValue"])
                ]),
                _: 2
              }, 1024)
            ], 64)) : scheme?.type === "oauth2" ? (openBlock(), createElementBlock(Fragment, { key: 4 }, [
              createVNode(unref(DataTableRow), null, {
                default: withCtx(() => [
                  Object.keys(scheme.flows).length > 1 ? (openBlock(), createElementBlock("div", _hoisted_2, [
                    createElementVNode("div", _hoisted_3, [
                      (openBlock(true), createElementBlock(Fragment, null, renderList(scheme?.flows, (_, key, ind) => {
                        return openBlock(), createElementBlock("button", {
                          key,
                          class: normalizeClass(["floating-bg text-c-3 relative cursor-pointer border-b-[1px] border-transparent py-1 text-base font-medium", {
                            "!text-c-1 !rounded-none border-b-[1px] !border-current": _ctx.layout !== "reference" && (activeFlow.value === key || ind === 0 && !activeFlow.value),
                            "!text-c-1 !rounded-none border-b-[1px] !border-current opacity-100": _ctx.layout === "reference" && (activeFlow.value === key || ind === 0 && !activeFlow.value)
                          }]),
                          type: "button",
                          onClick: ($event) => activeFlow.value = key
                        }, [
                          createElementVNode("span", _hoisted_5, toDisplayString(key), 1)
                        ], 10, _hoisted_4);
                      }), 128))
                    ])
                  ])) : createCommentVNode("", true)
                ]),
                _: 2
              }, 1024),
              (openBlock(true), createElementBlock(Fragment, null, renderList(scheme?.flows, (flow, key, ind) => {
                return openBlock(), createElementBlock(Fragment, { key }, [
                  activeFlow.value === key || ind === 0 && !activeFlow.value ? (openBlock(), createBlock(_sfc_main$2, mergeProps({
                    key: 0,
                    ref_for: true
                  }, dataTableInputProps, {
                    collection: _ctx.collection,
                    flow,
                    persistAuth: _ctx.persistAuth,
                    scheme,
                    server: _ctx.server,
                    workspace: _ctx.workspace
                  }), null, 16, ["collection", "flow", "persistAuth", "scheme", "server", "workspace"])) : createCommentVNode("", true)
                ], 64);
              }), 128))
            ], 64)) : scheme?.type === "openIdConnect" ? (openBlock(), createElementBlock("div", {
              key: 5,
              class: normalizeClass(["text-c-3 bg-b-1 flex min-h-[calc(4rem+1px)] items-center justify-center border-t border-b-0 px-4 text-base", { "rounded-b-lg": _ctx.layout === "reference" }])
            }, " Coming soon ", 2)) : createCommentVNode("", true)
          ], 64);
        }), 128))
      ], 64);
    };
  }
});
export {
  _sfc_main as default
};
