// @scalar/api-reference/dist/blocks/scalar-auth-selector-block/components/RequestAuthDataTable.vue2.js
import { defineComponent, ref, computed, watch, createElementBlock, openBlock, withModifiers, createCommentVNode, withDirectives, createVNode, normalizeClass, Fragment, renderList, createElementVNode, toDisplayString, unref, withCtx, vShow } from "vue";
import { DataTable } from "@scalar/api-client/components/DataTable";
import { useModal } from "@scalar/components";
import _sfc_main$2 from "./DeleteRequestAuthModal.vue.js";
import _sfc_main$1 from "./RequestAuthTab.vue.js";
const _hoisted_1 = ["onClick"];
const _hoisted_2 = { class: "relative z-10 font-medium whitespace-nowrap" };
const _hoisted_3 = {
  key: 0,
  class: "absolute inset-x-1 bottom-[var(--scalar-border-width)] left-1/2 z-1 h-px w-full -translate-x-1/2 bg-current"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "RequestAuthDataTable",
  props: {
    collection: {},
    environment: {},
    envVariables: {},
    layout: { default: "client" },
    persistAuth: { type: Boolean, default: false },
    selectedSchemeOptions: { default: () => [] },
    server: {},
    workspace: {}
  },
  setup(__props) {
    const deleteSchemeModal = useModal();
    const selectedScheme = ref(
      null
    );
    const activeAuthIndex = ref(0);
    const activeScheme = computed(() => {
      if (!__props.selectedSchemeOptions || __props.selectedSchemeOptions.length === 0) {
        return [];
      }
      const option = __props.selectedSchemeOptions[activeAuthIndex.value];
      if (!option) {
        return [];
      }
      const keys = option.id.split(",").filter(Boolean);
      return keys.length > 1 ? keys : [option.id];
    });
    const hasActiveSchemes = computed(() => {
      return activeScheme.value.length > 0;
    });
    watch(
      () => __props.selectedSchemeOptions,
      (newOptions) => {
        if (!newOptions || !newOptions[activeAuthIndex.value]) {
          activeAuthIndex.value = Math.max(0, activeAuthIndex.value - 1);
        }
      }
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("form", {
        onSubmit: _cache[1] || (_cache[1] = withModifiers(() => {
        }, ["prevent"]))
      }, [
        _ctx.selectedSchemeOptions.length > 1 ? (openBlock(), createElementBlock("div", {
          key: 0,
          class: normalizeClass(["box-content flex flex-wrap gap-x-2.5 overflow-hidden border border-b-0 px-3", _ctx.layout === "client" && "border-x-0"])
        }, [
          (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.selectedSchemeOptions, (option, index) => {
            return openBlock(), createElementBlock("div", {
              key: option.id,
              class: normalizeClass(["relative z-1 -mb-[var(--scalar-border-width)] flex h-8 cursor-pointer", [activeAuthIndex.value === index ? "text-c-1" : "text-c-3"]])
            }, [
              createElementVNode("button", {
                class: "floating-bg relative cursor-pointer border-b-[1px] border-transparent py-1 text-sm font-medium",
                type: "button",
                onClick: ($event) => activeAuthIndex.value = index
              }, [
                createElementVNode("span", _hoisted_2, toDisplayString(option.label), 1)
              ], 8, _hoisted_1),
              activeAuthIndex.value === index ? (openBlock(), createElementBlock("div", _hoisted_3)) : createCommentVNode("", true)
            ], 2);
          }), 128))
        ], 2)) : createCommentVNode("", true),
        withDirectives(createVNode(unref(DataTable), {
          class: normalizeClass(["flex-1", _ctx.layout === "reference" && "bg-b-1 rounded-b-lg border border-t-0"]),
          columns: [""],
          presentational: ""
        }, {
          default: withCtx(() => [
            createVNode(_sfc_main$1, {
              collection: _ctx.collection,
              envVariables: _ctx.envVariables,
              environment: _ctx.environment,
              layout: _ctx.layout,
              persistAuth: _ctx.persistAuth,
              securitySchemeUids: activeScheme.value,
              server: _ctx.server,
              workspace: _ctx.workspace
            }, null, 8, ["collection", "envVariables", "environment", "layout", "persistAuth", "securitySchemeUids", "server", "workspace"])
          ]),
          _: 1
        }, 8, ["class"]), [
          [vShow, hasActiveSchemes.value]
        ]),
        !hasActiveSchemes.value ? (openBlock(), createElementBlock("div", {
          key: 1,
          class: normalizeClass([
            "text-c-3 bg-b-1 flex min-h-16 items-center justify-center border-t px-4 text-sm",
            _ctx.layout === "reference" && "min-h-[calc(4rem+0.5px)] rounded-b-lg border"
          ])
        }, " No authentication selected ", 2)) : createCommentVNode("", true),
        createVNode(_sfc_main$2, {
          scheme: selectedScheme.value,
          state: unref(deleteSchemeModal),
          onClose: _cache[0] || (_cache[0] = ($event) => unref(deleteSchemeModal).hide())
        }, null, 8, ["scheme", "state"])
      ], 32);
    };
  }
});
export {
  _sfc_main as default
};
