// @scalar/api-reference/dist/components/Content/Tags/Tag.vue.js
import { defineComponent, createBlock, openBlock, withCtx, renderSlot } from "vue";
import ClassicLayout from "./components/ClassicLayout.vue.js";
import ModernLayout from "./components/ModernLayout.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Tag",
  props: {
    tag: {},
    layout: {},
    moreThanOneTag: { type: Boolean },
    isLoading: { type: Boolean },
    isCollapsed: { type: Boolean },
    eventBus: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return _ctx.layout === "classic" ? (openBlock(), createBlock(ClassicLayout, {
        key: 0,
        eventBus: _ctx.eventBus,
        isCollapsed: _ctx.isCollapsed,
        layout: _ctx.layout,
        tag: _ctx.tag
      }, {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "default")
        ]),
        _: 3
      }, 8, ["eventBus", "isCollapsed", "layout", "tag"])) : (openBlock(), createBlock(ModernLayout, {
        key: 1,
        eventBus: _ctx.eventBus,
        isCollapsed: _ctx.isCollapsed,
        isLoading: _ctx.isLoading,
        layout: _ctx.layout,
        moreThanOneTag: _ctx.moreThanOneTag,
        tag: _ctx.tag
      }, {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "default")
        ]),
        _: 3
      }, 8, ["eventBus", "isCollapsed", "isLoading", "layout", "moreThanOneTag", "tag"]));
    };
  }
});
export {
  _sfc_main as default
};
