// @scalar/api-client/dist/v2/features/app/App.vue.js
import { defineComponent as O, ref as h, computed as a, createElementBlock as A, openBlock as d, Fragment as H, createElementVNode as p, createVNode as v, unref as c, withCtx as M, createBlock as b, withDirectives as N, vShow as P, normalizeProps as L, guardReactiveProps as F } from "vue";
import { ScalarTeleportRoot as $ } from "@scalar/components";
import { isHttpMethod as z } from "@scalar/helpers/http/is-http-method";
import { getThemeStyles as j } from "@scalar/themes";
import { useColorMode as q } from "@scalar/use-hooks/useColorMode";
import { createWorkspaceEventBus as G } from "@scalar/workspace-store/events";
import { xScalarEnvironmentSchema as y } from "@scalar/workspace-store/schemas/extensions/document/x-scalar-environments";
import { coerceValue as x } from "@scalar/workspace-store/schemas/typebox-coerce";
import { useRoute as J, useRouter as K, RouterView as Q } from "vue-router";
import { useSidebarState as X } from "../../hooks/use-sidebar-state.js";
import { useWorkspaceClientEvents as Y } from "../../hooks/use-workspace-client-events.js";
import Z from "./components/AppSidebar.vue.js";
import _ from "./components/DesktopTabs.vue.js";
import ee from "./components/WebTopNav.vue.js";
const te = ["innerHTML"], oe = { class: "flex min-h-0 flex-1" }, ae = { class: "bg-b-1 flex-1" }, re = {}, be = /* @__PURE__ */ O({
  ...re,
  __name: "App",
  props: {
    layout: {},
    workspaceStore: {}
  },
  setup(o) {
    q(), typeof window < "u" && (window.dataDumpWorkspace = () => o.workspaceStore);
    const n = (e) => {
      const t = E.params[e];
      return typeof t == "string" ? t : void 0;
    }, E = J(), W = K(), k = G(), s = h("default"), i = h(!0), u = a(
      () => n("workspaceSlug") ?? "default"
    ), l = a(() => n("documentSlug")), m = a(() => l.value ? o.workspaceStore.workspace.documents[l.value] ?? null : null), S = a(() => {
      const e = n("pathEncoded");
      return e ? decodeURIComponent(e) : void 0;
    }), w = a(() => {
      const e = n("method");
      return e && z(e) ? e : void 0;
    }), f = a(() => n("exampleName")), { handleSelectItem: g, sidebarState: T } = X({
      workspaceStore: o.workspaceStore,
      workspaceSlug: u,
      documentSlug: l,
      path: S,
      method: w,
      exampleName: f
    });
    Y(k, m, o.workspaceStore);
    const D = a(() => {
      const e = o.workspaceStore.workspace["x-scalar-active-environment"];
      if (!e)
        return x(y, {});
      const t = o.workspaceStore.workspace["x-scalar-environments"]?.[e] ?? {
        variables: []
      }, r = m.value?.["x-scalar-environments"]?.[e] ?? {
        variables: []
      };
      return x(y, {
        ...t,
        ...r,
        variables: [...t.variables, ...r.variables]
      });
    }), I = a(() => {
      const e = o.workspaceStore.workspace["x-scalar-theme"];
      return e ? `<style>${j(e)}</style>` : "";
    }), R = a(
      () => o.workspaceStore.workspace["x-scalar-sidebar-width"] ?? 288
    ), B = a(
      () => !!(u.value && !l.value)
    ), U = (e) => o.workspaceStore.update("x-scalar-sidebar-width", e), V = () => W.push({
      name: "workspace",
      params: { workspaceSlug: u.value }
    }), C = a(() => ({
      document: m.value,
      environment: D.value,
      eventBus: k,
      exampleName: f.value,
      layout: o.layout,
      method: w.value,
      path: S.value,
      workspaceStore: o.workspaceStore
    }));
    return (e, t) => (d(), A(H, null, [
      p("div", { innerHTML: I.value }, null, 8, te),
      v(c($), null, {
        default: M(() => [
          e.layout === "desktop" ? (d(), b(_, { key: 0 })) : (d(), b(ee, {
            key: 1,
            modelValue: s.value,
            "onUpdate:modelValue": t[0] || (t[0] = (r) => s.value = r)
          }, null, 8, ["modelValue"])),
          p("main", oe, [
            N(v(Z, {
              isSidebarOpen: i.value,
              "onUpdate:isSidebarOpen": t[1] || (t[1] = (r) => i.value = r),
              workspace: s.value,
              "onUpdate:workspace": t[2] || (t[2] = (r) => s.value = r),
              isWorkspaceOpen: B.value,
              layout: e.layout,
              sidebarState: c(T),
              sidebarWidth: R.value,
              "onClick:workspace": V,
              onSelectItem: c(g),
              "onUpdate:sidebarWidth": U
            }, null, 8, ["isSidebarOpen", "workspace", "isWorkspaceOpen", "layout", "sidebarState", "sidebarWidth", "onSelectItem"]), [
              [P, i.value]
            ]),
            p("div", ae, [
              v(c(Q), L(F(C.value)), null, 16)
            ])
          ])
        ]),
        _: 1
      })
    ], 64));
  }
});
export {
  be as default
};
