// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/components/RequestAuthDataTableInput.vue.js
import { defineComponent as p, mergeModels as u, useModel as f, useId as V, createBlock as y, openBlock as v, unref as o, mergeProps as C, withCtx as d, renderSlot as s, createElementVNode as B } from "vue";
import b from "../../../components/data-table/DataTableInput.vue.js";
const q = ["for"], $ = /* @__PURE__ */ p({
  __name: "RequestAuthDataTableInput",
  props: /* @__PURE__ */ u({
    containerClass: {},
    environment: {},
    readOnly: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    type: {}
  }, {
    modelValue: { default: "", required: !0 },
    modelModifiers: {}
  }),
  emits: /* @__PURE__ */ u(["inputFocus", "inputBlur", "selectVariable"], ["update:modelValue"]),
  setup(i, { emit: m }) {
    const t = m, r = f(i, "modelValue"), a = V();
    return (e, n) => (v(), y(o(b), C({ id: o(a) }, e.$attrs, {
      modelValue: r.value,
      "onUpdate:modelValue": n[0] || (n[0] = (l) => r.value = l),
      canAddCustomEnumValue: !e.readOnly,
      containerClass: e.containerClass,
      environment: e.environment,
      lineWrapping: "",
      readOnly: e.readOnly,
      required: e.required,
      type: e.type,
      onInputBlur: n[1] || (n[1] = (l) => t("inputBlur")),
      onInputFocus: n[2] || (n[2] = (l) => t("inputFocus")),
      onSelectVariable: n[3] || (n[3] = (l) => t("selectVariable", l))
    }), {
      default: d(() => [
        B("label", { for: o(a) }, [
          s(e.$slots, "default")
        ], 8, q)
      ]),
      icon: d(() => [
        s(e.$slots, "icon")
      ]),
      _: 3
    }, 16, ["id", "modelValue", "canAddCustomEnumValue", "containerClass", "environment", "readOnly", "required", "type"]));
  }
});
export {
  $ as default
};
