// @scalar/api-client/dist/v2/features/modal/helpers/create-api-client-modal.js
import { useModal as d } from "@scalar/components";
import { reactive as p, createApp as u } from "vue";
import m from "../Modal.vue.js";
const M = ({ el: l, workspaceStore: o, mountOnInitialize: s = !0 }) => {
  const c = {
    path: "default",
    method: "default",
    example: "default",
    documentSlug: o.workspace["x-scalar-active-document"] || "default"
  }, r = p(c), t = d(), a = u(m, {
    workspaceStore: o,
    activeEntities: r,
    modalState: t
  });
  a.config.idPrefix = "scalar-client";
  const i = (e = l) => {
    if (!e) {
      console.error(
        "[@scalar/api-client] Could not create the API client Modal.",
        "Invalid HTML element provided.",
        "Read more: https://github.com/scalar/scalar/tree/main/packages/api-client"
      );
      return;
    }
    a.mount(e);
  };
  s && i();
  const n = (e) => Object.assign(r, c, e);
  return {
    /** The vue app instance for the modal, be careful with this */
    app: a,
    /** Open the API client modal and optionally route to an example */
    open: (e) => {
      t.open = !0, e && n(e);
    },
    /** Mount the client to a given element */
    mount: i,
    /** "Route" to the specified path, method and example */
    route: n,
    /** Controls the visibility of the modal */
    modalState: t
  };
};
export {
  M as createApiClientModal
};
