// @scalar/api-reference/dist/blocks/scalar-info-block/components/InfoBlock.vue.js
import { defineComponent, computed, createBlock, openBlock, withCtx, createVNode, renderSlot } from "vue";
import DownloadLink from "./DownloadLink.vue.js";
import IntroductionCard from "./IntroductionCard.vue.js";
import IntroductionLayout from "./IntroductionLayout.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "InfoBlock",
  props: {
    id: {},
    oasVersion: {},
    info: {},
    externalDocs: {},
    documentExtensions: {},
    infoExtensions: {},
    eventBus: {},
    options: {}
  },
  setup(__props) {
    const introCardsSlot = computed(
      () => __props.options.layout === "classic" ? "after" : "aside"
    );
    return (_ctx, _cache) => {
      return openBlock(), createBlock(IntroductionLayout, {
        id: _ctx.id,
        documentExtensions: _ctx.documentExtensions,
        eventBus: _ctx.eventBus,
        externalDocs: _ctx.externalDocs,
        headingSlugGenerator: _ctx.options.headingSlugGenerator,
        info: _ctx.info,
        infoExtensions: _ctx.infoExtensions,
        oasVersion: _ctx.oasVersion
      }, {
        [introCardsSlot.value]: withCtx(() => [
          createVNode(IntroductionCard, {
            row: _ctx.options.layout === "classic"
          }, {
            default: withCtx(() => [
              renderSlot(_ctx.$slots, "selectors")
            ]),
            _: 3
          }, 8, ["row"])
        ]),
        "download-link": withCtx(() => [
          createVNode(DownloadLink, {
            documentDownloadType: _ctx.options.documentDownloadType ?? "both"
          }, null, 8, ["documentDownloadType"])
        ]),
        _: 2
      }, 1032, ["id", "documentExtensions", "eventBus", "externalDocs", "headingSlugGenerator", "info", "infoExtensions", "oasVersion"]);
    };
  }
});
export {
  _sfc_main as default
};
