// @scalar/api-reference/dist/features/Operation/Operation.vue.js
import { defineComponent, computed, createElementBlock, createCommentVNode, openBlock, Fragment, createBlock } from "vue";
import { getResolvedRef } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { combineParams } from "./helpers/combine-params.js";
import { getFirstServer } from "./helpers/get-first-server.js";
import ClassicLayout from "./layouts/ClassicLayout.vue.js";
import ModernLayout from "./layouts/ModernLayout.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Operation",
  props: {
    id: {},
    method: {},
    path: {},
    pathValue: {},
    server: {},
    security: {},
    getSecurityScheme: { type: Function },
    isCollapsed: { type: Boolean },
    xScalarDefaultClient: {},
    eventBus: {},
    options: {}
  },
  setup(__props) {
    const operation = computed(() => {
      const entity = getResolvedRef(__props.pathValue?.[__props.method]);
      if (!entity) {
        return null;
      }
      const parameters = combineParams(__props.pathValue?.parameters, entity.parameters);
      return { ...entity, parameters };
    });
    const selectedSecuritySchemes = computed(
      () => __props.getSecurityScheme(operation.value?.security, __props.security || [])
    );
    const selectedServer = computed(
      () => getFirstServer(
        // 1) Operation
        operation.value?.servers,
        // 2) Path Item
        __props.pathValue?.servers,
        // 3) Document
        __props.server
      )
    );
    return (_ctx, _cache) => {
      return operation.value ? (openBlock(), createElementBlock(Fragment, { key: 0 }, [
        _ctx.options?.layout === "classic" ? (openBlock(), createBlock(ClassicLayout, {
          key: 0,
          id: _ctx.id,
          eventBus: _ctx.eventBus,
          isCollapsed: _ctx.isCollapsed,
          method: _ctx.method,
          operation: operation.value,
          options: _ctx.options,
          path: _ctx.path,
          securitySchemes: selectedSecuritySchemes.value,
          server: selectedServer.value,
          xScalarDefaultClient: _ctx.xScalarDefaultClient
        }, null, 8, ["id", "eventBus", "isCollapsed", "method", "operation", "options", "path", "securitySchemes", "server", "xScalarDefaultClient"])) : (openBlock(), createBlock(ModernLayout, {
          key: 1,
          id: _ctx.id,
          eventBus: _ctx.eventBus,
          method: _ctx.method,
          operation: operation.value,
          options: _ctx.options,
          path: _ctx.path,
          securitySchemes: selectedSecuritySchemes.value,
          server: selectedServer.value,
          xScalarDefaultClient: _ctx.xScalarDefaultClient
        }, null, 8, ["id", "eventBus", "method", "operation", "options", "path", "securitySchemes", "server", "xScalarDefaultClient"]))
      ], 64)) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
