// @scalar/api-client/dist/v2/blocks/operation-block/components/Header.vue2.js
import { defineComponent as m, createElementBlock as r, openBlock as n, createElementVNode as s, createVNode as d, createCommentVNode as l, normalizeClass as v, unref as i, createBlock as g } from "vue";
import { ScalarIcon as u } from "@scalar/components";
import f from "../../scalar-address-bar-block/components/AddressBar.vue.js";
import y from "../../../../components/OpenApiClientButton.vue.js";
const h = { class: "lg:min-h-header t-app__top-container flex w-full flex-wrap items-center justify-center border-b p-2 pt-2 lg:p-1 lg:pt-1" }, b = { class: "mb-2 flex w-1/2 flex-row items-center gap-1 lg:mb-0 lg:flex-1 lg:px-1" }, B = { class: "mb-2 flex w-1/2 flex-row items-center justify-end gap-1 lg:mb-0 lg:flex-1 lg:px-2.5" }, S = /* @__PURE__ */ m({
  __name: "Header",
  props: {
    path: {},
    method: {},
    layout: {},
    isSidebarOpen: { type: Boolean },
    showSidebar: { type: Boolean, default: !0 },
    hideClientButton: { type: Boolean, default: !1 },
    integration: {},
    documentUrl: {},
    source: {},
    server: {},
    servers: {},
    history: {},
    requestLoadingPercentage: {},
    eventBus: {},
    environment: {},
    events: {}
  },
  emits: ["execute", "update:path", "update:method"],
  setup(C, { emit: p }) {
    const a = p;
    return (e, t) => (n(), r("div", h, [
      s("div", b, [
        e.showSidebar ? (n(), r("div", {
          key: 0,
          class: v(["size-8", { hidden: e.layout === "modal" && !e.isSidebarOpen }])
        }, null, 2)) : l("", !0)
      ]),
      d(i(f), {
        environment: e.environment,
        eventBus: e.eventBus,
        events: e.events,
        history: e.history,
        layout: e.layout,
        method: e.method,
        path: e.path,
        percentage: e.requestLoadingPercentage,
        server: e.server,
        servers: e.servers,
        onExecute: t[0] || (t[0] = (o) => a("execute")),
        "onUpdate:method": t[1] || (t[1] = (o) => a("update:method", o)),
        "onUpdate:path": t[2] || (t[2] = (o) => a("update:path", o))
      }, null, 8, ["environment", "eventBus", "events", "history", "layout", "method", "path", "percentage", "server", "servers"]),
      s("div", B, [
        e.layout === "modal" && e.documentUrl && !e.hideClientButton ? (n(), g(i(y), {
          key: 0,
          buttonSource: "modal",
          class: "!w-fit lg:-mr-1",
          integration: e.integration ?? null,
          source: e.source ?? "api-reference",
          url: e.documentUrl
        }, null, 8, ["integration", "source", "url"])) : l("", !0),
        e.layout === "modal" ? (n(), r("button", {
          key: 1,
          class: "app-exit-button gitbook-hidden zoomed:static zoomed:p-1 fixed top-2 right-2 rounded-full p-2",
          type: "button",
          onClick: t[3] || (t[3] = (o) => e.eventBus.emit("hide:modal"))
        }, [
          d(i(u), {
            icon: "Close",
            size: "lg",
            thickness: "2"
          }),
          t[5] || (t[5] = s("span", { class: "sr-only" }, "Close Client", -1))
        ])) : l("", !0),
        e.layout === "modal" ? (n(), r("button", {
          key: 2,
          class: "text-c-1 hover:bg-b-2 active:text-c-1 gitbook-show -mr-1.5 rounded p-2",
          type: "button",
          onClick: t[4] || (t[4] = (o) => e.eventBus.emit("hide:modal"))
        }, [
          d(i(u), {
            icon: "Close",
            size: "md",
            thickness: "1.75"
          }),
          t[6] || (t[6] = s("span", { class: "sr-only" }, "Close Client", -1))
        ])) : l("", !0)
      ])
    ]));
  }
});
export {
  S as default
};
