// @scalar/api-reference/dist/features/Operation/components/callbacks/Callbacks.vue.js
import { defineComponent, createElementBlock, openBlock, createElementVNode, Fragment, renderList, unref, createBlock, createCommentVNode } from "vue";
import { isHttpMethod } from "@scalar/helpers/http/is-http-method";
import { getResolvedRef } from "@scalar/workspace-store/helpers/get-resolved-ref";
import Callback from "./Callback.vue.js";
const _hoisted_1 = {
  "aria-label": "Callbacks",
  class: "callbacks-list gap-3",
  role: "group"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Callbacks",
  props: {
    path: {},
    method: {},
    callbacks: {},
    eventBus: {},
    options: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        _cache[0] || (_cache[0] = createElementVNode("div", { class: "callbacks-title text-c-1 my-3 text-lg font-medium" }, " Callbacks ", -1)),
        (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.callbacks, (pathItem, name) => {
          return openBlock(), createElementBlock(Fragment, { key: name }, [
            (openBlock(true), createElementBlock(Fragment, null, renderList(unref(getResolvedRef)(pathItem), (methods, url) => {
              return openBlock(), createElementBlock(Fragment, null, [
                (openBlock(true), createElementBlock(Fragment, null, renderList(methods, (callback, callbackMethod) => {
                  return openBlock(), createElementBlock(Fragment, { key: callbackMethod }, [
                    unref(isHttpMethod)(callbackMethod) ? (openBlock(), createBlock(Callback, {
                      key: 0,
                      callback,
                      eventBus: _ctx.eventBus,
                      method: callbackMethod,
                      name,
                      operationMethod: __props.method,
                      options: _ctx.options,
                      path: _ctx.path,
                      url
                    }, null, 8, ["callback", "eventBus", "method", "name", "operationMethod", "options", "path", "url"])) : createCommentVNode("", true)
                  ], 64);
                }), 128))
              ], 64);
            }), 256))
          ], 64);
        }), 128))
      ]);
    };
  }
});
export {
  _sfc_main as default
};
