// @scalar/api-reference/dist/components/Content/Tags/components/TagSection.vue.js
import { defineComponent, createBlock, createCommentVNode, openBlock, unref, withCtx, withDirectives, createVNode, createTextVNode, toDisplayString, vShow } from "vue";
import { ScalarMarkdown } from "@scalar/components";
import ScreenReader from "../../../ScreenReader.vue.js";
import Section from "../../../Section/Section.vue.js";
import SectionHeader from "../../../Section/SectionHeader.vue.js";
import _sfc_main$1 from "../../../Anchor/Anchor.vue.js";
import SectionHeaderTag from "../../../Section/SectionHeaderTag.vue.js";
import SectionContent from "../../../Section/SectionContent.vue.js";
import SectionColumns from "../../../Section/SectionColumns.vue.js";
import SectionColumn from "../../../Section/SectionColumn.vue.js";
import OperationsList from "../../../OperationsList/OperationsList.vue.js";
import _sfc_main$2 from "../../../../features/specification-extension/SpecificationExtension.vue2.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "TagSection",
  props: {
    tag: {},
    headerId: {},
    isCollapsed: { type: Boolean },
    isLoading: { type: Boolean },
    eventBus: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return _ctx.tag ? (openBlock(), createBlock(unref(Section), {
        key: 0,
        id: _ctx.tag.id,
        role: "none",
        onIntersecting: _cache[1] || (_cache[1] = () => _ctx.eventBus?.emit("intersecting:nav-item", { id: _ctx.tag.id }))
      }, {
        default: withCtx(() => [
          withDirectives(createVNode(unref(SectionHeader), null, {
            default: withCtx(() => [
              createVNode(unref(_sfc_main$1), {
                onCopyAnchorUrl: _cache[0] || (_cache[0] = () => _ctx.eventBus?.emit("copy-url:nav-item", { id: _ctx.tag.id }))
              }, {
                default: withCtx(() => [
                  createVNode(unref(SectionHeaderTag), {
                    id: _ctx.headerId,
                    level: 2
                  }, {
                    default: withCtx(() => [
                      createTextVNode(toDisplayString(_ctx.tag.title) + " ", 1),
                      _ctx.isCollapsed ? (openBlock(), createBlock(ScreenReader, { key: 0 }, {
                        default: withCtx(() => [..._cache[2] || (_cache[2] = [
                          createTextVNode(" (Collapsed)", -1)
                        ])]),
                        _: 1
                      })) : createCommentVNode("", true)
                    ]),
                    _: 1
                  }, 8, ["id"])
                ]),
                _: 1
              })
            ]),
            _: 1
          }, 512), [
            [vShow, !_ctx.isLoading]
          ]),
          createVNode(unref(SectionContent), { loading: _ctx.isLoading }, {
            default: withCtx(() => [
              createVNode(unref(SectionColumns), null, {
                default: withCtx(() => [
                  createVNode(unref(SectionColumn), null, {
                    default: withCtx(() => [
                      createVNode(unref(ScalarMarkdown), {
                        clamp: _ctx.isCollapsed ? 7 : void 0,
                        value: _ctx.tag?.description ?? "",
                        withImages: ""
                      }, null, 8, ["clamp", "value"])
                    ]),
                    _: 1
                  }),
                  createVNode(unref(SectionColumn), null, {
                    default: withCtx(() => [
                      createVNode(unref(OperationsList), {
                        eventBus: _ctx.eventBus,
                        tag: _ctx.tag
                      }, null, 8, ["eventBus", "tag"])
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          }, 8, ["loading"]),
          createVNode(unref(_sfc_main$2), {
            value: _ctx.tag.xKeys
          }, null, 8, ["value"])
        ]),
        _: 1
      }, 8, ["id"])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
