// @scalar/workspace-store/dist/mutators/auth.js
import { generateUniqueValue } from "../helpers/generate-unique-value.js";
import { getResolvedRef } from "../helpers/get-resolved-ref.js";
import { mergeObjects } from "../helpers/merge-object.js";
const updateSelectedSecuritySchemes = (document, { selectedRequirements, newSchemes, meta }) => {
  if (!document) {
    return;
  }
  const getTarget = () => {
    if (meta.type === "document") {
      return document;
    }
    return getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  };
  const createdSchemes = newSchemes.map((scheme) => {
    const name = generateUniqueValue({
      defaultValue: scheme.name,
      validation: (value) => !document.components?.securitySchemes?.[value],
      maxRetries: 100
    });
    if (!name) {
      return;
    }
    if (!document.components) {
      document.components = {};
    }
    if (!document.components.securitySchemes) {
      document.components.securitySchemes = {};
    }
    document.components.securitySchemes[name] = scheme.scheme;
    return {
      [name]: []
    };
  }).filter(Boolean);
  const target = getTarget();
  const newSelectedSecuritySchemes = [...selectedRequirements, ...createdSchemes];
  if (!target) {
    return;
  }
  if (!target["x-scalar-selected-security"]) {
    target["x-scalar-selected-security"] = {
      selectedIndex: -1,
      selectedSchemes: []
    };
  }
  const selectedIndex = target["x-scalar-selected-security"].selectedIndex;
  target["x-scalar-selected-security"].selectedSchemes = newSelectedSecuritySchemes;
  if (newSelectedSecuritySchemes.length > 0 && selectedIndex < 0) {
    target["x-scalar-selected-security"].selectedIndex = 0;
  }
  if (selectedIndex >= newSelectedSecuritySchemes.length) {
    target["x-scalar-selected-security"].selectedIndex = newSelectedSecuritySchemes.length - 1;
  }
};
const updateSecurityScheme = (document, { payload, name }) => {
  const target = getResolvedRef(document?.components?.securitySchemes?.[name]);
  if (!target) {
    console.error(`Security scheme ${name} not found`);
    return;
  }
  if (target.type === payload.type) {
    mergeObjects(target, payload);
  }
  return target;
};
const updateSelectedAuthTab = (document, { index, meta }) => {
  if (!document) {
    return;
  }
  const getTarget = () => {
    if (meta.type === "document") {
      return document;
    }
    return getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  };
  const target = getTarget();
  if (!target) {
    return;
  }
  if (!target["x-scalar-selected-security"]) {
    target["x-scalar-selected-security"] = {
      selectedIndex: 0,
      selectedSchemes: []
    };
  }
  target["x-scalar-selected-security"].selectedIndex = index;
};
const updateSelectedScopes = (document, { id, name, scopes, meta }) => {
  if (!document) {
    return;
  }
  const getTarget = () => {
    if (meta.type === "document") {
      return document;
    }
    return getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  };
  const target = getTarget();
  if (!target) {
    return;
  }
  const selectedSchemes = target["x-scalar-selected-security"]?.selectedSchemes;
  if (!selectedSchemes) {
    return;
  }
  const scheme = selectedSchemes.find((scheme2) => JSON.stringify(Object.keys(scheme2)) === JSON.stringify(id));
  if (!scheme) {
    return;
  }
  scheme[name] = scopes;
};
const deleteSecurityScheme = (document, { names }) => {
  if (!document) {
    return;
  }
  const target = getResolvedRef(document.components?.securitySchemes);
  if (!target) {
    return;
  }
  names.forEach((name) => {
    delete target[name];
  });
  const filterSecuritySchemes = (schemes) => {
    return schemes.filter((scheme) => !names.some((name) => Object.keys(scheme).includes(name)));
  };
  if (document["x-scalar-selected-security"]) {
    const selectedSecurity = document["x-scalar-selected-security"];
    selectedSecurity.selectedSchemes = filterSecuritySchemes(selectedSecurity.selectedSchemes);
  }
  if (document["security"]) {
    document["security"] = filterSecuritySchemes(document["security"]);
  }
  Object.values(document.paths ?? {}).forEach((path) => {
    Object.values(path).forEach((operation) => {
      if (typeof operation !== "object") {
        return;
      }
      const resolvedOperation = getResolvedRef(operation);
      if ("security" in resolvedOperation && resolvedOperation["security"]) {
        resolvedOperation["security"] = filterSecuritySchemes(resolvedOperation["security"]);
      }
      if ("x-scalar-selected-security" in resolvedOperation && resolvedOperation["x-scalar-selected-security"]) {
        resolvedOperation["x-scalar-selected-security"].selectedSchemes = filterSecuritySchemes(
          resolvedOperation["x-scalar-selected-security"].selectedSchemes
        );
      }
    });
  });
};
export {
  deleteSecurityScheme,
  updateSecurityScheme,
  updateSelectedAuthTab,
  updateSelectedScopes,
  updateSelectedSecuritySchemes
};
