// @scalar/workspace-store/dist/mutators/index.js
import { cookieMutators } from "../mutators/cookie.js";
import { getDocument } from "../mutators/helpers.js";
import { requestMutators } from "../mutators/request.js";
import { securitySchemeMutators } from "../mutators/security-schemes.js";
function generateClientMutators(store) {
  const serverMutators = (target) => {
    const addServer2 = (server) => {
      if (!target) {
        return false;
      }
      target.push(server);
      return true;
    };
    const deleteServer2 = (url) => {
      if (!target) {
        return false;
      }
      const newTarget = [...target.filter((it) => it.url !== url)];
      target.splice(0, target.length);
      target.push(...newTarget);
      return true;
    };
    return {
      addServer: addServer2,
      deleteServer: deleteServer2
    };
  };
  const documentMutators = (documentName) => {
    const document = getDocument(store, documentName);
    if (document) {
      if (!document.servers) {
        document.servers = [];
      }
      if (!document.components) {
        document.components = {};
      }
      if (!document.components.securitySchemes) {
        document.components.securitySchemes = {};
      }
    }
    return {
      requestMutators: requestMutators(document),
      securitySchemeMutators: securitySchemeMutators(document?.components?.securitySchemes),
      cookieMutators: cookieMutators(document),
      serverMutators: serverMutators(document?.servers)
    };
  };
  const workspaceMutators = () => {
    const workspace = store.workspace;
    if (!workspace["x-scalar-client-config-servers"]) {
      workspace["x-scalar-client-config-servers"] = [];
    }
    if (!store.workspace["x-scalar-client-config-security-schemes"]) {
      store.workspace["x-scalar-client-config-security-schemes"] = {};
    }
    return {
      cookieMutators: cookieMutators(store.workspace),
      serverMutators: serverMutators(store.workspace["x-scalar-client-config-servers"]),
      securitySchemeMutators: securitySchemeMutators(store.workspace["x-scalar-client-config-security-schemes"])
    };
  };
  return {
    /**
     * Returns mutators for the workspace-level configuration.
     */
    workspace: () => workspaceMutators(),
    /**
     * Returns mutators for the currently active document.
     * Falls back to the first document if no active document is set.
     */
    active: () => documentMutators(store.workspace["x-scalar-active-document"] ?? Object.keys(store.workspace.documents)[0] ?? ""),
    /**
     * Returns mutators for a specific document by name.
     *
     * @param name - The name of the document
     */
    doc: (name) => documentMutators(name)
  };
}
import {
  deleteSecurityScheme,
  updateSecurityScheme,
  updateSelectedAuthTab,
  updateSelectedScopes,
  updateSelectedSecuritySchemes
} from "./auth.js";
import { toggleDocumentSecurity } from "./document.js";
import {
  upsertEnvironment,
  upsertEnvironmentVariable
} from "./environment.js";
import {
  addOperationParameter,
  addOperationRequestBodyFormRow,
  deleteAllOperationParameters,
  deleteOperationParameter,
  deleteOperationRequestBodyFormRow,
  updateOperationMethodDraft,
  updateOperationParameter,
  updateOperationPathDraft,
  updateOperationRequestBodyContentType,
  updateOperationRequestBodyExample,
  updateOperationRequestBodyFormRow,
  updateOperationSummary
} from "./operation.js";
import {
  addServer,
  deleteServer,
  updateSelectedServer,
  updateServer,
  updateServerVariables
} from "./server.js";
export {
  addOperationParameter,
  addOperationRequestBodyFormRow,
  addServer,
  deleteAllOperationParameters,
  deleteOperationParameter,
  deleteOperationRequestBodyFormRow,
  deleteSecurityScheme,
  deleteServer,
  generateClientMutators,
  toggleDocumentSecurity,
  updateOperationMethodDraft as updateOperationMethod,
  updateOperationParameter,
  updateOperationPathDraft as updateOperationPath,
  updateOperationRequestBodyContentType,
  updateOperationRequestBodyExample,
  updateOperationRequestBodyFormRow,
  updateOperationSummary,
  updateSecurityScheme,
  updateSelectedAuthTab,
  updateSelectedScopes,
  updateSelectedSecuritySchemes,
  updateSelectedServer,
  updateServer,
  updateServerVariables,
  upsertEnvironment,
  upsertEnvironmentVariable
};
