// @scalar/workspace-store/dist/mutators/operation.js
import { getResolvedRef } from "../helpers/get-resolved-ref.js";
const updateOperationSummary = ({
  document,
  meta,
  payload: { summary }
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  operation.summary = summary;
};
const updateOperationMethodDraft = ({
  document,
  meta,
  payload: { method }
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  operation["x-scalar-method"] = method;
};
const updateOperationPathDraft = ({
  document,
  meta,
  payload: { path }
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  operation["x-scalar-path"] = path;
  const pathVariables = Array.from(path.matchAll(/{([^\/}]+)}/g), (m) => m[1]);
  const pathVariablesWithoutPathParameters = operation.parameters?.filter((it) => getResolvedRef(it).in !== "path");
  operation.parameters = [
    ...pathVariablesWithoutPathParameters ?? [],
    ...pathVariables.map((it) => ({
      name: it ?? "",
      in: "path",
      required: true
    }))
  ];
};
const addOperationParameter = ({
  document,
  meta,
  payload,
  type
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  if (!operation.parameters) {
    operation.parameters = [];
  }
  operation.parameters.push({
    name: payload.key,
    in: type,
    required: type === "path" ? true : false,
    examples: {
      [meta.exampleKey]: {
        value: payload.value,
        "x-disabled": !payload.isEnabled
      }
    }
  });
};
const updateOperationParameter = ({
  document,
  meta,
  type,
  payload,
  index
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  const resolvedParameters = operation.parameters?.map((it) => getResolvedRef(it)).filter((it) => it.in === type) ?? [];
  const parameter = resolvedParameters[index];
  if (!parameter) {
    return;
  }
  parameter.name = payload.key ?? parameter.name ?? "";
  if ("examples" in parameter) {
    if (!parameter.examples) {
      parameter.examples = {};
    }
    const example = getResolvedRef(parameter.examples[meta.exampleKey]);
    if (!example) {
      return;
    }
    example.value = payload.value ?? example?.value ?? "";
    example["x-disabled"] = payload.isEnabled === void 0 ? example["x-disabled"] : !payload.isEnabled;
  }
};
const deleteOperationParameter = ({
  document,
  meta,
  index,
  type
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  const resolvedParameters = operation.parameters?.map((it) => getResolvedRef(it)).filter((it) => it.in === type) ?? [];
  const parameter = resolvedParameters[index];
  if (!parameter) {
    return;
  }
  const actualIndex = operation.parameters?.findIndex((it) => getResolvedRef(it) === parameter);
  operation.parameters?.splice(actualIndex, 1);
};
const deleteAllOperationParameters = ({
  document,
  meta,
  type
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  operation.parameters = operation.parameters?.filter((it) => getResolvedRef(it).in !== type) ?? [];
};
const updateOperationRequestBodyContentType = ({
  document,
  meta,
  payload
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  let requestBody = getResolvedRef(operation.requestBody);
  if (!requestBody) {
    operation.requestBody = {
      content: {}
    };
    requestBody = getResolvedRef(operation.requestBody);
  }
  if (!requestBody["x-scalar-selected-content-type"]) {
    requestBody["x-scalar-selected-content-type"] = {};
  }
  requestBody["x-scalar-selected-content-type"][meta.exampleKey] = payload.contentType;
};
const updateOperationRequestBodyExample = ({
  document,
  meta,
  payload,
  contentType
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  let requestBody = getResolvedRef(operation.requestBody);
  if (!requestBody) {
    operation.requestBody = {
      content: {}
    };
    requestBody = getResolvedRef(operation.requestBody);
  }
  if (!requestBody.content[contentType]) {
    requestBody.content[contentType] = {
      examples: {}
    };
  }
  const mediaType = requestBody.content[contentType];
  mediaType.examples ??= {};
  const examples = getResolvedRef(mediaType.examples);
  const example = getResolvedRef(examples[meta.exampleKey]);
  if (!example) {
    examples[meta.exampleKey] = {
      value: payload.value
    };
    return;
  }
  example.value = payload.value;
};
const addOperationRequestBodyFormRow = ({
  document,
  meta,
  payload,
  contentType
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  let requestBody = getResolvedRef(operation.requestBody);
  if (!requestBody) {
    operation.requestBody = {
      content: {}
    };
    requestBody = getResolvedRef(operation.requestBody);
  }
  if (!requestBody.content[contentType]) {
    requestBody.content[contentType] = {
      examples: {}
    };
  }
  if (!requestBody.content[contentType].examples) {
    requestBody.content[contentType].examples = {};
  }
  const examples = getResolvedRef(requestBody.content[contentType].examples);
  const example = getResolvedRef(examples[meta.exampleKey]);
  if (!example || !Array.isArray(example.value)) {
    examples[meta.exampleKey] = {
      value: [
        {
          name: payload.key,
          value: payload.value
        }
      ]
    };
    return;
  }
  example.value.push({
    name: payload.key ?? "",
    value: payload.value ?? ""
  });
};
const updateOperationRequestBodyFormRow = ({
  document,
  meta,
  index,
  payload,
  contentType
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  let requestBody = getResolvedRef(operation.requestBody);
  if (!requestBody) {
    operation.requestBody = {
      content: {}
    };
    requestBody = getResolvedRef(operation.requestBody);
  }
  if (!requestBody.content[contentType]) {
    return;
  }
  const examples = getResolvedRef(requestBody.content[contentType].examples);
  if (!examples) {
    return;
  }
  const example = getResolvedRef(examples[meta.exampleKey]);
  if (!example || !Array.isArray(example.value)) {
    return;
  }
  example.value[index] = {
    name: payload.key ?? example.value[index]?.name ?? "",
    value: payload.value === null ? void 0 : payload.value ?? example.value[index]?.value ?? ""
  };
};
const deleteOperationRequestBodyFormRow = ({
  document,
  meta,
  index,
  contentType
}) => {
  if (!document) {
    return;
  }
  const operation = getResolvedRef(document.paths?.[meta.path]?.[meta.method]);
  if (!operation) {
    return;
  }
  const requestBody = getResolvedRef(operation.requestBody);
  if (!requestBody) {
    return;
  }
  if (!requestBody.content[contentType]) {
    return;
  }
  const examples = getResolvedRef(requestBody.content[contentType].examples);
  if (!examples) {
    return;
  }
  const example = getResolvedRef(examples[meta.exampleKey]);
  if (!example || !Array.isArray(example.value)) {
    return;
  }
  example.value.splice(index, 1);
  if (example.value.length === 0) {
    delete requestBody.content[contentType].examples[meta.exampleKey];
  }
};
export {
  addOperationParameter,
  addOperationRequestBodyFormRow,
  deleteAllOperationParameters,
  deleteOperationParameter,
  deleteOperationRequestBodyFormRow,
  updateOperationMethodDraft,
  updateOperationParameter,
  updateOperationPathDraft,
  updateOperationRequestBodyContentType,
  updateOperationRequestBodyExample,
  updateOperationRequestBodyFormRow,
  updateOperationSummary
};
