// @scalar/workspace-store/dist/navigation/get-navigation-options.js
import { slug } from "github-slugger";
const getNavigationOptions = (documentName, config) => {
  const referenceConfig = config?.["x-scalar-reference-config"];
  const generateId = (props) => {
    const documentId = `${slug(documentName)}`;
    if (props.type === "text") {
      if (referenceConfig?.generateHeadingSlug) {
        return referenceConfig?.generateHeadingSlug({ slug: props.slug });
      }
      if (props.slug) {
        return `${documentId}/description/${props.slug}`;
      }
      return `${documentId}/`;
    }
    if (props.type === "tag") {
      if (referenceConfig?.generateTagSlug) {
        return `${documentId}/tag/${referenceConfig.generateTagSlug(props.tag)}`;
      }
      return `${documentId}/tag/${slug(props.tag.name ?? "")}`;
    }
    if (props.type === "operation") {
      const tagId = `${generateId({
        type: "tag",
        tag: props.parentTag.tag,
        parentId: props.parentTag.id
      })}`;
      if (referenceConfig?.generateOperationSlug) {
        return `${tagId}/${referenceConfig.generateOperationSlug({
          path: props.path,
          operationId: props.operation.operationId,
          method: props.method,
          summary: props.operation.summary
        })}`;
      }
      return `${tagId}/${props.method}${props.path}`;
    }
    if (props.type === "webhook") {
      const prefixTag = props.parentTag ? `${generateId({
        type: "tag",
        parentId: props.parentTag.id,
        tag: props.parentTag.tag
      })}/` : `${documentId}/`;
      if (referenceConfig?.generateWebhookSlug) {
        return `${prefixTag}webhook/${referenceConfig.generateWebhookSlug({
          name: props.name,
          method: props.method
        })}`;
      }
      return `${prefixTag}webhook/${props.method}/${slug(props.name)}`;
    }
    if (props.type === "model") {
      if (!props.name) {
        return `${documentId}/models`;
      }
      const prefixTag = props.parentTag ? `${generateId({
        type: "tag",
        parentId: props.parentTag.id,
        tag: props.parentTag.tag
      })}/` : `${documentId}/`;
      if (referenceConfig?.generateModelSlug) {
        return `${prefixTag}model/${referenceConfig.generateModelSlug({
          name: props.name
        })}`;
      }
      return `${prefixTag}model/${slug(props.name)}`;
    }
    if (props.type === "example") {
      return `${props.parentId}/example/${slug(props.name)}`;
    }
    if (props.type === "document") {
      return documentId;
    }
    console.warn("[WARNING]: unhandled id generation for navigation item:", props);
    return "unknown-id";
  };
  const hideModels = referenceConfig?.features?.showModels === false;
  const operationsSorter = referenceConfig?.operationsSorter;
  const tagsSorter = referenceConfig?.tagSort;
  return {
    hideModels,
    operationsSorter,
    tagsSorter,
    generateId
  };
};
export {
  getNavigationOptions
};
