// @scalar/api-client/dist/v2/hooks/use-workspace-client-events.js
import { mergeObjects as i } from "@scalar/workspace-store/helpers/merge-object";
import { toggleDocumentSecurity as d, upsertEnvironment as u, upsertEnvironmentVariable as v, deleteSecurityScheme as m, updateSelectedAuthTab as c, updateSecurityScheme as s, updateSelectedScopes as y, updateSelectedSecuritySchemes as S, addServer as h, updateServer as O, deleteServer as R, updateServerVariables as q, updateSelectedServer as b, updateOperationMethod as f, updateOperationPath as g, updateOperationSummary as w, addOperationParameter as x, updateOperationParameter as P, deleteOperationParameter as E, deleteAllOperationParameters as T, updateOperationRequestBodyContentType as C, updateOperationRequestBodyExample as F, addOperationRequestBodyFormRow as k, updateOperationRequestBodyFormRow as A, deleteOperationRequestBodyFormRow as V } from "@scalar/workspace-store/mutators";
const M = (r, a, n) => {
  const p = (e, o) => o === "document" ? e.value : n.workspace;
  r.on(
    "document:update:icon",
    (e) => a.value && (a.value["x-scalar-client-config-icon"] = e)
  ), r.on("document:update:info", (e) => a.value && i(a.value.info, e)), r.on("document:toggle:document-security", () => d(a.value)), r.on(
    "environment:upsert:environment",
    (e) => u(a.value, n.workspace, e)
  ), r.on(
    "environment:delete:environment",
    ({ environmentName: e, collectionType: o }) => delete p(a, o)?.["x-scalar-environments"]?.[e]
  ), r.on(
    "environment:upsert:environment-variable",
    (e) => v(p(a, e.collectionType), e)
  ), r.on(
    "environment:delete:environment-variable",
    ({ environmentName: e, index: o, collectionType: t }) => p(a, t)?.["x-scalar-environments"]?.[e]?.variables?.splice(o, 1)
  ), r.on("auth:delete:security-scheme", (e) => m(a.value, e)), r.on("auth:update:active-index", (e) => c(a.value, e)), r.on("auth:update:security-scheme", (e) => s(a.value, e)), r.on("auth:update:selected-scopes", (e) => y(a.value, e)), r.on(
    "auth:update:selected-security-schemes",
    (e) => S(a.value, e)
  ), r.on("server:add:server", () => h(a.value)), r.on("server:update:server", (e) => O(a.value, e)), r.on("server:delete:server", (e) => R(a.value, e)), r.on("server:update:variables", (e) => q(a.value, e)), r.on("server:update:selected", (e) => b(a.value, e)), r.on("operation:update:method", ({ payload: e, meta: o }) => {
    f({
      document: a.value,
      meta: o,
      payload: e
    });
  }), r.on("operation:update:path", ({ meta: e, payload: o }) => {
    g({
      document: a.value,
      meta: e,
      payload: o
    });
  }), r.on("operation:update:summary", ({ payload: e, meta: o }) => {
    w({
      document: a.value,
      meta: o,
      payload: e
    });
  }), r.on("operation:add:parameter", ({ type: e, meta: o, payload: t }) => {
    x({
      document: a.value,
      meta: o,
      payload: t,
      type: e
    });
  }), r.on("operation:update:parameter", ({ meta: e, payload: o, index: t, type: l }) => {
    P({
      document: a.value,
      meta: e,
      payload: o,
      index: t,
      type: l
    });
  }), r.on("operation:delete:parameter", ({ index: e, meta: o, type: t }) => {
    E({
      document: a.value,
      meta: o,
      index: e,
      type: t
    });
  }), r.on("operation:delete-all:parameters", ({ meta: e, type: o }) => {
    T({
      document: a.value,
      meta: e,
      type: o
    });
  }), r.on("operation:update:requestBody:contentType", ({ payload: e, meta: o }) => {
    C({
      document: a.value,
      meta: o,
      payload: e
    });
  }), r.on("operation:update:requestBody:value", ({ contentType: e, payload: o, meta: t }) => {
    F({
      document: a.value,
      meta: t,
      payload: o,
      contentType: e
    });
  }), r.on("operation:add:requestBody:formRow", ({ payload: e, contentType: o, meta: t }) => {
    k({
      document: a.value,
      meta: t,
      payload: e,
      contentType: o
    });
  }), r.on("operation:update:requestBody:formRow", ({ index: e, payload: o, contentType: t, meta: l }) => {
    A({
      document: a.value,
      meta: l,
      index: e,
      payload: o,
      contentType: t
    });
  }), r.on("operation:delete:requestBody:formRow", ({ index: e, contentType: o, meta: t }) => {
    V({
      document: a.value,
      meta: t,
      index: e,
      contentType: o
    });
  });
};
export {
  M as useWorkspaceClientEvents
};
