// @scalar/api-client/dist/v2/hooks/use-sidebar-state.js
import { isDefined as c } from "@scalar/helpers/array/is-defined";
import { createSidebarState as I, generateReverseIndex as b } from "@scalar/sidebar";
import { computed as i, toValue as w, watch as R } from "vue";
import { useRouter as C } from "vue-router";
const U = ({
  workspaceStore: l,
  workspaceSlug: h,
  documentSlug: x,
  path: f,
  method: y,
  exampleName: S
}) => {
  const m = C(), s = i(() => Object.values(w(l).workspace.documents).map((e) => e["x-scalar-navigation"]).filter(c)), n = I(s), a = (e, t) => {
    if (t)
      return t.type === e ? t : a(e, t.parent);
  }, d = ({
    document: e,
    path: t,
    method: r,
    example: o
  }) => JSON.stringify([e, t, r, o].filter(c)), u = i(
    () => b({
      items: s.value,
      nestedKey: "children",
      filter: (e) => e.type === "document" || e.type === "operation" || e.type === "example",
      getId: (e) => {
        const t = a("document", e), r = a("operation", e);
        return d({
          document: t?.name ?? "",
          path: r?.path,
          method: r?.method,
          example: e.type === "example" ? e.name : void 0
        });
      }
    })
  ), E = (e) => {
    const t = u.value.get(d(e));
    return t || u.value.get(
      d({
        document: e.document,
        path: e.path,
        method: e.method
      })
    );
  }, g = (e) => {
    const t = n.getEntryById(e);
    if (!t) {
      console.warn(`Could not find sidebar entry with id ${e} to select`);
      return;
    }
    if (t.type === "document")
      return n.setSelected(e), n.setExpanded(e, !n.isExpanded(e)), m.push({
        name: "document.overview",
        params: { documentSlug: t.name }
      });
    if (t.type === "operation") {
      if (n.isSelected(e)) {
        n.setExpanded(e, !n.isExpanded(e));
        return;
      }
      const r = t.children?.find((o) => o.type === "example");
      return r ? (n.setSelected(r.id), n.setExpanded(r.id, !0)) : n.setSelected(e), m.push({
        name: "example",
        params: {
          documentSlug: a("document", t)?.name,
          pathEncoded: encodeURIComponent(t.path),
          method: t.method,
          exampleName: r?.name ?? "default"
        }
      });
    }
    if (t.type === "example") {
      n.setSelected(e), n.setExpanded(e, !0);
      const r = a("operation", t);
      return m.push({
        name: "example",
        params: {
          documentSlug: a("document", t)?.name,
          pathEncoded: encodeURIComponent(r?.path ?? ""),
          method: r?.method,
          exampleName: t.name
        }
      });
    }
    n.setExpanded(e, !n.isExpanded(e));
  };
  return R(
    [h, x, f, y, S],
    ([e, t, r, o, v]) => {
      if (!t) {
        n.setSelected(null);
        return;
      }
      const p = E({
        document: t,
        path: r,
        method: o,
        example: v
      });
      p && (n.setSelected(p.id), n.setExpanded(p.id, !0));
    },
    {
      immediate: !0
    }
  ), {
    handleSelectItem: g,
    sidebarState: n
  };
};
export {
  U as useSidebarState
};
