// @scalar/workspace-store/dist/navigation/helpers/traverse-document.js
import { getNavigationOptions } from "../../navigation/get-navigation-options.js";
import { traverseDescription } from "./traverse-description.js";
import { traversePaths } from "./traverse-paths.js";
import { traverseSchemas } from "./traverse-schemas.js";
import { traverseTags } from "./traverse-tags.js";
import { traverseWebhooks } from "./traverse-webhooks.js";
const traverseDocument = (documentName, document, config) => {
  const { hideModels, tagsSorter, operationsSorter, generateId } = getNavigationOptions(documentName, config);
  const documentId = generateId({
    type: "document",
    info: document.info,
    name: documentName
  });
  const tagsMap = new Map(
    document.tags?.map((tag) => [
      tag.name ?? "Untitled Tag",
      { id: generateId({ type: "tag", tag, parentId: documentId }), parentId: documentId, tag, entries: [] }
    ]) ?? []
  );
  const entries = traverseDescription({
    generateId,
    parentId: documentId,
    info: document.info
  });
  traversePaths({ document, tagsMap, generateId, documentId });
  const untaggedWebhooksParentId = generateId({
    type: "webhook",
    name: "",
    parentId: documentId
  });
  const untaggedWebhooks = traverseWebhooks({
    document,
    generateId,
    tagsMap,
    untaggedWebhooksParentId,
    documentId
  });
  const tagsEntries = traverseTags({
    document,
    tagsMap,
    documentId,
    options: { tagsSorter, operationsSorter, generateId }
  });
  entries.push(...tagsEntries);
  if (untaggedWebhooks.length) {
    entries.push({
      type: "tag",
      id: untaggedWebhooksParentId,
      title: "Webhooks",
      name: "Webhooks",
      children: untaggedWebhooks,
      isGroup: false,
      isWebhooks: true
    });
  }
  if (!hideModels && document.components?.schemas) {
    const untaggedModels = traverseSchemas({
      documentId,
      document,
      generateId,
      tagsMap
    });
    if (untaggedModels.length) {
      entries.push({
        type: "models",
        id: generateId({
          type: "model",
          parentId: documentId
        }),
        title: "Models",
        name: "Models",
        children: untaggedModels
      });
    }
  }
  return {
    id: documentId,
    type: "document",
    title: document.info.title,
    name: documentName,
    children: entries
  };
};
export {
  traverseDocument
};
