// @scalar/api-client/dist/v2/blocks/scalar-operation-block/components/OperationParams.vue.js
import { defineComponent as R, computed as s, createBlock as i, openBlock as m, unref as n, withCtx as o, createVNode as d, createElementVNode as p, createCommentVNode as g, withModifiers as h, createTextVNode as u, toDisplayString as f } from "vue";
import { ScalarTooltip as y, ScalarButton as P } from "@scalar/components";
import A from "./OperationTable.vue.js";
import { getParameterExample as C } from "../helpers/get-parameter-example.js";
import { getParameterSchema as x } from "../helpers/get-parameter-schema.js";
import K from "../../../components/layout/CollapsibleSection.vue.js";
const k = { class: "text-c-2 request-meta-buttons flex whitespace-nowrap opacity-0 group-hover/params:opacity-100 has-[:focus-visible]:opacity-100" }, B = { class: "sr-only" }, E = /* @__PURE__ */ R({
  __name: "OperationParams",
  props: {
    parameters: {},
    exampleKey: {},
    title: {},
    label: {},
    invalidParams: {},
    globalRoute: {},
    showAddRowPlaceholder: { type: Boolean, default: !0 },
    environment: {}
  },
  emits: ["add", "update", "delete", "deleteAll"],
  setup(l, { emit: v }) {
    const r = v, w = s(
      () => l.parameters.map((e) => {
        const t = C(e, l.exampleKey);
        return {
          name: e.name,
          value: t?.value ?? "",
          globalRoute: l.globalRoute,
          schema: x(e),
          isRequired: e.required,
          isDisabled: t?.["x-disabled"] ?? !1
        };
      })
    ), b = s(() => l.parameters.length > 1);
    return (e, t) => (m(), i(n(K), {
      class: "group/params",
      itemCount: e.parameters.length
    }, {
      title: o(() => [
        u(f(e.title), 1)
      ]),
      actions: o(() => [
        p("div", k, [
          b.value ? (m(), i(n(y), {
            key: 0,
            content: "Clear optional parameters",
            placement: "left"
          }, {
            default: o(() => [
              d(n(P), {
                class: "pr-0.75 pl-1 transition-none",
                size: "sm",
                variant: "ghost",
                onClick: t[0] || (t[0] = h((a) => r("deleteAll"), ["stop"]))
              }, {
                default: o(() => [
                  t[4] || (t[4] = u(" Clear ", -1)),
                  p("span", B, "All " + f(e.title), 1)
                ]),
                _: 1
              })
            ]),
            _: 1
          })) : g("", !0)
        ])
      ]),
      default: o(() => [
        d(A, {
          class: "flex-1",
          columns: ["32px", "", ""],
          data: w.value,
          environment: e.environment,
          exampleKey: e.exampleKey,
          globalRoute: e.globalRoute,
          invalidParams: e.invalidParams,
          label: e.label,
          showAddRowPlaceholder: e.showAddRowPlaceholder,
          onAddRow: t[1] || (t[1] = (a) => r("add", a)),
          onDeleteRow: t[2] || (t[2] = (a) => r("delete", { index: a })),
          onUpdateRow: t[3] || (t[3] = (a, c) => r("update", { index: a, payload: c }))
        }, null, 8, ["data", "environment", "exampleKey", "globalRoute", "invalidParams", "label", "showAddRowPlaceholder"])
      ]),
      _: 1
    }, 8, ["itemCount"]));
  }
});
export {
  E as default
};
