// @scalar/api-client/dist/v2/blocks/scalar-operation-block/components/OperationBlock.vue2.js
import { defineComponent as L, computed as v, ref as G, useId as V, watch as X, createBlock as U, openBlock as u, unref as t, withCtx as q, withDirectives as h, createElementBlock as x, createVNode as s, vShow as f, Fragment as J, renderList as W, createCommentVNode as K, resolveDynamicComponent as Y, createElementVNode as Z, toDisplayString as ee } from "vue";
import { ScalarErrorBoundary as te } from "@scalar/components";
import { canMethodHaveBody as w } from "@scalar/helpers/http/can-method-have-body";
import { REGEX as oe } from "@scalar/helpers/regex/regex-helpers";
import { getResolvedRef as C } from "@scalar/workspace-store/helpers/get-resolved-ref";
import ae from "../../../../components/SectionFilter.vue.js";
import le from "../../../../components/ViewLayout/ViewLayoutSection.vue.js";
import ne from "./OperationBody.vue.js";
import B from "./OperationParams.vue.js";
import { groupBy as re } from "../helpers/group-by.js";
import de from "../../scalar-auth-selector-block/components/AuthSelector.vue.js";
const ie = { class: "group pointer-events-none flex flex-1 items-center gap-1 lg:pr-24" }, ue = ["for"], se = ["id", "placeholder", "value"], me = {
  key: 2,
  class: "text-c-1 flex h-8 items-center"
}, Ue = /* @__PURE__ */ L({
  __name: "OperationBlock",
  props: {
    method: {},
    path: {},
    operation: {},
    authMeta: {},
    exampleKey: {},
    securitySchemes: {},
    selectedSecurity: {},
    security: {},
    server: {},
    layout: {},
    plugins: {},
    eventBus: {},
    environment: {}
  },
  setup(E) {
    const R = E, {
      operation: r,
      method: A,
      layout: k,
      securitySchemes: S,
      path: _,
      exampleKey: d,
      security: P,
      eventBus: l,
      environment: m
    } = R, H = R.authMeta ?? { type: "document" }, n = v(() => ({
      method: A,
      path: _,
      exampleKey: d
    })), p = v(
      () => re(r.parameters?.map((e) => C(e)) ?? [], "in")
    ), O = [
      "Auth",
      "Variables",
      "Cookies",
      "Headers",
      "Query",
      "Body"
    ], c = G("All"), b = v(() => {
      const e = /* @__PURE__ */ new Set(["All", ...O]);
      return p.value.path?.length || e.delete("Variables"), w(A) || e.delete("Body"), g.value && e.delete("Auth"), [...e];
    }), i = v(
      () => Object.fromEntries(
        b.value.map((e) => [e, V()])
      )
    );
    X(
      () => A,
      (e) => {
        c.value === "Body" && !w(e) && (c.value = "All");
      }
    );
    const g = v(
      () => k === "modal" && !r.security && !Object.keys(S ?? {}).length
    ), F = v(() => r.summary ? r.summary : _.replace(oe.PROTOCOL, "") || "Request Name"), y = (e) => c.value === "All" || c.value === e, I = (e) => {
      const a = e.target.value;
      l.emit("operation:update:summary", {
        meta: n.value,
        payload: { summary: a }
      });
    }, o = (e) => ({
      onAdd: (a) => {
        l.emit("operation:add:parameter", {
          type: e,
          payload: {
            key: a.key ?? "",
            value: a.value ?? "",
            isEnabled: !0
          },
          meta: n.value
        });
      },
      onDelete: (a) => {
        l.emit("operation:delete:parameter", {
          type: e,
          index: a.index,
          meta: n.value
        });
      },
      onDeleteAll: () => {
        l.emit("operation:delete-all:parameters", {
          type: e,
          meta: n.value
        });
      },
      onUpdate: (a) => {
        l.emit("operation:update:parameter", {
          type: e,
          index: a.index,
          payload: a.payload,
          meta: n.value
        });
      }
    }), N = (e) => l.emit("operation:add:requestBody:formRow", {
      contentType: e.contentType,
      meta: n.value,
      payload: {
        key: e.data.key ?? "",
        value: e.data.value ?? ""
      }
    }), M = (e) => l.emit("operation:delete:requestBody:formRow", {
      contentType: e.contentType,
      index: e.index,
      meta: n.value
    }), Q = (e) => l.emit("operation:update:requestBody:contentType", {
      payload: { contentType: e.value },
      meta: n.value
    }), $ = (e) => l.emit("operation:update:requestBody:formRow", {
      contentType: e.contentType,
      meta: n.value,
      index: e.index,
      payload: {
        key: e.data.key ?? "",
        value: e.data.value ?? ""
      }
    }), j = (e) => l.emit("operation:update:requestBody:value", {
      contentType: e.contentType,
      payload: { value: e.value ?? "" },
      meta: n.value
    }), T = V();
    return (e, a) => (u(), U(le, {
      "aria-label": `Request: ${t(r).summary}`
    }, {
      title: q(() => [
        Z("div", ie, [
          t(k) !== "modal" ? (u(), x("label", {
            key: 0,
            class: "pointer-events-auto absolute top-0 left-0 h-full w-full cursor-text opacity-0",
            for: t(T)
          }, null, 8, ue)) : K("", !0),
          t(k) !== "modal" ? (u(), x("input", {
            key: 1,
            id: t(T),
            class: "text-c-1 group-hover-input pointer-events-auto relative z-10 -ml-0.5 h-8 w-full rounded pl-1.25 has-[:focus-visible]:outline md:-ml-1.25",
            placeholder: F.value,
            value: t(r).summary,
            onInput: I
          }, null, 40, se)) : (u(), x("span", me, ee(t(r).summary), 1))
        ]),
        s(ae, {
          modelValue: c.value,
          "onUpdate:modelValue": a[0] || (a[0] = (D) => c.value = D),
          filterIds: i.value,
          filters: b.value
        }, null, 8, ["modelValue", "filterIds", "filters"])
      ]),
      default: q(() => [
        h(s(t(de), {
          id: i.value.Auth,
          environment: t(m),
          eventBus: t(l),
          meta: t(H),
          security: t(P),
          securitySchemes: t(S),
          selectedSecurity: e.selectedSecurity,
          server: e.server,
          title: "Authorization"
        }, null, 8, ["id", "environment", "eventBus", "meta", "security", "securitySchemes", "selectedSecurity", "server"]), [
          [f, y("Auth") && !g.value]
        ]),
        h(s(B, {
          id: i.value.Variables,
          environment: t(m),
          exampleKey: t(d),
          parameters: p.value.path ?? [],
          showAddRowPlaceholder: !1,
          title: "Variables",
          onDelete: o("path").onDelete,
          onDeleteAll: o("path").onDeleteAll,
          onUpdate: o("path").onUpdate
        }, null, 8, ["id", "environment", "exampleKey", "parameters", "onDelete", "onDeleteAll", "onUpdate"]), [
          [f, y("Variables") && p.value.path?.length]
        ]),
        h(s(B, {
          id: i.value.Cookies,
          environment: t(m),
          exampleKey: t(d),
          parameters: p.value.cookie ?? [],
          showAddRowPlaceholder: !0,
          title: "Cookies",
          onAdd: o("cookie").onAdd,
          onDelete: o("cookie").onDelete,
          onDeleteAll: o("cookie").onDeleteAll,
          onUpdate: o("cookie").onUpdate
        }, null, 8, ["id", "environment", "exampleKey", "parameters", "onAdd", "onDelete", "onDeleteAll", "onUpdate"]), [
          [f, y("Cookies")]
        ]),
        h(s(B, {
          id: i.value.Headers,
          environment: t(m),
          exampleKey: t(d),
          parameters: p.value.header ?? [],
          title: "Headers",
          onAdd: o("header").onAdd,
          onDelete: o("header").onDelete,
          onDeleteAll: o("header").onDeleteAll,
          onUpdate: o("header").onUpdate
        }, null, 8, ["id", "environment", "exampleKey", "parameters", "onAdd", "onDelete", "onDeleteAll", "onUpdate"]), [
          [f, y("Headers")]
        ]),
        h(s(B, {
          id: i.value.Query,
          environment: t(m),
          exampleKey: t(d),
          parameters: p.value.query ?? [],
          title: "Query Parameters",
          onAdd: o("query").onAdd,
          onDelete: o("query").onDelete,
          onDeleteAll: o("query").onDeleteAll,
          onUpdate: o("query").onUpdate
        }, null, 8, ["id", "environment", "exampleKey", "parameters", "onAdd", "onDelete", "onDeleteAll", "onUpdate"]), [
          [f, y("Query")]
        ]),
        h(s(ne, {
          id: i.value.Body,
          environment: t(m),
          exampleKey: t(d),
          requestBody: t(C)(t(r).requestBody),
          title: "Request Body",
          "onAdd:formRow": N,
          "onDelete:fromRow": M,
          "onUpdate:contentType": Q,
          "onUpdate:formRow": $,
          "onUpdate:value": j
        }, null, 8, ["id", "environment", "exampleKey", "requestBody"]), [
          [f, y("Body") && t(w)(t(A))]
        ]),
        (u(!0), x(J, null, W(e.plugins, (D, z) => (u(), U(t(te), { key: z }, {
          default: q(() => [
            D?.components?.request ? (u(), U(Y(D.components.request), {
              key: 0,
              operation: t(r),
              selectedExample: t(d)
            }, null, 8, ["operation", "selectedExample"])) : K("", !0)
          ]),
          _: 2
        }, 1024))), 128))
      ]),
      _: 1
    }, 8, ["aria-label"]));
  }
});
export {
  Ue as default
};
