// @scalar/api-reference/dist/components/Content/Operations/TraversedEntry.vue.js
import { defineComponent, resolveComponent, createElementBlock, openBlock, Fragment, renderList, createBlock, withCtx, createCommentVNode, unref, createVNode } from "vue";
import { getResolvedRef } from "@scalar/workspace-store/helpers/get-resolved-ref";
import _sfc_main$5 from "../Models/Model.vue.js";
import _sfc_main$4 from "../Models/ModelTag.vue.js";
import _sfc_main$1 from "../../Lazy/Lazy.vue.js";
import SectionContainer from "../../Section/SectionContainer.vue.js";
import _sfc_main$2 from "../../../features/Operation/Operation.vue.js";
import _sfc_main$3 from "../Tags/Tag.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "TraversedEntry",
  props: {
    level: { default: 0 },
    entries: {},
    paths: {},
    webhooks: {},
    schemas: {},
    security: {},
    activeServer: {},
    getSecuritySchemes: { type: Function },
    xScalarDefaultClient: {},
    expandedItems: {},
    eventBus: {},
    options: {}
  },
  setup(__props) {
    const isTagGroup = (entry) => entry["type"] === "tag" && entry.isGroup === true;
    const isTag = (entry) => entry["type"] === "tag" && !isTagGroup(entry) && entry.id !== "models";
    const isOperation = (entry) => entry["type"] === "operation";
    const isWebhook = (entry) => entry["type"] === "webhook";
    const isModelsTag = (entry) => entry["type"] === "models";
    const isModel = (entry) => entry["type"] === "model";
    function getPathValue(entry) {
      return isWebhook(entry) ? __props.webhooks[entry.name] : __props.paths[entry.path];
    }
    return (_ctx, _cache) => {
      const _component_TraversedEntry = resolveComponent("TraversedEntry", true);
      return openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.entries, (entry) => {
        return openBlock(), createBlock(_sfc_main$1, {
          id: entry.id,
          key: `${entry.id}-${_ctx.options.layout}`
        }, {
          default: withCtx(() => [
            isOperation(entry) || isWebhook(entry) ? (openBlock(), createBlock(unref(SectionContainer), {
              key: 0,
              omit: _ctx.level !== 0
            }, {
              default: withCtx(() => [
                createVNode(unref(_sfc_main$2), {
                  id: entry.id,
                  eventBus: _ctx.eventBus,
                  getSecurityScheme: _ctx.getSecuritySchemes,
                  isCollapsed: !_ctx.expandedItems[entry.id],
                  method: entry.method,
                  options: {
                    ..._ctx.options,
                    isWebhook: isWebhook(entry)
                  },
                  path: isWebhook(entry) ? entry.name : entry.path,
                  pathValue: getPathValue(entry),
                  security: _ctx.security,
                  server: _ctx.activeServer,
                  xScalarDefaultClient: _ctx.xScalarDefaultClient
                }, null, 8, ["id", "eventBus", "getSecurityScheme", "isCollapsed", "method", "options", "path", "pathValue", "security", "server", "xScalarDefaultClient"])
              ]),
              _: 2
            }, 1032, ["omit"])) : isTag(entry) ? (openBlock(), createBlock(unref(_sfc_main$3), {
              key: 1,
              eventBus: _ctx.eventBus,
              isCollapsed: !_ctx.expandedItems[entry.id],
              isLoading: false,
              layout: _ctx.options.layout,
              moreThanOneTag: _ctx.entries.filter(isTag).length > 1,
              tag: entry
            }, {
              default: withCtx(() => [
                "children" in entry && entry.children?.length ? (openBlock(), createBlock(_component_TraversedEntry, {
                  key: 0,
                  activeServer: _ctx.activeServer,
                  entries: entry.children,
                  eventBus: _ctx.eventBus,
                  expandedItems: _ctx.expandedItems,
                  getSecuritySchemes: _ctx.getSecuritySchemes,
                  level: _ctx.level + 1,
                  options: _ctx.options,
                  paths: _ctx.paths,
                  schemas: _ctx.schemas,
                  security: _ctx.security,
                  webhooks: _ctx.webhooks,
                  xScalarDefaultClient: _ctx.xScalarDefaultClient
                }, null, 8, ["activeServer", "entries", "eventBus", "expandedItems", "getSecuritySchemes", "level", "options", "paths", "schemas", "security", "webhooks", "xScalarDefaultClient"])) : createCommentVNode("", true)
              ]),
              _: 2
            }, 1032, ["eventBus", "isCollapsed", "layout", "moreThanOneTag", "tag"])) : isTagGroup(entry) ? (openBlock(), createBlock(_component_TraversedEntry, {
              key: 2,
              activeServer: _ctx.activeServer,
              entries: entry.children || [],
              eventBus: _ctx.eventBus,
              expandedItems: _ctx.expandedItems,
              getSecuritySchemes: _ctx.getSecuritySchemes,
              level: _ctx.level + 1,
              options: _ctx.options,
              paths: _ctx.paths,
              schemas: _ctx.schemas,
              security: _ctx.security,
              webhooks: _ctx.webhooks,
              xScalarDefaultClient: _ctx.xScalarDefaultClient
            }, null, 8, ["activeServer", "entries", "eventBus", "expandedItems", "getSecuritySchemes", "level", "options", "paths", "schemas", "security", "webhooks", "xScalarDefaultClient"])) : isModelsTag(entry) && _ctx.schemas ? (openBlock(), createBlock(_sfc_main$4, {
              key: 3,
              id: entry.id,
              eventBus: _ctx.eventBus,
              isCollapsed: !_ctx.expandedItems[entry.id],
              options: {
                layout: _ctx.options.layout,
                expandAllModelSections: _ctx.options.expandAllModelSections,
                orderRequiredPropertiesFirst: _ctx.options.orderRequiredPropertiesFirst,
                orderSchemaPropertiesBy: _ctx.options.orderSchemaPropertiesBy
              }
            }, {
              default: withCtx(() => [
                createVNode(_component_TraversedEntry, {
                  activeServer: _ctx.activeServer,
                  entries: entry.children || [],
                  eventBus: _ctx.eventBus,
                  expandedItems: _ctx.expandedItems,
                  getSecuritySchemes: _ctx.getSecuritySchemes,
                  level: _ctx.level + 1,
                  options: _ctx.options,
                  paths: _ctx.paths,
                  schemas: _ctx.schemas,
                  security: _ctx.security,
                  webhooks: _ctx.webhooks,
                  xScalarDefaultClient: _ctx.xScalarDefaultClient
                }, null, 8, ["activeServer", "entries", "eventBus", "expandedItems", "getSecuritySchemes", "level", "options", "paths", "schemas", "security", "webhooks", "xScalarDefaultClient"])
              ]),
              _: 2
            }, 1032, ["id", "eventBus", "isCollapsed", "options"])) : isModel(entry) && _ctx.schemas ? (openBlock(), createBlock(_sfc_main$5, {
              key: 4,
              id: entry.id,
              eventBus: _ctx.eventBus,
              isCollapsed: !_ctx.expandedItems[entry.id],
              name: entry.name,
              options: {
                layout: _ctx.options.layout,
                orderRequiredPropertiesFirst: _ctx.options.orderRequiredPropertiesFirst,
                orderSchemaPropertiesBy: _ctx.options.orderSchemaPropertiesBy
              },
              schema: unref(getResolvedRef)(_ctx.schemas[entry.name])
            }, null, 8, ["id", "eventBus", "isCollapsed", "name", "options", "schema"])) : createCommentVNode("", true)
          ]),
          _: 2
        }, 1032, ["id"]);
      }), 128);
    };
  }
});
export {
  _sfc_main as default
};
