// @scalar/api-client/dist/v2/features/app/helpers/routes.js
const e = [
  // Global settings
  {
    name: "settings",
    path: "/settings",
    component: () => import("../../../components/TempReplaceMe.vue.js")
  },
  {
    path: "/workspace/:workspaceSlug",
    children: [
      {
        path: "document/:documentSlug",
        children: [
          // Example page
          {
            name: "example",
            path: "path/:pathEncoded/method/:method/example/:exampleName",
            component: () => import("../../operation/Operation.vue2.js")
          },
          // Document Page
          {
            name: "document",
            path: "",
            component: () => import("../../collection/DocumentCollection.vue.js"),
            children: [
              // Redirect to overview
              {
                name: "document.redirect",
                path: "",
                redirect: {
                  name: "document.overview"
                }
              },
              // Document overview
              {
                name: "document.overview",
                path: "overview",
                component: () => import("../../collection/components/Overview.vue.js")
              },
              // Document servers
              {
                name: "document.servers",
                path: "servers",
                component: () => import("../../collection/components/Servers.vue.js")
              },
              // Document environment
              {
                name: "document.environment",
                path: "environment",
                component: () => import("../../collection/components/Environment.vue.js")
              },
              // Document authentication
              {
                name: "document.authentication",
                path: "authentication",
                component: () => import("../../collection/components/Authentication.vue.js")
              },
              // Document cookies
              {
                name: "document.cookies",
                path: "cookies",
                component: () => import("../../collection/components/Cookies.vue.js")
              },
              // Document settings
              {
                name: "document.settings",
                path: "settings",
                component: () => import("../../collection/components/Settings.vue.js")
              }
            ]
          }
        ]
      },
      // Workspace page
      {
        name: "workspace",
        path: "",
        component: () => import("../../collection/WorkspaceCollection.vue.js"),
        children: [
          // Workspace environment
          {
            name: "workspace.environment",
            path: "environment",
            component: () => import("../../collection/components/Environment.vue.js")
          },
          // Workspace cookies
          {
            name: "workspace.cookies",
            path: "cookies",
            component: () => import("../../collection/components/Cookies.vue.js")
          },
          // Workspace settings
          {
            name: "workspace.settings",
            path: "settings",
            component: () => import("../../collection/components/Settings.vue.js")
          }
        ]
      }
    ]
  },
  {
    path: "/:pathMatch(.*)*",
    redirect: "/workspace/default/document/default/overview"
  }
];
export {
  e as ROUTES
};
