// @scalar/api-reference/dist/features/toolbar/ApiReferenceToolbarRegisterButton.vue.js
import { defineComponent, mergeModels, useModel, createBlock, openBlock, unref, withCtx, renderSlot, createTextVNode, nextTick } from "vue";
import { useLoadingState, ScalarButton } from "@scalar/components";
import { useToasts } from "@scalar/use-toasts";
import { DASHBOARD_REGISTER_URL } from "../../consts/urls.js";
import { uploadTempDocument } from "../../helpers/upload-temp-document.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ApiReferenceToolbarRegisterButton",
  props: /* @__PURE__ */ mergeModels({
    workspace: {},
    sdks: { default: () => [] }
  }, {
    "url": {},
    "urlModifiers": {}
  }),
  emits: ["update:url"],
  setup(__props) {
    const tempDocUrl = useModel(__props, "url");
    const { toast } = useToasts();
    const loading = useLoadingState();
    function openRegisterLink(docUrl) {
      const url = new URL(DASHBOARD_REGISTER_URL);
      url.searchParams.set("url", docUrl);
      __props.sdks.forEach((sdk) => url.searchParams.append("sdk", sdk));
      window.open(url.toString(), "_blank");
    }
    async function generateRegisterLink() {
      if (loading.isLoading || !__props.workspace) {
        return;
      }
      if (tempDocUrl.value) {
        openRegisterLink(tempDocUrl.value);
        return;
      }
      loading.startLoading();
      const document = __props.workspace.exportActiveDocument("json");
      if (!document) {
        toast("Unable to export active document", "error");
        loading.invalidate();
        return;
      }
      try {
        tempDocUrl.value = await uploadTempDocument(document);
        await loading.validate(600);
        openRegisterLink(tempDocUrl.value);
        await nextTick();
        loading.clear();
      } catch (error) {
        const message = error instanceof Error ? error.message : "An unknown error occurred";
        toast(message, "error");
        loading.invalidate();
      }
    }
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(ScalarButton), {
        class: "h-auto p-2.5",
        loading: unref(loading),
        onClick: generateRegisterLink
      }, {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "default", {}, () => [
            _cache[0] || (_cache[0] = createTextVNode("Generate", -1))
          ])
        ]),
        _: 3
      }, 8, ["loading"]);
    };
  }
});
export {
  _sfc_main as default
};
