// @scalar/api-reference/dist/features/toolbar/ApiReferenceToolbarConfigTheme.vue.js
import { defineComponent, useModel, computed, createBlock, openBlock, unref, withCtx, createVNode, createTextVNode, toDisplayString } from "vue";
import { ScalarCombobox, ScalarListboxInput } from "@scalar/components";
import { themeIds, themeLabels } from "@scalar/themes";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ApiReferenceToolbarConfigTheme",
  props: {
    "modelValue": {},
    "modelModifiers": {}
  },
  emits: ["update:modelValue"],
  setup(__props) {
    const model = useModel(__props, "modelValue");
    const options = computed(
      () => themeIds.map((id) => ({ id, label: themeLabels[id] }))
    );
    const selected = computed({
      get: () => {
        const theme = model.value ?? "default";
        return options.value.find((o) => o.id === theme) ?? options.value[0];
      },
      set: (option) => model.value = option.id
    });
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(ScalarCombobox), {
        modelValue: selected.value,
        "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => selected.value = $event),
        options: options.value,
        resize: ""
      }, {
        default: withCtx(({ open }) => [
          createVNode(unref(ScalarListboxInput), { open }, {
            default: withCtx(() => [
              createTextVNode(toDisplayString(selected.value.label), 1)
            ]),
            _: 1
          }, 8, ["open"])
        ]),
        _: 1
      }, 8, ["modelValue", "options"]);
    };
  }
});
export {
  _sfc_main as default
};
