// @scalar/api-client/dist/v2/features/global-cookies/components/CookiesTable.vue2.js
import { defineComponent as w, computed as D, createBlock as v, openBlock as f, unref as e, withCtx as o, createVNode as a, createElementBlock as U, createTextVNode as d, Fragment as x, renderList as y } from "vue";
import { ScalarButton as R } from "@scalar/components";
import { ScalarIconTrash as T } from "@scalar/icons";
import B from "../../../../components/DataTable/DataTable.vue.js";
import c from "../../../../components/DataTable/DataTableRow.vue.js";
import i from "../../../../components/DataTable/DataTableHeader.vue.js";
import $ from "../../../../components/DataTable/DataTableCheckbox.vue.js";
import p from "../../../../components/DataTable/DataTableCell.vue.js";
import b from "../../../../components/CodeInput/CodeInput.vue.js";
const j = /* @__PURE__ */ w({
  __name: "CookiesTable",
  props: {
    data: {},
    environment: {},
    envVariables: {}
  },
  emits: ["addRow", "updateRow", "deleteRow"],
  setup(m, { emit: C }) {
    const V = C, g = D(() => {
      const l = m.data.at(-1);
      return !l || l.name !== "" || l.value !== "" || l.domain !== "" ? [...m.data, { name: "", value: "", domain: "", isDisabled: !0 }] : m.data;
    }), u = ({
      index: l,
      payload: n
    }) => {
      if (l >= m.data.length) {
        V("addRow", n);
        return;
      }
      V("updateRow", l, n);
    }, k = ["32px", "1fr", "1fr", "1fr", "36px"];
    return (l, n) => (f(), v(e(B), {
      class: "group/table data-table h-min flex-1 rounded border",
      columns: k
    }, {
      default: o(() => [
        a(e(c), { class: "sr-only !block" }, {
          default: o(() => [
            a(e(i), null, {
              default: o(() => [...n[0] || (n[0] = [
                d("Enabled", -1)
              ])]),
              _: 1
            }),
            a(e(i), null, {
              default: o(() => [...n[1] || (n[1] = [
                d("Name", -1)
              ])]),
              _: 1
            }),
            a(e(i), null, {
              default: o(() => [...n[2] || (n[2] = [
                d("Value", -1)
              ])]),
              _: 1
            }),
            a(e(i), null, {
              default: o(() => [...n[3] || (n[3] = [
                d("Domain", -1)
              ])]),
              _: 1
            }),
            a(e(i), null, {
              default: o(() => [...n[4] || (n[4] = [
                d("Actions", -1)
              ])]),
              _: 1
            })
          ]),
          _: 1
        }),
        (f(!0), U(x, null, y(g.value, (s, r) => (f(), v(e(c), {
          id: r,
          key: r,
          class: "group/row"
        }, {
          default: o(() => [
            a(e($), {
              class: "!border-r",
              modelValue: !s.isDisabled,
              "onUpdate:modelValue": (t) => u({ index: r, payload: { isDisabled: !t } })
            }, null, 8, ["modelValue", "onUpdate:modelValue"]),
            a(e(p), null, {
              default: o(() => [
                a(e(b), {
                  "aria-label": "Cookie Name",
                  disableCloseBrackets: "",
                  disableTabIndent: "",
                  envVariables: l.envVariables,
                  environment: l.environment,
                  lineWrapping: "",
                  modelValue: s.name,
                  placeholder: "Name",
                  "onUpdate:modelValue": (t) => u({ index: r, payload: { name: t } })
                }, null, 8, ["envVariables", "environment", "modelValue", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            a(e(p), null, {
              default: o(() => [
                a(e(b), {
                  "aria-label": "Cookie Value",
                  disableTabIndent: "",
                  envVariables: l.envVariables,
                  environment: l.environment,
                  lineWrapping: "",
                  modelValue: s.value,
                  placeholder: "Value",
                  "onUpdate:modelValue": (t) => u({ index: r, payload: { value: t } })
                }, null, 8, ["envVariables", "environment", "modelValue", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            a(e(p), null, {
              default: o(() => [
                a(e(b), {
                  "aria-label": "Cookie Domain",
                  disableCloseBrackets: "",
                  disableTabIndent: "",
                  envVariables: l.envVariables,
                  environment: l.environment,
                  lineWrapping: "",
                  modelValue: s.domain,
                  placeholder: "Domain",
                  "onUpdate:modelValue": (t) => u({ index: r, payload: { domain: t } })
                }, null, 8, ["envVariables", "environment", "modelValue", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            a(e(p), { class: "flex items-center justify-center" }, {
              default: o(() => [
                a(e(R), {
                  class: "text-c-2 hover:text-c-1 hover:bg-b-2 hidden h-fit rounded p-1 group-hover/row:flex",
                  size: "sm",
                  variant: "ghost",
                  onClick: (t) => l.$emit("deleteRow", r)
                }, {
                  default: o(() => [
                    a(e(T), { class: "size-3.5" })
                  ]),
                  _: 1
                }, 8, ["onClick"])
              ]),
              _: 2
            }, 1024)
          ]),
          _: 2
        }, 1032, ["id"]))), 128))
      ]),
      _: 1
    }));
  }
});
export {
  j as default
};
