// @scalar/api-client/dist/v2/features/global-cookies/components/GlobalCookies.vue2.js
import { defineComponent as c, createBlock as u, openBlock as l, withCtx as s, createVNode as a, createElementVNode as n, createElementBlock as r, unref as k, Fragment as m, createTextVNode as d, toDisplayString as x } from "vue";
import { ScalarIconCookie as v } from "@scalar/icons";
import b from "../../../../components/ViewLayout/ViewLayout.vue.js";
import g from "../../../../components/ViewLayout/ViewLayoutContent.vue.js";
import w from "../../../../components/ViewLayout/ViewLayoutSection.vue.js";
import V from "./CookiesTable.vue.js";
const _ = { class: "mx-auto flex max-h-full w-full max-w-[720px] !flex-col gap-8 overflow-auto px-5 py-15" }, y = { class: "flex flex-col gap-2" }, C = { class: "flex items-center gap-2 text-xl font-bold" }, T = /* @__PURE__ */ c({
  __name: "GlobalCookies",
  props: {
    documentName: {},
    cookies: {},
    environment: {},
    envVariables: {}
  },
  emits: ["navigation:update:selection", "navigation:update:sidebarWidth", "cookie:add", "cookie:update", "cookie:delete"],
  setup(N, { emit: p }) {
    const i = p;
    return (o, e) => (l(), u(b, null, {
      default: s(() => [
        a(g, { class: "flex-1" }, {
          default: s(() => [
            a(w, null, {
              default: s(() => [
                n("div", _, [
                  n("div", y, [
                    n("h2", C, [
                      a(k(v)),
                      o.documentName ? (l(), r(m, { key: 1 }, [
                        d(x(o.documentName) + " cookies", 1)
                      ], 64)) : (l(), r(m, { key: 0 }, [
                        d(" Global cookies ")
                      ], 64))
                    ]),
                    e[3] || (e[3] = n("p", { class: "text-c-2 mb-4 text-sm" }, " Manage your global cookies here. ", -1))
                  ]),
                  a(V, {
                    data: o.cookies,
                    envVariables: o.envVariables,
                    environment: o.environment,
                    onAddRow: e[0] || (e[0] = (t) => i("cookie:add", t)),
                    onDeleteRow: e[1] || (e[1] = (t) => i("cookie:delete", t)),
                    onUpdateRow: e[2] || (e[2] = (t, f) => i("cookie:update", t, f))
                  }, null, 8, ["data", "envVariables", "environment"])
                ])
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  T as default
};
