// @scalar/api-client/dist/v2/features/collection/DocumentCollection.vue2.js
import { defineComponent as f, computed as d, createElementBlock as s, openBlock as a, createElementVNode as t, createVNode as l, withCtx as r, unref as u, createBlock as p, KeepAlive as v, resolveDynamicComponent as x } from "vue";
import { ScalarButton as h } from "@scalar/components";
import { LibraryIcon as y } from "@scalar/icons/library";
import { RouterView as w } from "vue-router";
import b from "../../../components/IconSelector.vue.js";
import _ from "./components/LabelInput.vue.js";
import k from "./components/Tabs.vue.js";
const B = { class: "custom-scroll h-full" }, V = {
  key: 0,
  class: "w-full md:mx-auto md:max-w-[720px]"
}, g = ["aria-label"], S = { class: "flex flex-row items-center gap-2" }, C = { class: "group relative ml-1.25" }, D = { class: "px-1.5 py-8" }, N = {
  key: 1,
  class: "flex w-full flex-1 items-center justify-center"
}, I = {
  name: "DocumentCollection"
}, z = /* @__PURE__ */ f({
  ...I,
  props: {
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    workspaceStore: {}
  },
  setup(m) {
    const i = d(() => m.document?.info?.title || "Untitled Document"), c = d(
      () => m.document?.["x-scalar-client-config-icon"] || "interface-content-folder"
    );
    return (e, o) => (a(), s("div", B, [
      e.document ? (a(), s("div", V, [
        t("div", {
          "aria-label": `title: ${i.value}`,
          class: "mx-auto flex h-fit w-full flex-col gap-2 pt-6 pb-3 md:mx-auto md:max-w-[720px]"
        }, [
          t("div", S, [
            l(b, {
              modelValue: c.value,
              placement: "bottom-start",
              "onUpdate:modelValue": o[0] || (o[0] = (n) => e.eventBus.emit("document:update:icon", n))
            }, {
              default: r(() => [
                l(u(h), {
                  class: "hover:bg-b-2 aspect-square h-7 w-7 cursor-pointer rounded border border-transparent p-0 hover:border-inherit",
                  variant: "ghost"
                }, {
                  default: r(() => [
                    l(u(y), {
                      class: "text-c-2 size-5",
                      src: c.value,
                      "stroke-width": "2"
                    }, null, 8, ["src"])
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["modelValue"])
          ]),
          t("div", C, [
            l(_, {
              class: "text-xl font-bold",
              inputId: "documentName",
              modelValue: i.value,
              "onUpdate:modelValue": o[1] || (o[1] = (n) => e.eventBus.emit("document:update:info", { title: n }))
            }, null, 8, ["modelValue"])
          ])
        ], 8, g),
        l(k, { type: "document" }),
        t("div", D, [
          l(u(w), null, {
            default: r(({ Component: n }) => [
              (a(), p(v, null, [
                (a(), p(x(n), {
                  collectionType: "document",
                  document: e.document,
                  environment: e.environment,
                  eventBus: e.eventBus,
                  layout: e.layout,
                  workspaceStore: e.workspaceStore
                }, null, 8, ["document", "environment", "eventBus", "layout", "workspaceStore"]))
              ], 1024))
            ]),
            _: 1
          })
        ])
      ])) : (a(), s("div", N, [...o[2] || (o[2] = [
        t("div", { class: "flex h-full flex-col items-center justify-center" }, [
          t("h1", { class: "text-2xl font-bold" }, "Document not found"),
          t("p", { class: "text-gray-500" }, " The document you are looking for does not exist. ")
        ], -1)
      ])]))
    ]));
  }
});
export {
  z as default
};
