// @scalar/api-client/dist/v2/features/collection/WorkspaceCollection.vue2.js
import { defineComponent as s, createElementBlock as c, openBlock as r, createElementVNode as e, createVNode as t, unref as l, toDisplayString as p } from "vue";
import { ScalarIconGlobe as n } from "@scalar/icons";
import { RouterView as m } from "vue-router";
import i from "./components/Tabs.vue.js";
const u = { class: "w-full md:mx-auto md:max-w-[720px]" }, d = ["aria-label"], f = { class: "flex flex-col gap-12 px-1.5 pt-8" }, _ = {}, B = /* @__PURE__ */ s({
  ..._,
  __name: "WorkspaceCollection",
  props: {
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    workspaceStore: {}
  },
  setup(x) {
    const o = "Workspace Title [replace with name]";
    return (a, v) => (r(), c("div", u, [
      e("div", {
        "aria-label": `title: ${o}`,
        class: "mx-auto flex h-fit w-full flex-col gap-2 pt-6 pb-3 md:mx-auto md:max-w-[720px]"
      }, [
        t(l(n), { class: "text-c-2 size-6" }),
        e("div", { class: "group relative ml-1.25" }, [
          e("span", { class: "text-c-1 flex h-8 items-center" }, p(o))
        ])
      ], 8, d),
      t(i, { type: "workspace" }),
      e("div", f, [
        t(l(m), {
          eventBus: a.eventBus,
          layout: a.layout,
          type: "workspace"
        }, null, 8, ["eventBus", "layout"])
      ])
    ]));
  }
});
export {
  B as default
};
