// @scalar/api-client/dist/v2/components/server/ServerDropdownItem.vue2.js
import { defineComponent as h, useId as x, computed as p, createElementBlock as i, openBlock as n, normalizeClass as k, createElementVNode as l, createCommentVNode as u, mergeProps as S, unref as s, createVNode as d, toDisplayString as g, withModifiers as y } from "vue";
import { ScalarListboxCheckbox as w, ScalarMarkdown as C } from "@scalar/components";
import _ from "../../../components/Server/ServerVariablesForm.vue.js";
const V = ["aria-expanded"], O = { class: "overflow-hidden text-ellipsis whitespace-nowrap" }, E = ["id"], I = { key: 0 }, N = { class: "description text-c-3 px-3 py-1.5" }, j = /* @__PURE__ */ h({
  __name: "ServerDropdownItem",
  props: {
    server: {},
    serverOption: {}
  },
  emits: ["update:variable", "update:selectedServer"],
  setup(o, { emit: m }) {
    const v = m, c = x(), b = () => Object.keys(o.server?.variables ?? {}).length > 0, t = p(() => o.serverOption.id === o.server?.url), a = p(() => t.value && b()), f = (e, r) => {
      v("update:variable", e, r);
    };
    return (e, r) => (n(), i("div", {
      class: k(["group/item flex min-h-fit flex-col rounded border", { "border-transparent": !t.value }])
    }, [
      l("button", S(a.value ? { "aria-controls": s(c) } : {}, {
        "aria-expanded": a.value,
        class: ["flex min-h-8 cursor-pointer items-center gap-1.5 rounded px-1.5", t.value ? "text-c-1 bg-b-2" : "hover:bg-b-2"],
        type: "button",
        onClick: r[0] || (r[0] = (B) => v("update:selectedServer"))
      }), [
        d(s(w), { selected: t.value }, null, 8, ["selected"]),
        l("span", O, g(e.serverOption.label), 1)
      ], 16, V),
      a.value ? (n(), i("div", {
        key: 0,
        id: s(c),
        class: "bg-b-2 divide divide-y rounded-b border-t *:pl-4",
        onClick: r[1] || (r[1] = y(() => {
        }, ["stop"]))
      }, [
        d(_, {
          variables: e.server?.variables,
          "onUpdate:variable": f
        }, null, 8, ["variables"]),
        e.server?.description ? (n(), i("div", I, [
          l("div", N, [
            d(s(C), {
              value: e.server.description
            }, null, 8, ["value"])
          ])
        ])) : u("", !0)
      ], 8, E)) : u("", !0)
    ], 2));
  }
});
export {
  j as default
};
