// @scalar/api-client/dist/v2/blocks/operation-block/OperationBlock.vue.js
import { defineComponent as m, computed as s, createElementBlock as d, openBlock as h, createElementVNode as v, createVNode as o, withCtx as a, unref as i } from "vue";
import y from "../../../components/ViewLayout/ViewLayout.vue.js";
import f from "../../../components/ViewLayout/ViewLayoutContent.vue.js";
import B from "./components/Header.vue.js";
import g from "../scalar-operation-block/components/OperationBlock.vue.js";
import S from "../scalar-response-block/components/ResponseBlock.vue.js";
const q = { class: "bg-b-1 flex h-full flex-col" }, b = { class: "lg:min-h-header flex w-full flex-wrap items-center justify-center p-2 lg:p-1" }, R = /* @__PURE__ */ m({
  __name: "OperationBlock",
  props: {
    eventBus: {},
    appVersion: {},
    path: {},
    method: {},
    layout: {},
    server: {},
    servers: {},
    history: {},
    requestLoadingPercentage: {},
    response: {},
    request: {},
    totalPerformedRequests: {},
    isSidebarOpen: { type: Boolean },
    showSidebar: { type: Boolean },
    hideClientButton: { type: Boolean },
    integration: {},
    documentUrl: {},
    source: {},
    operation: {},
    exampleKey: {},
    authMeta: {},
    securitySchemes: {},
    selectedSecurity: {},
    security: {},
    events: {},
    plugins: {},
    environment: {}
  },
  setup(u) {
    const n = u, p = s(
      () => n.operation["x-scalar-method"] ?? n.method
    ), l = s(() => n.operation["x-scalar-path"] ?? n.path);
    return (e, t) => (h(), d("div", q, [
      v("div", b, [
        o(B, {
          documentUrl: e.documentUrl,
          environment: e.environment,
          eventBus: e.eventBus,
          events: e.events,
          hideClientButton: e.hideClientButton,
          history: e.history,
          integration: e.integration,
          isSidebarOpen: e.isSidebarOpen,
          layout: e.layout,
          method: p.value,
          path: l.value,
          percentage: e.requestLoadingPercentage,
          server: e.server,
          servers: e.servers,
          showSidebar: e.showSidebar,
          source: e.source,
          onExecute: t[0] || (t[0] = () => e.eventBus.emit("operation:send:request", {
            meta: { path: e.path, method: e.method, exampleKey: e.exampleKey }
          })),
          "onUpdate:method": t[1] || (t[1] = (r) => e.eventBus.emit("operation:update:method", {
            meta: {
              method: e.method,
              path: e.path
            },
            payload: {
              method: r.value
            }
          })),
          "onUpdate:path": t[2] || (t[2] = (r) => e.eventBus.emit("operation:update:path", {
            meta: {
              method: e.method,
              path: e.path
            },
            payload: {
              path: r.value
            }
          }))
        }, null, 8, ["documentUrl", "environment", "eventBus", "events", "hideClientButton", "history", "integration", "isSidebarOpen", "layout", "method", "path", "percentage", "server", "servers", "showSidebar", "source"])
      ]),
      o(y, null, {
        default: a(() => [
          o(f, { class: "flex flex-1" }, {
            default: a(() => [
              o(i(g), {
                authMeta: e.authMeta,
                environment: e.environment,
                eventBus: e.eventBus,
                exampleKey: e.exampleKey,
                layout: e.layout,
                method: e.method,
                operation: e.operation,
                path: e.path,
                plugins: e.plugins,
                security: e.security,
                securitySchemes: e.securitySchemes,
                selectedSecurity: e.selectedSecurity
              }, null, 8, ["authMeta", "environment", "eventBus", "exampleKey", "layout", "method", "operation", "path", "plugins", "security", "securitySchemes", "selectedSecurity"]),
              o(i(S), {
                appVersion: e.appVersion,
                eventBus: e.eventBus,
                events: e.events,
                layout: e.layout,
                plugins: e.plugins,
                request: e.request,
                response: e.response,
                totalPerformedRequests: e.totalPerformedRequests,
                onSendRequest: t[3] || (t[3] = (r) => e.eventBus.emit("operation:send:request", {
                  meta: { path: e.path, method: e.method, exampleKey: e.exampleKey }
                }))
              }, null, 8, ["appVersion", "eventBus", "events", "layout", "plugins", "request", "response", "totalPerformedRequests"])
            ]),
            _: 1
          })
        ]),
        _: 1
      })
    ]));
  }
});
export {
  R as default
};
