// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarButton.vue.js
import { defineComponent as l, createBlock as o, openBlock as s, resolveDynamicComponent as r, mergeProps as c, unref as a, withCtx as u, renderSlot as t, createElementBlock as f, createCommentVNode as n, createElementVNode as b, createVNode as p } from "vue";
import { cva as m, useBindCx as v } from "@scalar/use-hooks/useBindCx";
import h from "./ScalarSidebarIndent.vue.js";
import y from "../ScalarIcon/ScalarIconLegacyAdapter.vue.js";
const B = {
  key: 0,
  class: "h-[1lh] *:size-4 mr-1"
}, k = { class: "flex-1 min-w-0" }, $ = {}, E = /* @__PURE__ */ l({
  ...$,
  inheritAttrs: !1,
  __name: "ScalarSidebarButton",
  props: {
    is: { default: "a" },
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: { default: 0 }
  },
  setup(g) {
    const i = m({
      base: [
        "group/button flex items-stretch rounded p-2 ",
        "font-sidebar leading-5 text-c-2 no-underline wrap-break-word"
      ],
      variants: {
        active: { true: "text-c-1 font-sidebar-active" },
        disabled: { true: "cursor-auto" },
        selected: { true: "cursor-auto bg-b-2 text-c-1 font-sidebar-active" }
      },
      compoundVariants: [
        { selected: !1, disabled: !1, class: "hover:bg-b-2" }
      ],
      defaultVariants: { selected: !1, disabled: !1, active: !1 }
    }), { cx: d } = v();
    return (e, C) => (s(), o(r(e.is), c({
      "aria-selected": e.selected,
      type: e.is === "button" ? "button" : void 0
    }, a(d)(a(i)({ selected: e.selected, disabled: e.disabled, active: e.active }))), {
      default: u(() => [
        t(e.$slots, "indent", {}, () => [
          p(h, {
            disabled: e.disabled,
            indent: e.indent,
            selected: e.selected
          }, null, 8, ["disabled", "indent", "selected"])
        ]),
        e.icon || e.$slots.icon ? (s(), f("div", B, [
          t(e.$slots, "icon", {}, () => [
            e.icon ? (s(), o(a(y), {
              key: 0,
              icon: e.icon
            }, null, 8, ["icon"])) : n("", !0)
          ])
        ])) : n("", !0),
        b("div", k, [
          t(e.$slots, "default")
        ]),
        t(e.$slots, "aside")
      ]),
      _: 3
    }, 16, ["aria-selected", "type"]));
  }
});
export {
  E as default
};
