// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarNestedItems.vue.js
import { defineComponent as y, mergeModels as C, useModel as k, useTemplateRef as g, ref as B, createElementBlock as d, openBlock as u, normalizeStyle as T, normalizeClass as z, renderSlot as o, createVNode as t, mergeProps as I, unref as r, withCtx as s, Transition as N, createCommentVNode as _, normalizeProps as A, guardReactiveProps as w, createTextVNode as L, nextTick as c } from "vue";
import { ScalarIconArrowRight as M, ScalarIconCaretLeft as P, ScalarIconListDashes as R } from "@scalar/icons";
import f from "./ScalarSidebarButton.vue.js";
import V from "./ScalarSidebarItems.vue.js";
/* empty css                        */
import h from "./ScalarSidebarSpacer.vue.js";
import { findScrollContainer as p } from "./findScrollContainer.js";
import { useSidebarGroups as q } from "./useSidebarGroups.js";
import { useSidebarNestedItem as E } from "./useSidebarNestedItems.js";
import O from "../ScalarIcon/ScalarIconLegacyAdapter.vue.js";
const j = {
  key: 0,
  class: "absolute inset-0 translate-x-full"
}, x = {}, Y = /* @__PURE__ */ y({
  ...x,
  inheritAttrs: !1,
  __name: "ScalarSidebarNestedItems",
  props: /* @__PURE__ */ C({
    is: {},
    icon: { type: [Object, Function], default: R },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: {},
    controlled: { type: Boolean }
  }, {
    open: { type: Boolean, default: !1 },
    openModifiers: {}
  }),
  emits: ["update:open"],
  setup(m) {
    const l = k(m, "open");
    E(l);
    const { level: v } = q({ reset: !0 }), a = g("el"), i = B(0);
    function b() {
      i.value = p(a.value).scrollTop ?? 0, c(
        () => a.value?.querySelector('button:not([aria-expanded="true"])')?.focus({ preventScroll: !0 })
      );
    }
    function $() {
      c(() => {
        a.value?.querySelector("button")?.focus({ preventScroll: !0 }), p(a.value).scrollTop = i.value;
      });
    }
    return (e, n) => (u(), d("li", {
      ref_key: "el",
      ref: a,
      class: z(["group/item group/nested-items contents", { "group/nested-items-open": l.value }]),
      style: T({ "--nested-items-offset": `${i.value}px` })
    }, [
      o(e.$slots, "button", {}, () => [
        t(f, I({
          is: "button",
          "aria-expanded": l.value,
          class: "text-c-1 font-sidebar-active",
          disabled: e.disabled,
          indent: r(v),
          selected: e.selected
        }, e.$attrs, {
          onClick: n[0] || (n[0] = (S) => !e.controlled && (l.value = !0))
        }), {
          icon: s(() => [
            o(e.$slots, "icon", {}, () => [
              t(r(O), {
                icon: e.icon,
                weight: "bold"
              }, null, 8, ["icon"])
            ])
          ]),
          aside: s(() => [
            o(e.$slots, "aside", {}, () => [
              t(r(M), { class: "size-4 text-c-2" })
            ])
          ]),
          default: s(() => [
            o(e.$slots, "default")
          ]),
          _: 3
        }, 16, ["aria-expanded", "disabled", "indent", "selected"])
      ]),
      t(N, {
        duration: 300,
        enterActiveClass: "top-(--nested-items-offset)",
        leaveActiveClass: "top-(--nested-items-offset)",
        onEnter: b,
        onLeave: $
      }, {
        default: s(() => [
          l.value ? (u(), d("div", j, [
            t(V, A(w(e.$attrs)), {
              default: s(() => [
                o(e.$slots, "back", {}, () => [
                  t(f, {
                    is: "button",
                    class: "text-c-1 font-sidebar-active",
                    onClick: n[1] || (n[1] = (S) => !e.controlled && (l.value = !1))
                  }, {
                    icon: s(() => [
                      t(r(P), { class: "size-4 -m-px text-c-2" })
                    ]),
                    default: s(() => [
                      o(e.$slots, "back-label", {}, () => [
                        n[2] || (n[2] = L("Back", -1))
                      ])
                    ]),
                    _: 3
                  })
                ]),
                t(h, { class: "h-3" }),
                o(e.$slots, "items")
              ]),
              _: 3
            }, 16)
          ])) : _("", !0)
        ]),
        _: 3
      })
    ], 6));
  }
});
export {
  Y as default
};
