// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarGroup.vue.js
import { defineComponent as c, mergeModels as f, useModel as v, createElementBlock as b, openBlock as r, renderSlot as t, createBlock as y, createCommentVNode as g, unref as l, createVNode as s, mergeProps as d, createSlots as B, withCtx as n, resolveDynamicComponent as $, normalizeProps as C } from "vue";
import { useBindCx as S } from "@scalar/use-hooks/useBindCx";
import k from "./ScalarSidebarButton.vue.js";
import _ from "./ScalarSidebarGroupToggle.vue.js";
import h from "./ScalarSidebarIndent.vue.js";
import { useSidebarGroups as x } from "./useSidebarGroups.js";
const A = { class: "group/item flex flex-col gap-px" }, M = {}, D = /* @__PURE__ */ c({
  ...M,
  inheritAttrs: !1,
  __name: "ScalarSidebarGroup",
  props: /* @__PURE__ */ f({
    is: { default: "ul" },
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: {},
    controlled: { type: Boolean }
  }, {
    open: { type: Boolean, default: !1 },
    openModifiers: {}
  }),
  emits: ["update:open"],
  setup(p) {
    const o = v(p, "open"), { level: a } = x({ increment: !0 }), { stylingAttrsCx: u, otherAttrs: m } = S();
    return (e, i) => (r(), b("li", A, [
      t(e.$slots, "button", {
        level: l(a),
        open: o.value
      }, () => [
        s(k, d({
          is: "button",
          active: e.active,
          "aria-expanded": o.value,
          class: "group/group-button",
          disabled: e.disabled,
          indent: l(a),
          selected: e.selected
        }, l(m), {
          onClick: i[0] || (i[0] = (G) => !e.controlled && (o.value = !o.value))
        }), B({
          indent: n(() => [
            s(h, {
              class: "mr-0",
              indent: l(a)
            }, null, 8, ["indent"])
          ]),
          icon: n(() => [
            t(e.$slots, "icon", { open: o.value }, () => [
              s(_, {
                class: "text-c-3",
                open: o.value
              }, null, 8, ["open"])
            ])
          ]),
          default: n(() => [
            t(e.$slots, "default", { open: o.value })
          ]),
          _: 2
        }, [
          e.$slots.aside ? {
            name: "aside",
            fn: n(() => [
              t(e.$slots, "aside", { open: o.value })
            ]),
            key: "0"
          } : void 0
        ]), 1040, ["active", "aria-expanded", "disabled", "indent", "selected"])
      ]),
      o.value ? (r(), y($(e.is), C(d({ key: 0 }, l(u)("group/items flex flex-col gap-px"))), {
        default: n(() => [
          t(e.$slots, "items", { open: o.value })
        ]),
        _: 3
      }, 16)) : g("", !0)
    ]));
  }
});
export {
  D as default
};
