// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarItem.vue.js
import { defineComponent as r, useSlots as s, createElementBlock as i, openBlock as a, createVNode as l, mergeProps as p, unref as c, createSlots as d, renderList as _, withCtx as m, renderSlot as u } from "vue";
import f from "./ScalarSidebarButton.vue.js";
import { useSidebarGroups as S } from "./useSidebarGroups.js";
const B = { class: "group/item contents" }, b = {}, C = /* @__PURE__ */ r({
  ...b,
  inheritAttrs: !1,
  __name: "ScalarSidebarItem",
  props: {
    is: {},
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: { default: () => {
    } }
  },
  setup(h) {
    const o = s(), { level: n } = S();
    return (e, y) => (a(), i("li", B, [
      l(f, p({ ...e.$attrs, ...e.$props }, {
        indent: e.indent ?? c(n)
      }), d({ _: 2 }, [
        _(o, ($, t) => ({
          name: t,
          fn: m(() => [
            u(e.$slots, t)
          ])
        }))
      ]), 1040, ["indent"])
    ]));
  }
});
export {
  C as default
};
