// @scalar/components/dist/components/ScalarMenu/ScalarMenuProduct.vue.js
import { defineComponent as n, createBlock as s, openBlock as a, unref as t, mergeProps as i, withCtx as c, renderSlot as l } from "vue";
import { useBindCx as p, cva as d } from "@scalar/use-hooks/useBindCx";
import u from "./ScalarMenuLink.vue.js";
const v = /* @__PURE__ */ n({
  inheritAttrs: !1,
  __name: "ScalarMenuProduct",
  props: {
    is: { default: "a" },
    selected: { type: Boolean },
    icon: { type: [Object, Function] }
  },
  setup(f) {
    const { cx: r } = p(), o = d({
      base: "gap-1.5",
      variants: {
        selected: {
          true: "pointer-events-none bg-b-2 dark:bg-b-3",
          false: "cursor-pointer hover:bg-b-2 dark:hover:bg-b-3"
        }
      }
    });
    return (e, b) => (a(), s(t(u), i({
      is: e.is,
      icon: e.icon,
      strong: "",
      target: "_blank"
    }, t(r)(t(o)({ selected: e.selected }))), {
      default: c(() => [
        l(e.$slots, "default")
      ]),
      _: 3
    }, 16, ["is", "icon"]));
  }
});
export {
  v as default
};
