// @scalar/api-reference/dist/features/Operation/components/Header.vue.js
import { defineComponent, createBlock, createCommentVNode, openBlock, unref } from "vue";
import { getResolvedRef } from "@scalar/workspace-store/helpers/get-resolved-ref";
import SchemaProperty from "../../../components/Content/Schema/SchemaProperty.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Header",
  props: {
    header: {},
    name: {},
    breadcrumb: {},
    eventBus: {},
    orderSchemaPropertiesBy: {},
    orderRequiredPropertiesFirst: { type: Boolean }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return "schema" in _ctx.header && _ctx.header.schema ? (openBlock(), createBlock(SchemaProperty, {
        key: 0,
        breadcrumb: _ctx.breadcrumb ? [..._ctx.breadcrumb, "headers"] : void 0,
        description: _ctx.header.description,
        eventBus: _ctx.eventBus,
        name: _ctx.name,
        options: {
          orderRequiredPropertiesFirst: _ctx.orderRequiredPropertiesFirst,
          orderSchemaPropertiesBy: _ctx.orderSchemaPropertiesBy
        },
        schema: unref(getResolvedRef)(_ctx.header.schema)
      }, null, 8, ["breadcrumb", "description", "eventBus", "name", "options", "schema"])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
