// @scalar/api-client/dist/v2/features/operation/Operation.vue.js
import { defineComponent as d, computed as u, createBlock as v, createElementBlock as y, openBlock as s, unref as r, createElementVNode as h } from "vue";
import { getResolvedRef as f } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { createStoreEvents as p } from "../../../store/events.js";
import S from "../../blocks/operation-block/OperationBlock.vue.js";
const B = {
  key: 1,
  class: "flex h-full w-full items-center justify-center"
}, N = {}, R = /* @__PURE__ */ d({
  ...N,
  __name: "Operation",
  props: {
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    workspaceStore: {}
  },
  setup(e) {
    const n = u(
      () => e.path && e.method ? f(e.document?.paths?.[e.path]?.[e.method]) : void 0
    ), a = u(
      () => (n.value?.security?.length ?? 0) > 0 && JSON.stringify(n.value?.security) !== "[{}]"
    ), i = u(() => !n.value || !e.document ? [] : a.value ? n.value.security ?? [] : JSON.stringify(n.value?.security) === "[{}]" ? [...e.document.security ?? [], {}] : e.document.security ?? []), l = u(() => a.value ? n.value?.["x-scalar-selected-security"] : e.document?.["x-scalar-selected-security"]), c = u(() => a.value ? {
      type: "operation",
      path: e.path ?? "",
      method: e.method ?? "get"
    } : {
      type: "document"
    }), m = "2.11.0";
    return (t, o) => t.path && t.method && t.exampleName && n.value ? (s(), v(r(S), {
      key: 0,
      appVersion: r(m),
      authMeta: c.value,
      environment: t.environment,
      eventBus: t.eventBus,
      events: r(p)(),
      exampleKey: t.exampleName,
      history: [],
      layout: t.layout,
      method: t.method,
      operation: n.value,
      path: t.path,
      plugins: [],
      security: i.value,
      securitySchemes: t.document?.components?.securitySchemes ?? {},
      selectedSecurity: l.value,
      server: void 0,
      servers: [],
      totalPerformedRequests: 0
    }, null, 8, ["appVersion", "authMeta", "environment", "eventBus", "events", "exampleKey", "layout", "method", "operation", "path", "security", "securitySchemes", "selectedSecurity"])) : (s(), y("div", B, [...o[0] || (o[0] = [
      h("span", { class: "text-c-3" }, "Select an operation to view details", -1)
    ])]));
  }
});
export {
  R as default
};
