// @scalar/components/dist/components/ScalarTextInput/ScalarTextInput.vue.js
import { defineComponent as y, mergeModels as v, useModel as x, ref as _, onMounted as $, createBlock as k, openBlock as t, unref as r, mergeProps as c, withCtx as w, createElementVNode as i, renderSlot as o, createElementBlock as n, createCommentVNode as l, withDirectives as C, vModelDynamic as g, toDisplayString as V } from "vue";
import { useBindCx as b } from "@scalar/use-hooks/useBindCx";
import B from "../ScalarForm/ScalarFormInput.vue.js";
const M = { class: "flex flex-1 relative" }, A = {
  key: 0,
  class: "select-none whitespace-nowrap text-sm text-transparent"
}, D = ["readonly", "aria-readonly"], S = {
  key: 1,
  class: "absolute inset-0 select-none overflow-hidden whitespace-nowrap text-sm"
}, E = {
  key: 0,
  class: "text-c-2"
}, N = { class: "text-transparent" }, z = {
  key: 1,
  class: "text-c-2"
}, I = {}, j = /* @__PURE__ */ y({
  ...I,
  inheritAttrs: !1,
  __name: "ScalarTextInput",
  props: /* @__PURE__ */ v({
    readonly: { type: Boolean }
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(d) {
    const a = x(d, "modelValue"), s = _(), { stylingAttrsCx: p, otherAttrs: u } = b();
    $(() => {
      "autofocus" in u.value && s.value?.focus();
    });
    function m() {
      d.readonly ? s.value?.select() : s.value?.focus();
    }
    return (e, f) => (t(), k(r(B), c(
      { is: "div" },
      r(p)(
        "cursor-text text-c-1",
        e.readonly ? "" : "focus-within:bg-b-1"
      ),
      { onClick: m }
    ), {
      default: w(() => [
        i("div", M, [
          e.$slots.prefix ? (t(), n("span", A, [
            o(e.$slots, "prefix")
          ])) : l("", !0),
          C(i("input", c({
            ref_key: "input",
            ref: s,
            "onUpdate:modelValue": f[0] || (f[0] = (h) => a.value = h),
            readonly: e.readonly,
            "aria-readonly": e.readonly || void 0,
            class: "z-1 min-w-0 flex-1 border-none bg-transparent text-sm placeholder:font-[inherit] focus-within:outline-none"
          }, r(u)), null, 16, D), [
            [g, a.value]
          ]),
          e.$slots.prefix || e.$slots.suffix ? (t(), n("div", S, [
            e.$slots.prefix ? (t(), n("span", E, [
              o(e.$slots, "prefix")
            ])) : l("", !0),
            i("span", N, V(a.value || e.$attrs.placeholder), 1),
            e.$slots.suffix ? (t(), n("span", z, [
              o(e.$slots, "suffix")
            ])) : l("", !0)
          ])) : l("", !0)
        ]),
        o(e.$slots, "aside")
      ]),
      _: 3
    }, 16));
  }
});
export {
  j as default
};
