// @scalar/sidebar/dist/helpers/create-sidebar-state.js
import { computed as c, toValue as u, ref as k } from "vue";
import { generateReverseIndex as m } from "./generate-reverse-index.js";
const E = (o, t) => {
  const n = c(
    () => m({
      items: u(o),
      nestedKey: t?.key ?? "children"
    })
  ), s = k({}), l = k({}), p = (e) => {
    const f = (r) => {
      r && (s.value[r.id] = !0, "parent" in r && r.parent && f(r.parent));
    };
    t?.hooks?.onBeforeSelect && t.hooks.onBeforeSelect(e), s.value = {}, e !== null && f(n.value.get(e)), t?.hooks?.onAfterSelect && t.hooks.onAfterSelect(e);
  }, v = (e, f) => {
    const r = (a) => {
      a && (l.value[a.id] = !0, "parent" in a && a.parent && r(a.parent));
    };
    t?.hooks?.onBeforeExpand && t.hooks.onBeforeExpand(e), f === !1 ? l.value[e] = !1 : r(n.value.get(e)), t?.hooks?.onAfterExpand && t.hooks.onAfterExpand(e);
  }, x = (e) => l.value[e] ?? !1, d = (e) => s.value[e] ?? !1, h = (e) => n.value.get(e);
  return {
    items: c(() => u(o)),
    index: n,
    selectedItems: s,
    expandedItems: l,
    setSelected: p,
    setExpanded: v,
    isExpanded: x,
    isSelected: d,
    getEntryById: h
  };
};
export {
  E as createSidebarState
};
