// @scalar/api-reference/dist/blocks/scalar-server-selector-block/components/ServerSelector.vue.js
import { defineComponent, useId, computed, watch, nextTick, createElementBlock, openBlock, Fragment, createElementVNode, createVNode, createBlock, createCommentVNode, normalizeClass, unref } from "vue";
import { ServerVariablesForm } from "@scalar/api-client/components/Server";
import { ScalarMarkdown } from "@scalar/components";
import { emitCustomEvent } from "@scalar/workspace-store/events";
import { templateRef } from "@vueuse/core";
import _sfc_main$1 from "./Selector.vue.js";
const _hoisted_1 = ["id"];
const __default__ = {};
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...__default__,
  __name: "ServerSelector",
  props: {
    xSelectedServer: {},
    servers: {}
  },
  setup(__props) {
    const id = useId();
    const updateServer = (newServer) => {
      emitCustomEvent(containerRef.value, "scalar-update-selected-server", {
        value: newServer
      });
    };
    const updateServerVariable = (key, value) => {
      if (!containerRef.value) {
        return;
      }
      emitCustomEvent(
        containerRef.value,
        "scalar-update-selected-server-variables",
        { key, value }
      );
    };
    const server = computed(() => {
      return __props.servers.find((s) => s.url === __props.xSelectedServer);
    });
    watch(
      () => __props.servers,
      (newServers) => {
        if (!newServers.length) {
          return;
        }
        if (__props.xSelectedServer) {
          return;
        }
        nextTick(() => {
          if (newServers[0]) {
            updateServer(newServers[0].url);
          }
        });
      },
      { immediate: true }
    );
    const containerRef = templateRef("containerRef");
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(Fragment, null, [
        _cache[0] || (_cache[0] = createElementVNode("label", { class: "bg-b-2 flex h-8 items-center rounded-t-lg border border-b-0 px-3 py-2.5 font-medium" }, " Server ", -1)),
        createElementVNode("div", {
          id: unref(id),
          ref_key: "containerRef",
          ref: containerRef,
          class: normalizeClass(["border", {
            "rounded-b-lg": !server.value?.description && !server.value?.variables
          }])
        }, [
          _ctx.servers.length ? (openBlock(), createBlock(_sfc_main$1, {
            key: 0,
            servers: _ctx.servers,
            target: unref(id),
            xSelectedServer: _ctx.xSelectedServer,
            "onUpdate:modelValue": updateServer
          }, null, 8, ["servers", "target", "xSelectedServer"])) : createCommentVNode("", true)
        ], 10, _hoisted_1),
        createVNode(unref(ServerVariablesForm), {
          layout: "reference",
          variables: server.value?.variables,
          "onUpdate:variable": updateServerVariable
        }, null, 8, ["variables"]),
        server.value?.description ? (openBlock(), createBlock(unref(ScalarMarkdown), {
          key: 0,
          class: "text-c-3 rounded-b-lg border border-t-0 px-3 py-1.5",
          value: server.value.description
        }, null, 8, ["value"])) : createCommentVNode("", true)
      ], 64);
    };
  }
});
export {
  _sfc_main as default
};
