// @scalar/api-client/dist/v2/features/app/components/AppSidebar.vue2.js
import { defineComponent as I, mergeModels as k, useModel as m, computed as v, capitalize as B, createBlock as S, openBlock as u, unref as a, createSlots as g, withCtx as i, createVNode as d, createTextVNode as b, toDisplayString as x, createElementVNode as o, normalizeClass as M, createElementBlock as A, createCommentVNode as w } from "vue";
import { ScalarSidebarItem as P, ScalarButton as y } from "@scalar/components";
import { ScalarIconGlobe as q } from "@scalar/icons";
import z from "../../../../assets/rabbit.ascii.js";
import N from "../../../../assets/rabbitjump.ascii.js";
import O from "../../../../components/ScalarAsciiArt.vue.js";
import V from "../../../components/sidebar/Sidebar.vue.js";
const $ = {
  key: 0,
  class: "empty-sidebar-item-content overflow-hidden px-2.5 py-2.5"
}, G = { class: "rabbit-ascii relative m-auto mt-2 h-[68px] w-[60px]" }, U = { class: "gap-1.5 p-2" }, T = /* @__PURE__ */ I({
  __name: "AppSidebar",
  props: /* @__PURE__ */ k({
    layout: {},
    sidebarState: {},
    isWorkspaceOpen: { type: Boolean }
  }, {
    workspace: {
      required: !0,
      default: "default"
    },
    workspaceModifiers: {},
    isSidebarOpen: { type: Boolean, required: !0 },
    isSidebarOpenModifiers: {},
    sidebarWidth: {
      required: !0,
      default: 288
    },
    sidebarWidthModifiers: {}
  }),
  emits: /* @__PURE__ */ k(["open:commandPalette", "click:workspace", "selectItem"], ["update:workspace", "update:isSidebarOpen", "update:sidebarWidth"]),
  setup(r, { emit: W }) {
    const s = W, n = m(r, "workspace"), C = v(
      () => B(n.value) + " Workspace"
    ), c = m(r, "isSidebarOpen"), f = m(r, "sidebarWidth"), p = v(() => r.sidebarState.items.value.length <= 1);
    return (l, e) => (u(), S(a(V), {
      isSidebarOpen: c.value,
      "onUpdate:isSidebarOpen": e[3] || (e[3] = (t) => c.value = t),
      sidebarWidth: f.value,
      "onUpdate:sidebarWidth": e[4] || (e[4] = (t) => f.value = t),
      workspace: n.value,
      "onUpdate:workspace": e[5] || (e[5] = (t) => n.value = t),
      sidebarState: l.sidebarState,
      layout: l.layout,
      onSelectItem: e[6] || (e[6] = (t) => s("selectItem", t))
    }, g({
      workspaceButton: i(() => [
        d(a(P), {
          is: "button",
          icon: a(q),
          active: l.isWorkspaceOpen,
          onClick: e[0] || (e[0] = (t) => s("click:workspace"))
        }, {
          default: i(() => [
            b(x(C.value), 1)
          ]),
          _: 1
        }, 8, ["icon", "active"])
      ]),
      _: 2
    }, [
      l.layout !== "modal" ? {
        name: "footer",
        fn: i(() => [
          o("div", {
            class: M({
              "empty-sidebar-item border-t": p.value
            })
          }, [
            p.value ? (u(), A("div", $, [
              o("div", G, [
                d(O, {
                  art: a(z),
                  class: "rabbitsit font-bold"
                }, null, 8, ["art"]),
                d(O, {
                  art: a(N),
                  class: "rabbitjump absolute top-0 left-0 font-bold"
                }, null, 8, ["art"])
              ]),
              e[7] || (e[7] = o("div", { class: "mt-2 mb-2 text-center text-sm text-balance" }, [
                o("b", { class: "font-medium" }, "Let's Get Started"),
                o("p", { class: "mt-2 leading-3" }, " Create request, folder, collection or import from OpenAPI/Postman ")
              ], -1))
            ])) : w("", !0),
            o("div", U, [
              p.value ? (u(), S(a(y), {
                key: 0,
                class: "w-full",
                size: "sm",
                onClick: e[1] || (e[1] = (t) => s("open:commandPalette", "import"))
              }, {
                default: i(() => [...e[8] || (e[8] = [
                  b(" Import Collection ", -1)
                ])]),
                _: 1
              })) : w("", !0),
              d(a(y), {
                class: "w-full",
                hotkey: "K",
                size: "sm",
                variant: "outlined",
                onClick: e[2] || (e[2] = (t) => s("open:commandPalette"))
              }, {
                default: i(() => [...e[9] || (e[9] = [
                  b(" Add Item ", -1)
                ])]),
                _: 1
              })
            ])
          ], 2)
        ]),
        key: "0"
      } : void 0
    ]), 1032, ["isSidebarOpen", "sidebarWidth", "workspace", "sidebarState", "layout"]));
  }
});
export {
  T as default
};
