// @scalar/api-reference/dist/helpers/map-config-to-client-store.js
import { createApiClientModal } from "@scalar/api-client/layouts/Modal";
import { createWorkspaceStore, createActiveEntitiesStore, WORKSPACE_SYMBOL, ACTIVE_ENTITIES_SYMBOL } from "@scalar/api-client/store";
import { mutateSecuritySchemeDiff } from "@scalar/api-client/views/Request/libs";
import { filterSecurityRequirements } from "@scalar/api-client/views/Request/RequestSection";
import { getServersFromDocument } from "@scalar/oas-utils/helpers";
import { emitCustomEvent } from "@scalar/workspace-store/events";
import { watchDebounced } from "@vueuse/core";
import microdiff from "microdiff";
import { toValue, provide, watch, toRaw, computed } from "vue";
import { convertSecurityScheme } from "./convert-security-scheme.js";
import { useLegacyStoreEvents } from "../hooks/use-legacy-store-events.js";
function mapConfigToClientStore({
  config,
  workspaceStore,
  el,
  root,
  dereferencedDocument,
  documentUrl
}) {
  const store = createWorkspaceStore({
    useLocalStorage: false,
    proxyUrl: toValue(config).proxyUrl,
    theme: toValue(config).theme,
    showSidebar: toValue(config).showSidebar,
    hideClientButton: toValue(config).hideClientButton,
    _integration: toValue(config)._integration
  });
  let client = null;
  const activeEntities = createActiveEntitiesStore(store);
  provide(WORKSPACE_SYMBOL, store);
  provide(ACTIVE_ENTITIES_SYMBOL, activeEntities);
  useLegacyStoreEvents(workspaceStore, store, activeEntities, root);
  watch(
    el,
    () => {
      console.info(`[CLIENT]: Client element changed. ${el.value ? "Mounting client..." : "Unmounting client..."}`);
      if (!el.value) {
        client?.app?.unmount();
        return;
      }
      const clientConfig = toValue(config);
      const mount = createApiClientModal({
        el: el.value,
        configuration: {
          ...clientConfig,
          plugins: typeof clientConfig.onBeforeRequest === "function" ? [
            () => ({
              name: "on-before-request",
              hooks: {
                onBeforeRequest: clientConfig.onBeforeRequest
              }
            })
          ] : []
        },
        store
      });
      client = mount.client;
    },
    {
      immediate: true
    }
  );
  watchDebounced(
    () => toValue(config),
    (newConfig, oldConfig) => {
      if (!oldConfig || !activeEntities.activeCollection.value) {
        return;
      }
      microdiff(oldConfig, newConfig).forEach((d) => {
        if (d.path[0] === "authentication") {
          mutateSecuritySchemeDiff(d, activeEntities, store);
        }
      });
    },
    { deep: true, debounce: 300 }
  );
  watch(
    [() => toValue(config).servers, () => toValue(dereferencedDocument)?.servers, () => toValue(config).baseServerURL],
    ([newServers], [oldServers]) => {
      if (newServers || oldServers) {
        const servers = getServersFromDocument(newServers ?? toValue(dereferencedDocument)?.servers, {
          baseServerURL: toValue(config).baseServerURL
        });
        emitCustomEvent(el.value, "scalar-replace-servers", {
          servers,
          options: {
            disableOldStoreUpdate: true
          }
        });
      }
    }
  );
  watchDebounced(
    () => toValue(dereferencedDocument),
    (newDocument, oldDocument) => {
      if (!newDocument) {
        return;
      }
      const diff = microdiff(newDocument, oldDocument || {});
      if (!diff?.length) {
        return;
      }
      if (activeEntities.activeCollection.value) {
        client?.resetStore();
      }
      return store.importSpecFile(void 0, "default", {
        dereferencedDocument: toRaw(newDocument),
        shouldLoad: true,
        documentUrl: toValue(documentUrl),
        useCollectionSecurity: true,
        ...toValue(config)
      });
    }
  );
  const { activeCollection, activeEnvVariables, activeEnvironment, activeWorkspace } = activeEntities;
  const activeServer = computed(() => {
    if (!activeCollection.value) {
      return void 0;
    }
    if (activeCollection.value.selectedServerUid) {
      const server = store.servers[activeCollection.value.selectedServerUid];
      if (server) {
        return server;
      }
    }
    return store.servers[activeCollection.value.servers[0] ?? ""];
  });
  const getSecuritySchemes = (operationSecurity, documentSecurity) => {
    return filterSecurityRequirements(
      operationSecurity || documentSecurity || [],
      activeCollection.value?.selectedSecuritySchemeUids || [],
      store.securitySchemes
    ).map(convertSecurityScheme);
  };
  return {
    activeServer,
    activeEnvVariables,
    activeEnvironment,
    activeWorkspace,
    getSecuritySchemes,
    openClient: (payload) => client?.open(payload)
  };
}
export {
  mapConfigToClientStore
};
