// @scalar/api-reference/dist/components/Section/CompactSection.vue2.js
import { defineComponent, createElementBlock, openBlock, createElementVNode, createBlock, createCommentVNode, normalizeClass, createVNode, unref, withCtx, renderSlot } from "vue";
import { ScalarIconCaretRight } from "@scalar/icons";
import Section from "./Section.vue.js";
import _sfc_main$1 from "../Anchor/Anchor.vue.js";
const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["id", "aria-controls", "aria-expanded"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "CompactSection",
  props: {
    id: {},
    label: {},
    modelValue: { type: Boolean }
  },
  emits: ["update:modelValue", "copyAnchorUrl"],
  setup(__props, { emit: __emit }) {
    const emit = __emit;
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("section", {
        "aria-label": _ctx.label,
        class: "collapsible-section"
      }, [
        createElementVNode("button", {
          id: _ctx.id,
          "aria-controls": _ctx.id,
          "aria-expanded": _ctx.modelValue,
          class: normalizeClass(["collapsible-section-trigger", { "collapsible-section-trigger-open": _ctx.modelValue }]),
          type: "button",
          onClick: _cache[1] || (_cache[1] = ($event) => emit("update:modelValue", !_ctx.modelValue))
        }, [
          createVNode(unref(ScalarIconCaretRight), {
            class: normalizeClass(["size-3 transition-transform duration-100", { "rotate-90": _ctx.modelValue }]),
            weight: "bold"
          }, null, 8, ["class"]),
          createVNode(unref(_sfc_main$1), {
            class: "collapsible-section-header",
            onCopyAnchorUrl: _cache[0] || (_cache[0] = () => emit("copyAnchorUrl"))
          }, {
            default: withCtx(() => [
              renderSlot(_ctx.$slots, "heading", {}, void 0, true)
            ]),
            _: 3
          })
        ], 10, _hoisted_2),
        _ctx.modelValue ? (openBlock(), createBlock(Section, {
          key: 0,
          class: "collapsible-section-content",
          label: _ctx.label
        }, {
          default: withCtx(() => [
            renderSlot(_ctx.$slots, "default", {}, void 0, true)
          ]),
          _: 3
        }, 8, ["label"])) : createCommentVNode("", true)
      ], 8, _hoisted_1);
    };
  }
});
export {
  _sfc_main as default
};
