// @scalar/api-reference/dist/features/test-request-button/TestRequestButton.vue2.js
import { defineComponent, ref, createElementBlock, openBlock, withModifiers, createVNode, createElementVNode, unref, withCtx, createTextVNode, toDisplayString } from "vue";
import { ScalarIconPlay } from "@scalar/icons";
import { emitCustomEvent } from "@scalar/workspace-store/events";
import ScreenReader from "../../components/ScreenReader.vue.js";
const _hoisted_1 = ["method"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "TestRequestButton",
  props: {
    method: {},
    path: {}
  },
  setup(__props) {
    const el = ref(null);
    const handleClick = () => {
      emitCustomEvent(el.value, "scalar-open-client", { method: __props.method, path: __props.path });
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("button", {
        ref_key: "el",
        ref: el,
        class: "show-api-client-button",
        method: _ctx.method,
        type: "button",
        onClick: withModifiers(handleClick, ["stop"])
      }, [
        createVNode(unref(ScalarIconPlay), {
          class: "size-3",
          weight: "fill"
        }),
        _cache[0] || (_cache[0] = createElementVNode("span", null, "Test Request", -1)),
        createVNode(ScreenReader, null, {
          default: withCtx(() => [
            createTextVNode("(" + toDisplayString(_ctx.method) + " " + toDisplayString(_ctx.path) + ")", 1)
          ]),
          _: 1
        })
      ], 8, _hoisted_1);
    };
  }
});
export {
  _sfc_main as default
};
