// @scalar/sidebar/dist/components/ScalarSidebar.vue.js
import { defineComponent as u, createBlock as d, openBlock as s, unref as i, withCtx as l, renderSlot as o, createVNode as f, createElementVNode as c, createElementBlock as S, Fragment as E, renderList as y, mergeProps as $ } from "vue";
import { ScalarSidebar as b, ScalarSidebarItems as g } from "@scalar/components";
import k from "./SidebarItem.vue.js";
const D = /* @__PURE__ */ u({
  __name: "ScalarSidebar",
  props: {
    layout: {},
    items: {},
    isSelected: { type: Function },
    isExpanded: { type: Function },
    options: {}
  },
  emits: ["reorder", "selectItem"],
  setup(I, { emit: m }) {
    const n = m, p = (e, t) => {
      n("reorder", e, t);
    };
    return (e, t) => (s(), d(i(b), { class: "flex min-h-0 flex-col" }, {
      default: l(() => [
        o(e.$slots, "header"),
        o(e.$slots, "default", {}, () => [
          f(i(g), { class: "custom-scroll pt-0" }, {
            default: l(() => [
              o(e.$slots, "before"),
              (s(!0), S(E, null, y(e.items, (a) => (s(), d(k, {
                key: a.id,
                isExpanded: e.isExpanded,
                isSelected: e.isSelected,
                item: a,
                layout: e.layout,
                options: e.options,
                onOnDragEnd: p,
                onSelectItem: t[0] || (t[0] = (r) => n("selectItem", r))
              }, {
                decorator: l((r) => [
                  o(e.$slots, "decorator", $({ ref_for: !0 }, r))
                ]),
                _: 3
              }, 8, ["isExpanded", "isSelected", "item", "layout", "options"]))), 128))
            ]),
            _: 3
          }),
          t[1] || (t[1] = c("div", { class: "flex-1" }, null, -1))
        ]),
        o(e.$slots, "footer")
      ]),
      _: 3
    }));
  }
});
export {
  D as default
};
