// @scalar/sidebar/dist/components/SidebarHttpBadge.vue2.js
import { defineComponent as l, createBlock as t, openBlock as r, normalizeClass as s, withCtx as n, createElementVNode as d, renderSlot as m, createCommentVNode as p, unref as a, normalizeStyle as i } from "vue";
import { getHttpMethodInfo as h } from "@scalar/helpers/http/http-info";
import { ScalarIconWebhooksLogo as c } from "@scalar/icons";
import f from "./HttpMethod.vue.js";
const w = /* @__PURE__ */ l({
  __name: "SidebarHttpBadge",
  props: {
    method: {},
    active: { type: Boolean },
    webhook: { type: Boolean }
  },
  setup(y) {
    return (e, o) => (r(), t(f, {
      class: s([
        "sidebar-heading-type",
        `sidebar-heading-type--${e.method.toLowerCase()}`,
        { "sidebar-heading-type-active": e.active }
      ]),
      method: e.method,
      property: "--method-color",
      short: ""
    }, {
      default: n(() => [
        o[0] || (o[0] = d("span", { class: "sr-only" }, "HTTP Method: ", -1)),
        m(e.$slots, "default", {}, () => [
          e.webhook ? (r(), t(a(c), {
            key: 0,
            style: i({
              color: a(h)(e.method).colorVar
            }),
            weight: "bold"
          }, null, 8, ["style"])) : p("", !0)
        ], !0)
      ]),
      _: 3
    }, 8, ["class", "method"]));
  }
});
export {
  w as default
};
