// @scalar/sidebar/dist/components/SidebarItem.vue.js
import { defineComponent as B, resolveComponent as G, createBlock as n, openBlock as o, unref as r, withCtx as i, createTextVNode as s, toDisplayString as m, createElementBlock as p, Fragment as u, renderList as E, renderSlot as y, mergeProps as v, createSlots as I, createVNode as c, createCommentVNode as f } from "vue";
import { ScalarSidebarSection as F, ScalarSidebarGroup as N, ScalarSidebarGroupToggle as O, ScalarSidebarItem as T } from "@scalar/components";
import { Draggable as V } from "@scalar/draggable";
import { addWordBreaks as k } from "@scalar/helpers/string/add-word-breaks";
import { ScalarIconFolder as A } from "@scalar/icons";
import D from "./SidebarHttpBadge.vue.js";
const z = /* @__PURE__ */ B({
  __name: "SidebarItem",
  props: {
    item: {},
    layout: {},
    isSelected: { type: Function },
    isExpanded: { type: Function },
    options: {}
  },
  emits: ["selectItem", "onDragEnd"],
  setup(w, { emit: $ }) {
    const a = $, h = (e) => "children" in e && Array.isArray(e.children) && e.children.length > 0, C = (e) => "isGroup" in e && e.isGroup, g = (e) => w.layout === "reference" ? e : e.filter(
      (t) => t.type === "operation" || t.type === "example" || t.type === "tag"
    ), S = (e, t) => {
      a("onDragEnd", e, t);
    };
    return (e, t) => {
      const b = G("SidebarItem", !0);
      return o(), n(r(V), {
        id: e.item.id,
        class: "flex flex-1 flex-col",
        isDraggable: e.layout === "client",
        parentIds: [],
        onOnDragEnd: S
      }, {
        default: i(() => [
          h(e.item) && C(e.item) ? (o(), n(r(F), { key: 0 }, {
            items: i(() => [
              (o(!0), p(u, null, E(g(e.item.children), (d) => (o(), n(b, {
                key: d.id,
                isExpanded: e.isExpanded,
                isSelected: e.isSelected,
                item: d,
                layout: e.layout,
                options: e.options,
                onOnDragEnd: S,
                onSelectItem: t[0] || (t[0] = (l) => a("selectItem", l))
              }, {
                decorator: i((l) => [
                  y(e.$slots, "decorator", v({ ref_for: !0 }, l))
                ]),
                _: 3
              }, 8, ["isExpanded", "isSelected", "item", "layout", "options"]))), 128))
            ]),
            default: i(() => [
              s(m(e.item.title) + " ", 1)
            ]),
            _: 3
          })) : h(e.item) && (e.layout === "reference" && !(e.item.type === "operation" || e.item.type === "webhook") || e.layout === "client") ? (o(), n(r(N), {
            key: 1,
            active: e.isSelected(e.item.id),
            controlled: "",
            open: e.isExpanded(e.item.id),
            onClick: t[2] || (t[2] = () => a("selectItem", e.item.id))
          }, I({
            items: i(() => [
              (o(!0), p(u, null, E(g(e.item.children), (d) => (o(), n(b, {
                key: d.id,
                isExpanded: e.isExpanded,
                isSelected: e.isSelected,
                item: d,
                layout: e.layout,
                options: e.options,
                parentIds: [],
                onOnDragEnd: S,
                onSelectItem: t[1] || (t[1] = (l) => a("selectItem", l))
              }, {
                decorator: i((l) => [
                  y(e.$slots, "decorator", v({ ref_for: !0 }, l))
                ]),
                _: 3
              }, 8, ["isExpanded", "isSelected", "item", "layout", "options"]))), 128))
            ]),
            default: i(() => [
              s(" " + m(e.item.title) + " ", 1)
            ]),
            _: 2
          }, [
            e.item.type === "document" ? {
              name: "icon",
              fn: i(({ open: d }) => [
                c(r(A), { class: "text-c-3 block group-hover/group-button:hidden" }),
                c(r(O), {
                  class: "text-c-3 hidden group-hover/group-button:block",
                  open: d
                }, null, 8, ["open"])
              ]),
              key: "0"
            } : void 0,
            "method" in e.item || e.$slots.decorator ? {
              name: "aside",
              fn: i(() => [
                e.$slots.decorator ? y(e.$slots, "decorator", {
                  key: 0,
                  item: e.item
                }) : f("", !0),
                "method" in e.item ? (o(), n(D, {
                  key: 1,
                  active: e.isSelected(e.item.id),
                  class: "ml-2 h-4 self-start",
                  method: e.item.method,
                  webhook: e.item.type === "webhook"
                }, null, 8, ["active", "method", "webhook"])) : f("", !0)
              ]),
              key: "1"
            } : void 0
          ]), 1032, ["active", "open"])) : (o(), n(r(T), {
            key: 2,
            is: "button",
            class: "text-left",
            selected: e.isSelected(e.item.id),
            onClick: t[3] || (t[3] = () => a("selectItem", e.item.id))
          }, I({
            default: i(() => [
              e.item.type === "model" ? (o(), p(u, { key: 0 }, [
                s(m(r(k)(e.item.title, { preset: "property" })), 1)
              ], 64)) : (o(), p(u, { key: 1 }, [
                s(m(e.options?.operationTitleSource === "path" && "path" in e.item ? r(k)(e.item.path) : r(k)(e.item.title)), 1)
              ], 64))
            ]),
            _: 2
          }, [
            "method" in e.item || e.$slots.decorator ? {
              name: "aside",
              fn: i(() => [
                e.$slots.decorator ? y(e.$slots, "decorator", {
                  key: 0,
                  item: e.item
                }) : f("", !0),
                "method" in e.item ? (o(), n(D, {
                  key: 1,
                  active: e.isSelected(e.item.id),
                  class: "ml-2 h-4 self-start",
                  method: e.item.method,
                  webhook: e.item.type === "webhook"
                }, null, 8, ["active", "method", "webhook"])) : f("", !0)
              ]),
              key: "0"
            } : void 0
          ]), 1032, ["selected"]))
        ]),
        _: 3
      }, 8, ["id", "isDraggable"]);
    };
  }
});
export {
  z as default
};
