// @scalar/components/dist/components/ScalarCheckboxInput/ScalarCheckboxInput.vue.js
import { defineComponent as u, mergeModels as m, useModel as p, createBlock as c, openBlock as f, unref as o, mergeProps as s, withCtx as _, createVNode as y, createElementVNode as r, withDirectives as h, renderSlot as v, vModelDynamic as x } from "vue";
import { useBindCx as V } from "@scalar/use-hooks/useBindCx";
import b from "./ScalarCheckbox.vue.js";
import g from "../ScalarForm/ScalarFormInput.vue.js";
const k = { class: "flex-1 text-left min-w-0 truncate" }, C = ["type"], B = {}, D = /* @__PURE__ */ u({
  ...B,
  inheritAttrs: !1,
  __name: "ScalarCheckboxInput",
  props: /* @__PURE__ */ m({
    type: { default: "checkbox" }
  }, {
    modelValue: { type: Boolean },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(n) {
    const e = p(n, "modelValue"), { stylingAttrsCx: a, otherAttrs: i } = V();
    return (t, l) => (f(), c(o(g), s(
      { is: "label" },
      o(a)("cursor-pointer gap-2 hover:bg-b-2", { "text-c-1": e.value })
    ), {
      default: _(() => [
        y(b, {
          selected: e.value,
          type: t.type,
          class: "shrink-0"
        }, null, 8, ["selected", "type"]),
        r("div", k, [
          v(t.$slots, "default")
        ]),
        h(r("input", s({
          type: t.type,
          "onUpdate:modelValue": l[0] || (l[0] = (d) => e.value = d),
          class: "sr-only"
        }, o(i)), null, 16, C), [
          [x, e.value]
        ])
      ]),
      _: 3
    }, 16));
  }
});
export {
  D as default
};
