// @scalar/api-client/dist/v2/features/settings/Settings.vue.js
import { defineComponent as j, createElementBlock as g, openBlock as a, createElementVNode as t, createVNode as i, withCtx as l, createBlock as u, createCommentVNode as p, unref as r, normalizeClass as d, createTextVNode as n, toDisplayString as v, Fragment as C, renderList as U, normalizeStyle as h } from "vue";
import { cva as M, ScalarButton as m, cx as y, ScalarIcon as b } from "@scalar/components";
import { themeLabels as T } from "@scalar/themes";
import B from "../../../components/ImportCollection/IntegrationLogo.vue.js";
import L from "./components/Appearance.vue.js";
import x from "./components/Section.vue.js";
import { getThemeColors as w } from "./helpers/get-theme-colors.js";
const N = { class: "bg-b-1 h-full w-full overflow-auto" }, R = { class: "mr-auto ml-auto flex w-full max-w-[720px] flex-col gap-8 px-5 py-5" }, V = { class: "flex flex-col gap-2" }, A = { class: "bg-c-accent text-b-1 flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] border-transparent p-1" }, E = { class: "flex flex-col gap-2" }, F = { class: "grid grid-cols-2 gap-2" }, O = { class: "flex items-center gap-2" }, D = { class: "flex items-center gap-1" }, W = { class: "grid grid-cols-2 gap-2" }, _ = { class: "flex items-center gap-2" }, q = { class: "flex items-center gap-1" }, G = { class: "size-7 rounded-xl" }, k = "https://proxy.scalar.com", ee = /* @__PURE__ */ j({
  __name: "Settings",
  props: {
    proxyUrl: {},
    customProxyUrl: {},
    activeThemeId: {},
    colorMode: {}
  },
  emits: ["update:proxyUrl", "update:themeId", "update:colorMode"],
  setup(H, { emit: S }) {
    const c = S, z = [
      "default",
      "alternate",
      "purple",
      "solarized",
      "saturn",
      "kepler"
    ], P = ["elysiajs", "fastify"], f = M({
      base: "w-full shadow-none text-c-1 justify-start pl-2 gap-2 border",
      variants: {
        active: {
          true: "bg-primary text-c-1 hover:bg-inherit",
          false: "bg-b-1 hover:bg-b-2"
        }
      }
    });
    return (s, e) => (a(), g("div", N, [
      t("div", R, [
        e[14] || (e[14] = t("div", null, [
          t("h2", { class: "mt-10 text-xl font-bold" }, "Settings")
        ], -1)),
        i(x, null, {
          title: l(() => [...e[4] || (e[4] = [
            n(" CORS Proxy ", -1)
          ])]),
          description: l(() => [...e[5] || (e[5] = [
            n(" Browsers block cross-origin requests for security. We provide a public proxy to ", -1),
            t("a", {
              class: "hover:text-c-1 underline-offset-2",
              href: "https://en.wikipedia.org/wiki/Cross-origin_resource_sharing",
              target: "_blank"
            }, " bypass CORS issues ", -1),
            n(" . Check the ", -1),
            t("a", {
              class: "hover:text-c-1 underline-offset-2",
              href: "https://github.com/scalar/scalar/tree/main/examples/proxy-server",
              target: "_blank"
            }, " source code on GitHub ", -1),
            n(" . ", -1)
          ])]),
          default: l(() => [
            t("div", V, [
              i(r(m), {
                class: d(
                  r(y)(
                    r(f)({
                      active: s.proxyUrl === k
                    })
                  )
                ),
                onClick: e[0] || (e[0] = (o) => c("update:proxyUrl", k))
              }, {
                default: l(() => [
                  t("div", {
                    class: d(["flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1", {
                      "bg-c-accent text-b-1 border-transparent": s.proxyUrl === k
                    }])
                  }, [
                    s.proxyUrl === k ? (a(), u(r(b), {
                      key: 0,
                      icon: "Checkmark",
                      size: "xs",
                      thickness: "3.5"
                    })) : p("", !0)
                  ], 2),
                  e[6] || (e[6] = n(" Use proxy.scalar.com (default) ", -1))
                ]),
                _: 1
              }, 8, ["class"]),
              s.customProxyUrl && s.customProxyUrl !== k ? (a(), u(r(m), {
                key: 0,
                class: d(
                  r(y)(
                    r(f)({
                      active: s.proxyUrl === s.customProxyUrl
                    })
                  )
                ),
                onClick: e[1] || (e[1] = (o) => c("update:proxyUrl", s.customProxyUrl))
              }, {
                default: l(() => [
                  t("div", A, [
                    s.customProxyUrl === s.proxyUrl ? (a(), u(r(b), {
                      key: 0,
                      icon: "Checkmark",
                      size: "xs",
                      thickness: "3.5"
                    })) : p("", !0)
                  ]),
                  n(" Use custom proxy (" + v(s.customProxyUrl) + ") ", 1)
                ]),
                _: 1
              }, 8, ["class"])) : p("", !0),
              i(r(m), {
                class: d(r(y)(r(f)({ active: !s.proxyUrl }))),
                onClick: e[2] || (e[2] = (o) => c("update:proxyUrl", null))
              }, {
                default: l(() => [
                  t("div", {
                    class: d(["flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1", !s.proxyUrl && "bg-c-accent text-b-1 border-transparent"])
                  }, [
                    s.proxyUrl ? p("", !0) : (a(), u(r(b), {
                      key: 0,
                      icon: "Checkmark",
                      size: "xs",
                      thickness: "3.5"
                    }))
                  ], 2),
                  e[7] || (e[7] = n(" Skip the proxy ", -1))
                ]),
                _: 1
              }, 8, ["class"])
            ])
          ]),
          _: 1
        }),
        i(x, null, {
          title: l(() => [...e[8] || (e[8] = [
            n(" Themes ", -1)
          ])]),
          description: l(() => [...e[9] || (e[9] = [
            n(" We've got a whole rainbow of themes for you to play with: ", -1)
          ])]),
          default: l(() => [
            t("div", E, [
              t("div", F, [
                (a(), g(C, null, U(z, (o) => i(r(m), {
                  key: o,
                  class: d(
                    r(y)(
                      r(f)({
                        active: s.activeThemeId === o
                      })
                    )
                  ),
                  onClick: ($) => c("update:themeId", o)
                }, {
                  default: l(() => [
                    t("div", O, [
                      t("div", {
                        class: d(["flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1", {
                          "bg-c-accent text-b-1 border-transparent": s.activeThemeId === o
                        }])
                      }, [
                        s.activeThemeId === o ? (a(), u(r(b), {
                          key: 0,
                          icon: "Checkmark",
                          size: "xs",
                          thickness: "3.5"
                        })) : p("", !0)
                      ], 2),
                      n(" " + v(r(T)[o]), 1)
                    ]),
                    t("div", D, [
                      t("span", {
                        class: "border-c-3 -mr-3 inline-block h-5 w-5 rounded-full",
                        style: h({
                          backgroundColor: r(w)(o).light
                        })
                      }, null, 4),
                      t("span", {
                        class: "border-c-3 -mr-3 inline-block h-5 w-5 rounded-full",
                        style: h({
                          backgroundColor: r(w)(o).dark
                        })
                      }, null, 4),
                      t("span", {
                        class: "border-c-3 inline-block h-5 w-5 rounded-full",
                        style: h({
                          backgroundColor: r(w)(o).accent
                        })
                      }, null, 4)
                    ])
                  ]),
                  _: 2
                }, 1032, ["class", "onClick"])), 64))
              ])
            ])
          ]),
          _: 1
        }),
        i(x, null, {
          title: l(() => [...e[10] || (e[10] = [
            n(" Framework Themes ", -1)
          ])]),
          description: l(() => [...e[11] || (e[11] = [
            n(" Are you a real fan? Show your support by using your favorite framework's theme! ", -1)
          ])]),
          default: l(() => [
            t("div", W, [
              (a(), g(C, null, U(P, (o) => i(r(m), {
                key: o,
                class: d(
                  r(y)(
                    r(f)({
                      active: s.activeThemeId === o
                    })
                  )
                ),
                onClick: ($) => c("update:themeId", o)
              }, {
                default: l(() => [
                  t("div", _, [
                    t("div", {
                      class: d(["flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1", {
                        "bg-c-accent text-b-1 border-transparent": s.activeThemeId === o
                      }])
                    }, [
                      s.activeThemeId === o ? (a(), u(r(b), {
                        key: 0,
                        icon: "Checkmark",
                        size: "xs",
                        thickness: "3.5"
                      })) : p("", !0)
                    ], 2),
                    n(" " + v(r(T)[o]), 1)
                  ]),
                  t("div", q, [
                    t("div", G, [
                      i(B, { integration: o }, null, 8, ["integration"])
                    ])
                  ])
                ]),
                _: 2
              }, 1032, ["class", "onClick"])), 64))
            ])
          ]),
          _: 1
        }),
        i(x, null, {
          title: l(() => [...e[12] || (e[12] = [
            n(" Appearance ", -1)
          ])]),
          description: l(() => [...e[13] || (e[13] = [
            n(" Choose between light, dark, or system-based appearance for your workspace. ", -1)
          ])]),
          default: l(() => [
            i(L, {
              colorMode: s.colorMode,
              "onUpdate:colorMode": e[3] || (e[3] = (o) => c("update:colorMode", o))
            }, null, 8, ["colorMode"])
          ]),
          _: 1
        })
      ])
    ]));
  }
});
export {
  ee as default
};
